{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: matplotlib in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (3.6.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (1.4.4)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (4.38.0)\n",
            "Requirement already satisfied: pyparsing>=2.2.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (3.0.9)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (9.2.0)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (2.8.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (0.11.0)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (1.0.7)\n",
            "Requirement already satisfied: packaging>=20.0 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (23.0)\n",
            "Requirement already satisfied: numpy>=1.19 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from matplotlib) (1.23.5)\n",
            "Requirement already satisfied: six>=1.5 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)\n",
            "Note: you may need to restart the kernel to use updated packages.\n",
            "Requirement already satisfied: pandas in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (1.5.3)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pandas) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pandas) (2022.7.1)\n",
            "Requirement already satisfied: numpy>=1.20.3 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pandas) (1.23.5)\n",
            "Requirement already satisfied: six>=1.5 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)\n",
            "Note: you may need to restart the kernel to use updated packages.\n",
            "Requirement already satisfied: numpy in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (1.23.5)\n",
            "Note: you may need to restart the kernel to use updated packages.\n",
            "Collecting Bio\n",
            "  Using cached bio-1.5.9-py3-none-any.whl (276 kB)\n",
            "Requirement already satisfied: tqdm in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from Bio) (4.64.1)\n",
            "Requirement already satisfied: requests in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from Bio) (2.28.2)\n",
            "Collecting biopython>=1.80\n",
            "  Using cached biopython-1.81-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "Collecting mygene\n",
            "  Using cached mygene-3.2.2-py2.py3-none-any.whl (5.4 kB)\n",
            "Requirement already satisfied: pandas in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from Bio) (1.5.3)\n",
            "Collecting gprofiler-official\n",
            "  Using cached gprofiler_official-1.0.0-py3-none-any.whl (9.3 kB)\n",
            "Collecting pooch\n",
            "  Using cached pooch-1.7.0-py3-none-any.whl (60 kB)\n",
            "Requirement already satisfied: numpy in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from biopython>=1.80->Bio) (1.23.5)\n",
            "Collecting biothings-client>=0.2.6\n",
            "  Using cached biothings_client-0.3.0-py2.py3-none-any.whl (29 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pandas->Bio) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pandas->Bio) (2022.7.1)\n",
            "Requirement already satisfied: platformdirs>=2.5.0 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pooch->Bio) (3.0.0)\n",
            "Requirement already satisfied: packaging>=20.0 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from pooch->Bio) (23.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from requests->Bio) (2022.12.7)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from requests->Bio) (2.1.1)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from requests->Bio) (1.26.14)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from requests->Bio) (3.4)\n",
            "Requirement already satisfied: six>=1.5 in /opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages (from python-dateutil>=2.8.1->pandas->Bio) (1.16.0)\n",
            "Installing collected packages: biopython, pooch, gprofiler-official, biothings-client, mygene, Bio\n",
            "Successfully installed Bio-1.5.9 biopython-1.81 biothings-client-0.3.0 gprofiler-official-1.0.0 mygene-3.2.2 pooch-1.7.0\n",
            "Note: you may need to restart the kernel to use updated packages.\n"
          ]
        }
      ],
      "source": [
        "%pip install matplotlib\n",
        "%pip install pandas\n",
        "%pip install numpy\n",
        "%pip install Bio"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "mOhWHTBBmN6b"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import os\n",
        "import numpy as np\n",
        "from Bio import Entrez\n",
        "\n",
        "Entrez.email = 'gchiriaco@campus.unimib.it'\n",
        "\n",
        "import xml.etree.ElementTree as ET\n",
        "import os"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Y2xnakIEmN6l",
        "outputId": "7d4fc5b2-4c31-4c6c-e963-74e4c2d98e8a"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>sample_name</th>\n",
              "      <th>acid_reflux</th>\n",
              "      <th>acne_medication</th>\n",
              "      <th>acne_medication_otc</th>\n",
              "      <th>add_adhd</th>\n",
              "      <th>age_cat</th>\n",
              "      <th>age_corrected</th>\n",
              "      <th>age_years</th>\n",
              "      <th>alcohol_consumption</th>\n",
              "      <th>alcohol_frequency</th>\n",
              "      <th>...</th>\n",
              "      <th>vioscreen_zinc</th>\n",
              "      <th>vitamin_b_supplement_frequency</th>\n",
              "      <th>vitamin_d_supplement_frequency</th>\n",
              "      <th>vivid_dreams</th>\n",
              "      <th>weight_cat</th>\n",
              "      <th>weight_change</th>\n",
              "      <th>weight_kg</th>\n",
              "      <th>weight_units</th>\n",
              "      <th>whole_eggs</th>\n",
              "      <th>whole_grain_frequency</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>10317.000001000</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>false</td>\n",
              "      <td>false</td>\n",
              "      <td>Diagnosed by a medical professional (doctor, p...</td>\n",
              "      <td>60s</td>\n",
              "      <td>64.0</td>\n",
              "      <td>64</td>\n",
              "      <td>true</td>\n",
              "      <td>Daily</td>\n",
              "      <td>...</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Never</td>\n",
              "      <td>Regularly (3-5 times/week)</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Remained stable</td>\n",
              "      <td>52</td>\n",
              "      <td>kilograms</td>\n",
              "      <td>Never</td>\n",
              "      <td>Occasionally (1-2 times/week)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>10317.000001001</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>false</td>\n",
              "      <td>false</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>50s</td>\n",
              "      <td>53.0</td>\n",
              "      <td>53.0</td>\n",
              "      <td>true</td>\n",
              "      <td>Rarely (a few times/month)</td>\n",
              "      <td>...</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Remained stable</td>\n",
              "      <td>110.0</td>\n",
              "      <td>kilograms</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>10317.000001002</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>false</td>\n",
              "      <td>false</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>50s</td>\n",
              "      <td>53.0</td>\n",
              "      <td>53.0</td>\n",
              "      <td>true</td>\n",
              "      <td>Regularly (3-5 times/week)</td>\n",
              "      <td>...</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>56.0</td>\n",
              "      <td>kilograms</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>10317.000001004</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>false</td>\n",
              "      <td>false</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>40s</td>\n",
              "      <td>44.0</td>\n",
              "      <td>44.0</td>\n",
              "      <td>true</td>\n",
              "      <td>Rarely (a few times/month)</td>\n",
              "      <td>...</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Remained stable</td>\n",
              "      <td>86.0</td>\n",
              "      <td>kilograms</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>10317.000001008</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>false</td>\n",
              "      <td>false</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>60s</td>\n",
              "      <td>66.0</td>\n",
              "      <td>66.0</td>\n",
              "      <td>false</td>\n",
              "      <td>Never</td>\n",
              "      <td>...</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Increased more than 10 pounds</td>\n",
              "      <td>74.0</td>\n",
              "      <td>kilograms</td>\n",
              "      <td>Not provided</td>\n",
              "      <td>Not provided</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 661 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "       sample_name   acid_reflux acne_medication acne_medication_otc  \\\n",
              "0  10317.000001000  Not provided           false               false   \n",
              "1  10317.000001001  Not provided           false               false   \n",
              "2  10317.000001002  Not provided           false               false   \n",
              "3  10317.000001004  Not provided           false               false   \n",
              "4  10317.000001008  Not provided           false               false   \n",
              "\n",
              "                                            add_adhd age_cat age_corrected  \\\n",
              "0  Diagnosed by a medical professional (doctor, p...     60s          64.0   \n",
              "1                                       Not provided     50s          53.0   \n",
              "2                                       Not provided     50s          53.0   \n",
              "3                                       Not provided     40s          44.0   \n",
              "4                                       Not provided     60s          66.0   \n",
              "\n",
              "  age_years alcohol_consumption           alcohol_frequency  ...  \\\n",
              "0        64                true                       Daily  ...   \n",
              "1      53.0                true  Rarely (a few times/month)  ...   \n",
              "2      53.0                true  Regularly (3-5 times/week)  ...   \n",
              "3      44.0                true  Rarely (a few times/month)  ...   \n",
              "4      66.0               false                       Never  ...   \n",
              "\n",
              "  vioscreen_zinc vitamin_b_supplement_frequency  \\\n",
              "0   Not provided                          Never   \n",
              "1   Not provided                   Not provided   \n",
              "2   Not provided                   Not provided   \n",
              "3   Not provided                   Not provided   \n",
              "4   Not provided                   Not provided   \n",
              "\n",
              "  vitamin_d_supplement_frequency  vivid_dreams weight_cat  \\\n",
              "0     Regularly (3-5 times/week)  Not provided        NaN   \n",
              "1                   Not provided  Not provided        NaN   \n",
              "2                   Not provided  Not provided        NaN   \n",
              "3                   Not provided  Not provided        NaN   \n",
              "4                   Not provided  Not provided        NaN   \n",
              "\n",
              "                   weight_change weight_kg weight_units    whole_eggs  \\\n",
              "0                Remained stable        52    kilograms         Never   \n",
              "1                Remained stable     110.0    kilograms  Not provided   \n",
              "2                   Not provided      56.0    kilograms  Not provided   \n",
              "3                Remained stable      86.0    kilograms  Not provided   \n",
              "4  Increased more than 10 pounds      74.0    kilograms  Not provided   \n",
              "\n",
              "           whole_grain_frequency  \n",
              "0  Occasionally (1-2 times/week)  \n",
              "1                   Not provided  \n",
              "2                   Not provided  \n",
              "3                   Not provided  \n",
              "4                   Not provided  \n",
              "\n",
              "[5 rows x 661 columns]"
            ]
          },
          "execution_count": 3,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "data = pd.read_csv('sample_information_American_gut_microbiome.csv.txt', sep=\"\\t\", dtype=object)\n",
        "data.head()"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Change all the true/false values to Yes/No"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "data[data=='true']='Yes'\n",
        "data[data=='false']='No'\n",
        "data[data=='I do not have this condition']='No'"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v_malcSOmN6z",
        "outputId": "5b9d35a9-c266-4525-c1d9-fc84845463cd"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "['sample_name',\n",
              " 'acid_reflux',\n",
              " 'acne_medication',\n",
              " 'acne_medication_otc',\n",
              " 'add_adhd',\n",
              " 'age_cat',\n",
              " 'age_corrected',\n",
              " 'age_years',\n",
              " 'alcohol_consumption',\n",
              " 'alcohol_frequency',\n",
              " 'alcohol_types',\n",
              " 'alcohol_types_beercider',\n",
              " 'alcohol_types_red_wine',\n",
              " 'alcohol_types_sour_beers',\n",
              " 'alcohol_types_spiritshard_alcohol',\n",
              " 'alcohol_types_unspecified',\n",
              " 'alcohol_types_white_wine',\n",
              " 'allergic_to',\n",
              " 'allergic_to_i_have_no_food_allergies_that_i_know_of',\n",
              " 'allergic_to_other',\n",
              " 'allergic_to_peanuts',\n",
              " 'allergic_to_shellfish',\n",
              " 'allergic_to_tree_nuts',\n",
              " 'allergic_to_unspecified',\n",
              " 'altitude',\n",
              " 'alzheimers',\n",
              " 'animal_age',\n",
              " 'animal_free_text',\n",
              " 'animal_gender',\n",
              " 'animal_origin',\n",
              " 'animal_type',\n",
              " 'anonymized_name',\n",
              " 'antibiotic_history',\n",
              " 'appendix_removed',\n",
              " 'artificial_sweeteners',\n",
              " 'asd',\n",
              " 'assigned_from_geo',\n",
              " 'autoimmune',\n",
              " 'birth_year',\n",
              " 'bmi',\n",
              " 'bmi_cat',\n",
              " 'bmi_corrected',\n",
              " 'body_habitat',\n",
              " 'body_product',\n",
              " 'body_site',\n",
              " 'bowel_movement_frequency',\n",
              " 'bowel_movement_quality',\n",
              " 'breastmilk_formula_ensure',\n",
              " 'cancer',\n",
              " 'cancer_treatment',\n",
              " 'cardiovascular_disease',\n",
              " 'cat',\n",
              " 'cdiff',\n",
              " 'census_region',\n",
              " 'chickenpox',\n",
              " 'clinical_condition',\n",
              " 'collection_date',\n",
              " 'collection_date_only',\n",
              " 'collection_month',\n",
              " 'collection_season',\n",
              " 'collection_time',\n",
              " 'collection_timestamp',\n",
              " 'consume_animal_products_abx',\n",
              " 'contraceptive',\n",
              " 'coprophage',\n",
              " 'cosmetics_frequency',\n",
              " 'country',\n",
              " 'country_of_birth',\n",
              " 'country_residence',\n",
              " 'csection',\n",
              " 'deodorant_use',\n",
              " 'depression_bipolar_schizophrenia',\n",
              " 'depth',\n",
              " 'description',\n",
              " 'diabetes',\n",
              " 'diabetes_type',\n",
              " 'diet',\n",
              " 'diet_type',\n",
              " 'dna_extracted',\n",
              " 'dog',\n",
              " 'dominant_hand',\n",
              " 'drinking_water_source',\n",
              " 'drinks_per_session',\n",
              " 'economic_region',\n",
              " 'elevation',\n",
              " 'env_biome',\n",
              " 'env_feature',\n",
              " 'env_material',\n",
              " 'env_package',\n",
              " 'epilepsy_or_seizure_disorder',\n",
              " 'exercise_frequency',\n",
              " 'exercise_location',\n",
              " 'fed_as_infant',\n",
              " 'fermented_consumed',\n",
              " 'fermented_consumed_beer',\n",
              " 'fermented_consumed_chicha',\n",
              " 'fermented_consumed_cider',\n",
              " 'fermented_consumed_cottage_cheese',\n",
              " 'fermented_consumed_fermented_beansmisonatto',\n",
              " 'fermented_consumed_fermented_breadsourdoughinjera',\n",
              " 'fermented_consumed_fermented_fish',\n",
              " 'fermented_consumed_fermented_tofu',\n",
              " 'fermented_consumed_fish_sauce',\n",
              " 'fermented_consumed_kefir_milk',\n",
              " 'fermented_consumed_kefir_water',\n",
              " 'fermented_consumed_kimchi',\n",
              " 'fermented_consumed_kombucha',\n",
              " 'fermented_consumed_mead',\n",
              " 'fermented_consumed_other',\n",
              " 'fermented_consumed_pickled_vegetables',\n",
              " 'fermented_consumed_sauerkraut',\n",
              " 'fermented_consumed_sour_creamcreme_fraiche',\n",
              " 'fermented_consumed_tempeh',\n",
              " 'fermented_consumed_unspecified',\n",
              " 'fermented_consumed_wine',\n",
              " 'fermented_consumed_yogurtlassi',\n",
              " 'fermented_frequency',\n",
              " 'fermented_increased',\n",
              " 'fermented_other',\n",
              " 'fermented_plant_frequency',\n",
              " 'fermented_produce_commercial',\n",
              " 'fermented_produce_commercial_beer',\n",
              " 'fermented_produce_commercial_chicha',\n",
              " 'fermented_produce_commercial_cider',\n",
              " 'fermented_produce_commercial_cottage_cheese',\n",
              " 'fermented_produce_commercial_fermented_beansmisonatto',\n",
              " 'fermented_produce_commercial_fermented_breadsourdoughinjera',\n",
              " 'fermented_produce_commercial_fermented_fish',\n",
              " 'fermented_produce_commercial_fermented_tofu',\n",
              " 'fermented_produce_commercial_fish_sauce',\n",
              " 'fermented_produce_commercial_kefir_milk',\n",
              " 'fermented_produce_commercial_kefir_water',\n",
              " 'fermented_produce_commercial_kimchi',\n",
              " 'fermented_produce_commercial_kombucha',\n",
              " 'fermented_produce_commercial_mead',\n",
              " 'fermented_produce_commercial_other',\n",
              " 'fermented_produce_commercial_pickled_vegetables',\n",
              " 'fermented_produce_commercial_sauerkraut',\n",
              " 'fermented_produce_commercial_sour_creamcreme_fraiche',\n",
              " 'fermented_produce_commercial_tempeh',\n",
              " 'fermented_produce_commercial_unspecified',\n",
              " 'fermented_produce_commercial_wine',\n",
              " 'fermented_produce_commercial_yogurtlassi',\n",
              " 'fermented_produce_personal',\n",
              " 'fermented_produce_personal_beer',\n",
              " 'fermented_produce_personal_chicha',\n",
              " 'fermented_produce_personal_cider',\n",
              " 'fermented_produce_personal_cottage_cheese',\n",
              " 'fermented_produce_personal_fermented_beansmisonatto',\n",
              " 'fermented_produce_personal_fermented_breadsourdoughinjera',\n",
              " 'fermented_produce_personal_fermented_fish',\n",
              " 'fermented_produce_personal_fermented_tofu',\n",
              " 'fermented_produce_personal_fish_sauce',\n",
              " 'fermented_produce_personal_kefir_milk',\n",
              " 'fermented_produce_personal_kefir_water',\n",
              " 'fermented_produce_personal_kimchi',\n",
              " 'fermented_produce_personal_kombucha',\n",
              " 'fermented_produce_personal_mead',\n",
              " 'fermented_produce_personal_other',\n",
              " 'fermented_produce_personal_pickled_vegetables',\n",
              " 'fermented_produce_personal_sauerkraut',\n",
              " 'fermented_produce_personal_sour_creamcreme_fraiche',\n",
              " 'fermented_produce_personal_tempeh',\n",
              " 'fermented_produce_personal_unspecified',\n",
              " 'fermented_produce_personal_wine',\n",
              " 'fermented_produce_personal_yogurtlassi',\n",
              " 'flossing_frequency',\n",
              " 'flu_vaccine_date',\n",
              " 'food_source',\n",
              " 'food_source_human_food',\n",
              " 'food_source_pet_store_food',\n",
              " 'food_source_unspecified',\n",
              " 'food_source_wild_food',\n",
              " 'food_special',\n",
              " 'food_special_grain_free',\n",
              " 'food_special_organic',\n",
              " 'food_special_unspecified',\n",
              " 'food_type',\n",
              " 'frozen_dessert_frequency',\n",
              " 'fruit_frequency',\n",
              " 'fungal_overgrowth',\n",
              " 'geo_loc_name',\n",
              " 'gluten',\n",
              " 'has_physical_specimen',\n",
              " 'height_cm',\n",
              " 'height_units',\n",
              " 'high_fat_red_meat_frequency',\n",
              " 'homecooked_meals_frequency',\n",
              " 'host',\n",
              " 'host_common_name',\n",
              " 'host_subject_id',\n",
              " 'host_taxid',\n",
              " 'hours_outside',\n",
              " 'humans_free_text',\n",
              " 'ibd',\n",
              " 'ibd_diagnosis',\n",
              " 'ibd_diagnosis_refined',\n",
              " 'ibs',\n",
              " 'kidney_disease',\n",
              " 'lactose',\n",
              " 'last_move',\n",
              " 'last_travel',\n",
              " 'latitude',\n",
              " 'level_of_education',\n",
              " 'liver_disease',\n",
              " 'living_status',\n",
              " 'livingwith',\n",
              " 'longitude',\n",
              " 'lowgrain_diet_type',\n",
              " 'lung_disease',\n",
              " 'meat_eggs_frequency',\n",
              " 'mental_illness',\n",
              " 'mental_illness_type',\n",
              " 'mental_illness_type_anorexia_nervosa',\n",
              " 'mental_illness_type_bipolar_disorder',\n",
              " 'mental_illness_type_bulimia_nervosa',\n",
              " 'mental_illness_type_depression',\n",
              " 'mental_illness_type_ptsd_posttraumatic_stress_disorder',\n",
              " 'mental_illness_type_schizophrenia',\n",
              " 'mental_illness_type_substance_abuse',\n",
              " 'mental_illness_type_unspecified',\n",
              " 'migraine',\n",
              " 'milk_cheese_frequency',\n",
              " 'milk_substitute_frequency',\n",
              " 'multivitamin',\n",
              " 'nail_biter',\n",
              " 'name',\n",
              " 'non_food_allergies',\n",
              " 'non_food_allergies_beestings',\n",
              " 'non_food_allergies_drug_eg_penicillin',\n",
              " 'non_food_allergies_pet_dander',\n",
              " 'non_food_allergies_poison_ivyoak',\n",
              " 'non_food_allergies_sun',\n",
              " 'non_food_allergies_unspecified',\n",
              " 'olive_oil',\n",
              " 'one_liter_of_water_a_day_frequency',\n",
              " 'other_animals_free_text',\n",
              " 'other_supplement_frequency',\n",
              " 'pets_other',\n",
              " 'pets_other_freetext',\n",
              " 'physical_specimen_location',\n",
              " 'physical_specimen_remaining',\n",
              " 'pku',\n",
              " 'pm_age',\n",
              " 'pm_concern',\n",
              " 'pm_education',\n",
              " 'pm_email',\n",
              " 'pm_ethnicity',\n",
              " 'pm_experience_worth_it',\n",
              " 'pm_gastro_problems',\n",
              " 'pm_gastro_problems_crohns_disease_or_ulcerative_colitis',\n",
              " 'pm_gastro_problems_frequent_more_than_once_a_week_constipation',\n",
              " 'pm_gastro_problems_frequent_more_than_once_a_week_diarrhea',\n",
              " 'pm_gastro_problems_gastrointestinal_cancer',\n",
              " 'pm_gastro_problems_i_have_had_surgery_on_my_intestines',\n",
              " 'pm_gastro_problems_irritable_bowel_syndrome_ibs',\n",
              " 'pm_gastro_problems_other',\n",
              " 'pm_gastro_problems_unspecified',\n",
              " 'pm_gender',\n",
              " 'pm_health',\n",
              " 'pm_lifestyle_change',\n",
              " 'pm_lifestyle_change_how',\n",
              " 'pm_lifestyle_change_how_change_in_alcohol_or_tobacco_use_ie_starting_stopping_or_changing_amount_consumed',\n",
              " 'pm_lifestyle_change_how_change_in_diet_ie_taking_a_probiotic',\n",
              " 'pm_lifestyle_change_how_change_in_exercise',\n",
              " 'pm_lifestyle_change_how_change_in_physical_environment_ie_adopting_or_getting_rid_of_a_pet_cleaning_more_or_less_often',\n",
              " 'pm_lifestyle_change_how_change_in_use_of_nutritional_supplements_or_vitamins',\n",
              " 'pm_lifestyle_change_how_other',\n",
              " 'pm_lifestyle_change_how_unspecified',\n",
              " 'pm_name',\n",
              " 'pm_participation_reason',\n",
              " 'pm_participation_reason_desire_to_improve_my_health',\n",
              " 'pm_participation_reason_general_curiosityinterest_in_learning_about_my_microbiome',\n",
              " 'pm_participation_reason_i_have_gastrointestinal_problems',\n",
              " 'pm_participation_reason_i_have_other_microbiomerelevant_health_problems',\n",
              " 'pm_participation_reason_interest_in_contributing_to_science',\n",
              " 'pm_participation_reason_other',\n",
              " 'pm_participation_reason_people_in_my_family_or_social_network_have_done_it',\n",
              " 'pm_participation_reason_professional_interest_in_the_microbiome',\n",
              " 'pm_participation_reason_the_study_seemed_fun_and_entertaining',\n",
              " 'pm_participation_reason_unspecified',\n",
              " 'pm_recontact',\n",
              " 'pm_shared',\n",
              " 'pm_shared_pcp',\n",
              " 'pm_shared_pcp_changes',\n",
              " 'pm_shared_pcp_changes_what',\n",
              " 'pm_shared_pcp_changes_what_change_in_behaviorallifestyle_recommendations_eg_start_probiotic',\n",
              " 'pm_shared_pcp_changes_what_change_in_medication_ie_starting_stopping_or_changing_the_dose_of_a_prescription',\n",
              " 'pm_shared_pcp_changes_what_ordered_additional_labs_tests_or_procedures',\n",
              " 'pm_shared_pcp_changes_what_other',\n",
              " 'pm_shared_pcp_changes_what_referral_to_a_specialist_eg_gastroenterologist',\n",
              " 'pm_shared_pcp_changes_what_unspecified',\n",
              " 'pm_shared_who',\n",
              " 'pm_shared_who_colleagues',\n",
              " 'pm_shared_who_family_members',\n",
              " 'pm_shared_who_friends',\n",
              " 'pm_shared_who_nutritionistdietician',\n",
              " 'pm_shared_who_other',\n",
              " 'pm_shared_who_other_medical_or_health_professional',\n",
              " 'pm_shared_who_posteddiscussed_on_an_online_patienthealth_platform_ie_patientslikeme',\n",
              " 'pm_shared_who_posteddiscussed_on_data_sharing_platform_ie_open_humans',\n",
              " 'pm_shared_who_posteddiscussed_on_social_networking_site_ie_facebook',\n",
              " 'pm_shared_who_primary_care_physician',\n",
              " 'pm_shared_who_specialty_physician_eg_gastroenterologist',\n",
              " 'pm_shared_who_unspecified',\n",
              " 'pm_understand_results',\n",
              " 'pm_useful',\n",
              " 'pool_frequency',\n",
              " 'poultry_frequency',\n",
              " 'pregnant',\n",
              " 'prepared_meals_frequency',\n",
              " 'probiotic_frequency',\n",
              " 'public',\n",
              " 'qiita_empo_1',\n",
              " 'qiita_empo_2',\n",
              " 'qiita_empo_3',\n",
              " 'qiita_study_id',\n",
              " 'race',\n",
              " 'ready_to_eat_meals_frequency',\n",
              " 'red_meat_frequency',\n",
              " 'roommates',\n",
              " 'roommates_in_study',\n",
              " 'salted_snacks_frequency',\n",
              " 'sample_type',\n",
              " 'scientific_name',\n",
              " 'seafood_frequency',\n",
              " 'seasonal_allergies',\n",
              " 'setting',\n",
              " 'sex',\n",
              " 'sibo',\n",
              " 'skin_condition',\n",
              " 'sleep_duration',\n",
              " 'smoking_frequency',\n",
              " 'softener',\n",
              " 'specialized_diet',\n",
              " 'specialized_diet_exclude_dairy',\n",
              " 'specialized_diet_exclude_nightshades',\n",
              " 'specialized_diet_exclude_refined_sugars',\n",
              " 'specialized_diet_fodmap',\n",
              " 'specialized_diet_halaal',\n",
              " 'specialized_diet_i_do_not_eat_a_specialized_diet',\n",
              " 'specialized_diet_kosher',\n",
              " 'specialized_diet_modified_paleo_diet',\n",
              " 'specialized_diet_other_restrictions_not_described_here',\n",
              " 'specialized_diet_paleodiet_or_primal_diet',\n",
              " 'specialized_diet_raw_food_diet',\n",
              " 'specialized_diet_unspecified',\n",
              " 'specialized_diet_westenprice_or_other_lowgrain_low_processed_fo',\n",
              " 'specialized_diet_westenprice_or_other_lowgrain_low_processed_food_diet',\n",
              " 'state',\n",
              " 'subset_age',\n",
              " 'subset_antibiotic_history',\n",
              " 'subset_bmi',\n",
              " 'subset_diabetes',\n",
              " 'subset_healthy',\n",
              " 'subset_ibd',\n",
              " 'sugar_sweetened_drink_frequency',\n",
              " 'sugary_sweets_frequency',\n",
              " 'surf_board_type',\n",
              " 'surf_frequency',\n",
              " 'surf_loal_break_frequency',\n",
              " 'surf_local_break',\n",
              " 'surf_shower_frequency',\n",
              " 'surf_stance',\n",
              " 'surf_sunscreen',\n",
              " 'surf_sunscreen_frequency',\n",
              " 'surf_travel_distance',\n",
              " 'surf_travel_frequency',\n",
              " 'surf_wax',\n",
              " 'surf_weetsuit',\n",
              " 'survey_id',\n",
              " 'taxon_id',\n",
              " 'teethbrushing_frequency',\n",
              " 'thyroid',\n",
              " 'title',\n",
              " 'toilet_water_access',\n",
              " 'tonsils_removed',\n",
              " 'types_of_plants',\n",
              " 'vegetable_frequency',\n",
              " 'vioscreen_a_bev',\n",
              " 'vioscreen_a_cal',\n",
              " 'vioscreen_acesupot',\n",
              " 'vioscreen_activity_level',\n",
              " 'vioscreen_add_sug',\n",
              " 'vioscreen_addsugar',\n",
              " 'vioscreen_adsugtot',\n",
              " 'vioscreen_age',\n",
              " 'vioscreen_alanine',\n",
              " 'vioscreen_alcohol',\n",
              " 'vioscreen_alcohol_servings',\n",
              " 'vioscreen_alphacar',\n",
              " 'vioscreen_alphtoce',\n",
              " 'vioscreen_alphtoco',\n",
              " 'vioscreen_arginine',\n",
              " 'vioscreen_ash',\n",
              " 'vioscreen_aspartam',\n",
              " 'vioscreen_aspartic',\n",
              " 'vioscreen_avcarb',\n",
              " 'vioscreen_bcodeid',\n",
              " 'vioscreen_betacar',\n",
              " 'vioscreen_betacryp',\n",
              " 'vioscreen_betaine',\n",
              " 'vioscreen_betatoco',\n",
              " 'vioscreen_biochana',\n",
              " 'vioscreen_bmi',\n",
              " 'vioscreen_caffeine',\n",
              " 'vioscreen_calcium',\n",
              " 'vioscreen_calcium_avg',\n",
              " 'vioscreen_calcium_dose',\n",
              " 'vioscreen_calcium_freq',\n",
              " 'vioscreen_calcium_from_dairy_servings',\n",
              " 'vioscreen_calcium_servings',\n",
              " 'vioscreen_calories',\n",
              " 'vioscreen_carbo',\n",
              " 'vioscreen_cholest',\n",
              " 'vioscreen_choline',\n",
              " 'vioscreen_clac9t11',\n",
              " 'vioscreen_clat10c12',\n",
              " 'vioscreen_copper',\n",
              " 'vioscreen_coumest',\n",
              " 'vioscreen_cystine',\n",
              " 'vioscreen_d_cheese',\n",
              " 'vioscreen_d_milk',\n",
              " 'vioscreen_d_tot_soym',\n",
              " 'vioscreen_d_total',\n",
              " 'vioscreen_d_yogurt',\n",
              " 'vioscreen_daidzein',\n",
              " 'vioscreen_database',\n",
              " 'vioscreen_delttoco',\n",
              " 'vioscreen_discfat_oil',\n",
              " 'vioscreen_discfat_sol',\n",
              " 'vioscreen_dob',\n",
              " 'vioscreen_eer',\n",
              " 'vioscreen_email',\n",
              " 'vioscreen_erythr',\n",
              " 'vioscreen_f_citmlb',\n",
              " 'vioscreen_f_nj_citmlb',\n",
              " 'vioscreen_f_nj_other',\n",
              " 'vioscreen_f_nj_total',\n",
              " 'vioscreen_f_other',\n",
              " 'vioscreen_f_total',\n",
              " 'vioscreen_fat',\n",
              " 'vioscreen_fiber',\n",
              " 'vioscreen_fibh2o',\n",
              " 'vioscreen_fibinso',\n",
              " 'vioscreen_finished',\n",
              " 'vioscreen_fish_servings',\n",
              " 'vioscreen_fol_deqv',\n",
              " 'vioscreen_fol_nat',\n",
              " 'vioscreen_fol_syn',\n",
              " 'vioscreen_formontn',\n",
              " 'vioscreen_fried_fish_servings',\n",
              " 'vioscreen_fried_food_servings',\n",
              " 'vioscreen_frt5_day',\n",
              " 'vioscreen_frtsumm',\n",
              " 'vioscreen_fructose',\n",
              " 'vioscreen_fruit_servings',\n",
              " 'vioscreen_g_nwhl',\n",
              " 'vioscreen_g_total',\n",
              " 'vioscreen_g_whl',\n",
              " 'vioscreen_galactos',\n",
              " 'vioscreen_gammtoco',\n",
              " 'vioscreen_gender',\n",
              " 'vioscreen_genistn',\n",
              " 'vioscreen_glac',\n",
              " 'vioscreen_gltc',\n",
              " 'vioscreen_glucose',\n",
              " 'vioscreen_glutamic',\n",
              " 'vioscreen_glycine',\n",
              " 'vioscreen_glycitn',\n",
              " 'vioscreen_grams',\n",
              " 'vioscreen_hei2010_dairy',\n",
              " 'vioscreen_hei2010_empty_calories',\n",
              " 'vioscreen_hei2010_fatty_acids',\n",
              " 'vioscreen_hei2010_fruit',\n",
              " 'vioscreen_hei2010_greens_beans',\n",
              " 'vioscreen_hei2010_protien_foods',\n",
              " 'vioscreen_hei2010_refined_grains',\n",
              " 'vioscreen_hei2010_score',\n",
              " 'vioscreen_hei2010_sea_foods_plant_protiens',\n",
              " 'vioscreen_hei2010_sodium',\n",
              " 'vioscreen_hei2010_veg',\n",
              " 'vioscreen_hei2010_whole_fruit',\n",
              " 'vioscreen_hei2010_whole_grains',\n",
              " 'vioscreen_hei_drk_g_org_veg_leg',\n",
              " 'vioscreen_hei_fruit',\n",
              " 'vioscreen_hei_grains',\n",
              " 'vioscreen_hei_meat_beans',\n",
              " 'vioscreen_hei_milk',\n",
              " 'vioscreen_hei_non_juice_frt',\n",
              " 'vioscreen_hei_oils',\n",
              " 'vioscreen_hei_sat_fat',\n",
              " 'vioscreen_hei_score',\n",
              " 'vioscreen_hei_sodium',\n",
              " 'vioscreen_hei_sol_fat_alc_add_sug',\n",
              " 'vioscreen_hei_veg',\n",
              " 'vioscreen_hei_whl_grains',\n",
              " 'vioscreen_height',\n",
              " 'vioscreen_histidin',\n",
              " 'vioscreen_inositol',\n",
              " 'vioscreen_iron',\n",
              " 'vioscreen_isoleuc',\n",
              " 'vioscreen_isomalt',\n",
              " 'vioscreen_joules',\n",
              " 'vioscreen_juice_servings',\n",
              " 'vioscreen_lactitol',\n",
              " 'vioscreen_lactose',\n",
              " 'vioscreen_legumes',\n",
              " 'vioscreen_leucine',\n",
              " 'vioscreen_line_gi',\n",
              " 'vioscreen_low_fat_dairy_serving',\n",
              " 'vioscreen_lutzeax',\n",
              " 'vioscreen_lycopene',\n",
              " 'vioscreen_lysine',\n",
              " 'vioscreen_m_egg',\n",
              " 'vioscreen_m_fish_hi',\n",
              " 'vioscreen_m_fish_lo',\n",
              " 'vioscreen_m_frank',\n",
              " 'vioscreen_m_meat',\n",
              " 'vioscreen_m_mpf',\n",
              " 'vioscreen_m_nutsd',\n",
              " 'vioscreen_m_organ',\n",
              " 'vioscreen_m_poult',\n",
              " 'vioscreen_m_soy',\n",
              " 'vioscreen_magnes',\n",
              " 'vioscreen_maltitol',\n",
              " 'vioscreen_maltose',\n",
              " 'vioscreen_mangan',\n",
              " 'vioscreen_mannitol',\n",
              " 'vioscreen_methhis3',\n",
              " 'vioscreen_methion',\n",
              " 'vioscreen_mfa141',\n",
              " 'vioscreen_mfa161',\n",
              " 'vioscreen_mfa181',\n",
              " 'vioscreen_mfa201',\n",
              " 'vioscreen_mfa221',\n",
              " 'vioscreen_mfatot',\n",
              " 'vioscreen_multi_calcium_avg',\n",
              " 'vioscreen_multi_calcium_dose',\n",
              " 'vioscreen_multivitamin',\n",
              " 'vioscreen_multivitamin_freq',\n",
              " 'vioscreen_natoco',\n",
              " 'vioscreen_nccglbr',\n",
              " 'vioscreen_nccglgr',\n",
              " 'vioscreen_niacin',\n",
              " 'vioscreen_niacineq',\n",
              " 'vioscreen_nitrogen',\n",
              " 'vioscreen_non_fried_fish_servings',\n",
              " 'vioscreen_nutrient_recommendation',\n",
              " 'vioscreen_omega3',\n",
              " 'vioscreen_oxalic',\n",
              " 'vioscreen_oxalicm',\n",
              " 'vioscreen_pantothe',\n",
              " 'vioscreen_pectins',\n",
              " 'vioscreen_pfa182',\n",
              " 'vioscreen_pfa183',\n",
              " 'vioscreen_pfa184',\n",
              " 'vioscreen_pfa204',\n",
              " 'vioscreen_pfa205',\n",
              " 'vioscreen_pfa225',\n",
              " 'vioscreen_pfa226',\n",
              " 'vioscreen_pfatot',\n",
              " 'vioscreen_phenylal',\n",
              " 'vioscreen_phosphor',\n",
              " 'vioscreen_phytic',\n",
              " 'vioscreen_pinitol',\n",
              " 'vioscreen_potass',\n",
              " 'vioscreen_procdate',\n",
              " 'vioscreen_proline',\n",
              " 'vioscreen_protanim',\n",
              " 'vioscreen_protein',\n",
              " 'vioscreen_protocol',\n",
              " 'vioscreen_protveg',\n",
              " 'vioscreen_questionnaire',\n",
              " 'vioscreen_recno',\n",
              " 'vioscreen_retinol',\n",
              " 'vioscreen_rgrain',\n",
              " 'vioscreen_ribofla',\n",
              " 'vioscreen_sacchar',\n",
              " 'vioscreen_salad_vegetable_servings',\n",
              " 'vioscreen_satoco',\n",
              " 'vioscreen_scf',\n",
              " 'vioscreen_scfv',\n",
              " 'vioscreen_selenium',\n",
              " 'vioscreen_serine',\n",
              " 'vioscreen_sfa100',\n",
              " 'vioscreen_sfa120',\n",
              " 'vioscreen_sfa140',\n",
              " 'vioscreen_sfa160',\n",
              " 'vioscreen_sfa170',\n",
              " 'vioscreen_sfa180',\n",
              " 'vioscreen_sfa200',\n",
              " 'vioscreen_sfa220',\n",
              " 'vioscreen_sfa40',\n",
              " 'vioscreen_sfa60',\n",
              " 'vioscreen_sfa80',\n",
              " 'vioscreen_sfatot',\n",
              " 'vioscreen_sodium',\n",
              " 'vioscreen_sorbitol',\n",
              " 'vioscreen_srvid',\n",
              " 'vioscreen_starch',\n",
              " 'vioscreen_started',\n",
              " 'vioscreen_subject_id',\n",
              " 'vioscreen_sucpoly',\n",
              " 'vioscreen_sucrlose',\n",
              " 'vioscreen_sucrose',\n",
              " 'vioscreen_sweet_servings',\n",
              " 'vioscreen_tagatose',\n",
              " 'vioscreen_tfa161t',\n",
              " 'vioscreen_tfa181t',\n",
              " 'vioscreen_tfa182t',\n",
              " 'vioscreen_tgrain',\n",
              " 'vioscreen_thiamin',\n",
              " 'vioscreen_threonin',\n",
              " 'vioscreen_time',\n",
              " 'vioscreen_totaltfa',\n",
              " 'vioscreen_totcla',\n",
              " 'vioscreen_totfolat',\n",
              " 'vioscreen_totsugar',\n",
              " 'vioscreen_tryptoph',\n",
              " 'vioscreen_tyrosine',\n",
              " 'vioscreen_user_id',\n",
              " 'vioscreen_v_drkgr',\n",
              " 'vioscreen_v_orange',\n",
              " 'vioscreen_v_other',\n",
              " 'vioscreen_v_potato',\n",
              " 'vioscreen_v_starcy',\n",
              " 'vioscreen_v_tomato',\n",
              " 'vioscreen_v_total',\n",
              " 'vioscreen_valine',\n",
              " 'vioscreen_veg5_day',\n",
              " 'vioscreen_vegetable_servings',\n",
              " 'vioscreen_vegsumm',\n",
              " 'vioscreen_visit',\n",
              " 'vioscreen_vita_iu',\n",
              " 'vioscreen_vita_rae',\n",
              " 'vioscreen_vita_re',\n",
              " 'vioscreen_vitb12',\n",
              " 'vioscreen_vitb6',\n",
              " 'vioscreen_vitc',\n",
              " 'vioscreen_vitd',\n",
              " 'vioscreen_vitd2',\n",
              " 'vioscreen_vitd3',\n",
              " 'vioscreen_vitd_iu',\n",
              " 'vioscreen_vite_iu',\n",
              " 'vioscreen_vitk',\n",
              " 'vioscreen_water',\n",
              " 'vioscreen_weight',\n",
              " 'vioscreen_wgrain',\n",
              " 'vioscreen_whole_grain_servings',\n",
              " 'vioscreen_xylitol',\n",
              " 'vioscreen_zinc',\n",
              " 'vitamin_b_supplement_frequency',\n",
              " 'vitamin_d_supplement_frequency',\n",
              " 'vivid_dreams',\n",
              " 'weight_cat',\n",
              " 'weight_change',\n",
              " 'weight_kg',\n",
              " 'weight_units',\n",
              " 'whole_eggs',\n",
              " 'whole_grain_frequency']"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "list(data.columns)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Selecting only the samples from feces"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "dexCFx6Lz_Y0"
      },
      "outputs": [],
      "source": [
        "\n",
        "data = data.loc[data.body_site == \"UBERON:feces\",:]"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Removing all the subjects with special diseases and conditions"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [],
      "source": [
        "cond_no_disease = ( (data['alcohol_frequency']!='Daily') &\n",
        "                    ((data['autoimmune']=='No') | (data['autoimmune']=='Not provided')) &\n",
        "                    ((data['cancer']=='No') | (data['cancer']=='Not provided')) &\n",
        "                    ((data['cdiff']=='No') | (data['cdiff']=='Not provided')) & #Clostridioides difficile\n",
        "                    ((data['diabetes']=='No') | (data['diabetes']=='Not provided')) &\n",
        "                    ((data['ibd']=='No') | (data['ibd']=='Not provided')) & #inflammatory bowel disease\n",
        "                    ((data['ibs']=='No') | (data['ibs']=='Not provided')) & #inflamatiry bowel syndrome\n",
        "                    ((data['kidney_disease']=='No') | (data['kidney_disease']=='Not provided')) &\n",
        "                    ((data['liver_disease']=='No') | (data['liver_disease']=='Not provided')) &\n",
        "                    ((data['mental_illness_type_anorexia_nervosa']=='No') | (data['mental_illness_type_anorexia_nervosa']=='Not provided')) &\n",
        "                    ((data['mental_illness_type_bulimia_nervosa']=='No') | (data['mental_illness_type_bulimia_nervosa']=='Not provided')) &\n",
        "                    ((data['sibo']=='No') | (data['sibo']=='Not provided'))) #small intestinal bacterial overgrowth\n",
        "\n",
        "data_no_disease = data.loc[cond_no_disease, :]"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Number of subjects that removed refined sugars from their diet"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NgUfOVBWmk8X",
        "outputId": "c0934532-c3c2-4869-cb0f-268953267c81"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Not provided    7417\n",
              "No              4633\n",
              "Yes              633\n",
              "Unspecified       22\n",
              "Name: specialized_diet_exclude_refined_sugars, dtype: int64"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "data_no_disease.specialized_diet_exclude_refined_sugars.value_counts()"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Selecting only Male/Females and subject with age between 20s and 60s"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "30s             2359\n",
              "40s             2288\n",
              "50s             2221\n",
              "60s             1735\n",
              "20s             1352\n",
              "Not provided     748\n",
              "child            733\n",
              "70+              594\n",
              "teen             382\n",
              "Unspecified      287\n",
              "baby               6\n",
              "Name: age_cat, dtype: int64"
            ]
          },
          "execution_count": 9,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "data_no_disease.age_cat.value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "female          6178\n",
              "male            5976\n",
              "Not provided     526\n",
              "unspecified       18\n",
              "other              7\n",
              "Name: sex, dtype: int64"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "data_no_disease.sex.value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "id": "D7hMR2IWmN70"
      },
      "outputs": [],
      "source": [
        "data_f = data_no_disease.loc[((data_no_disease['age_cat'] == '20s') |\n",
        "                          (data_no_disease['age_cat'] == '30s') |\n",
        "                          (data_no_disease['age_cat'] == '40s') |\n",
        "                          (data_no_disease['age_cat'] == '50s')|\n",
        "                          (data_no_disease['age_cat'] == '60s'))  &\n",
        "                          ((data_no_disease.sex == 'female') | (data_no_disease.sex == 'male')) ,:]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 175
        },
        "id": "0ALggz4cmN7M",
        "outputId": "c488a39d-494d-4561-e599-dd5e648efead"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th>specialized_diet_exclude_refined_sugars</th>\n",
              "      <th>No</th>\n",
              "      <th>Not provided</th>\n",
              "      <th>Unspecified</th>\n",
              "      <th>Yes</th>\n",
              "      <th>All</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>age_cat</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>20s</th>\n",
              "      <td>573.0</td>\n",
              "      <td>698.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>59.0</td>\n",
              "      <td>1330</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30s</th>\n",
              "      <td>869.0</td>\n",
              "      <td>1295.0</td>\n",
              "      <td>3.0</td>\n",
              "      <td>128.0</td>\n",
              "      <td>2295</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40s</th>\n",
              "      <td>817.0</td>\n",
              "      <td>1330.0</td>\n",
              "      <td>3.0</td>\n",
              "      <td>105.0</td>\n",
              "      <td>2255</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>50s</th>\n",
              "      <td>771.0</td>\n",
              "      <td>1284.0</td>\n",
              "      <td>6.0</td>\n",
              "      <td>101.0</td>\n",
              "      <td>2162</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>60s</th>\n",
              "      <td>557.0</td>\n",
              "      <td>1026.0</td>\n",
              "      <td>4.0</td>\n",
              "      <td>92.0</td>\n",
              "      <td>1679</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>All</th>\n",
              "      <td>3587.0</td>\n",
              "      <td>5633.0</td>\n",
              "      <td>16.0</td>\n",
              "      <td>485.0</td>\n",
              "      <td>9721</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "specialized_diet_exclude_refined_sugars      No  Not provided  Unspecified  \\\n",
              "age_cat                                                                      \n",
              "20s                                       573.0         698.0          NaN   \n",
              "30s                                       869.0        1295.0          3.0   \n",
              "40s                                       817.0        1330.0          3.0   \n",
              "50s                                       771.0        1284.0          6.0   \n",
              "60s                                       557.0        1026.0          4.0   \n",
              "All                                      3587.0        5633.0         16.0   \n",
              "\n",
              "specialized_diet_exclude_refined_sugars    Yes   All  \n",
              "age_cat                                               \n",
              "20s                                       59.0  1330  \n",
              "30s                                      128.0  2295  \n",
              "40s                                      105.0  2255  \n",
              "50s                                      101.0  2162  \n",
              "60s                                       92.0  1679  \n",
              "All                                      485.0  9721  "
            ]
          },
          "execution_count": 12,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "pd.crosstab(data_f.age_cat,\n",
        "            data_f.specialized_diet_exclude_refined_sugars, \n",
        "            values=data_f.sample_name,\n",
        "            aggfunc=len, \n",
        "            margins = True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th>specialized_diet_exclude_refined_sugars</th>\n",
              "      <th>No</th>\n",
              "      <th>Not provided</th>\n",
              "      <th>Unspecified</th>\n",
              "      <th>Yes</th>\n",
              "      <th>All</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>sex</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>female</th>\n",
              "      <td>1824</td>\n",
              "      <td>2945</td>\n",
              "      <td>12</td>\n",
              "      <td>250</td>\n",
              "      <td>5031</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>male</th>\n",
              "      <td>1763</td>\n",
              "      <td>2688</td>\n",
              "      <td>4</td>\n",
              "      <td>235</td>\n",
              "      <td>4690</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>All</th>\n",
              "      <td>3587</td>\n",
              "      <td>5633</td>\n",
              "      <td>16</td>\n",
              "      <td>485</td>\n",
              "      <td>9721</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "specialized_diet_exclude_refined_sugars    No  Not provided  Unspecified  Yes  \\\n",
              "sex                                                                             \n",
              "female                                   1824          2945           12  250   \n",
              "male                                     1763          2688            4  235   \n",
              "All                                      3587          5633           16  485   \n",
              "\n",
              "specialized_diet_exclude_refined_sugars   All  \n",
              "sex                                            \n",
              "female                                   5031  \n",
              "male                                     4690  \n",
              "All                                      9721  "
            ]
          },
          "execution_count": 13,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "pd.crosstab(data_f.sex,\n",
        "            data_f.specialized_diet_exclude_refined_sugars, \n",
        "            values=data_f.sample_name,\n",
        "            aggfunc=len, \n",
        "            margins = True)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Creation of the Control/Treatment groups:\n",
        "- Control: subjects without a particular health condition, that don't follow any specilized diet\n",
        "\n",
        "- Treatment: subjects without a paritcular health condition, that exclude refined sugars from their diet and don't follow any other specilized diet"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "['specialized_diet_exclude_dairy',\n",
              " 'specialized_diet_exclude_nightshades',\n",
              " 'specialized_diet_exclude_refined_sugars',\n",
              " 'specialized_diet_fodmap',\n",
              " 'specialized_diet_halaal',\n",
              " 'specialized_diet_i_do_not_eat_a_specialized_diet',\n",
              " 'specialized_diet_kosher',\n",
              " 'specialized_diet_modified_paleo_diet',\n",
              " 'specialized_diet_other_restrictions_not_described_here',\n",
              " 'specialized_diet_paleodiet_or_primal_diet',\n",
              " 'specialized_diet_raw_food_diet',\n",
              " 'specialized_diet_unspecified',\n",
              " 'specialized_diet_westenprice_or_other_lowgrain_low_processed_fo',\n",
              " 'specialized_diet_westenprice_or_other_lowgrain_low_processed_food_diet']"
            ]
          },
          "execution_count": 14,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "[x for x in list(data_f.columns) if x.lower().startswith(\"specialized_diet_\")]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_f= data_f.drop(\"specialized_diet_unspecified\",axis=1)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Function that assigns every subject to a group"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Not provided    5633\n",
              "No              3587\n",
              "Yes              485\n",
              "Unspecified       16\n",
              "Name: specialized_diet_exclude_refined_sugars, dtype: int64"
            ]
          },
          "execution_count": 31,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "data_f.specialized_diet_exclude_refined_sugars.value_counts()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 32,
      "metadata": {},
      "outputs": [],
      "source": [
        "def get_groups(data):\n",
        "    exclude_sugar = None\n",
        "    other_diets = False\n",
        "    specialized_diet = False\n",
        "\n",
        "    list_col = list(data.index)\n",
        "    list_col = [x for x in list_col if x.lower().startswith(\"specialized_diet_\")]\n",
        "    for col in list_col:\n",
        "        if col ==\"specialized_diet_i_do_not_eat_a_specialized_diet\":\n",
        "            if data[col] == \"No\":\n",
        "                specialized_diet = True\n",
        "        elif col == \"specialized_diet_exclude_refined_sugars\":\n",
        "            if data[col] == \"Yes\":\n",
        "                exclude_sugar = True\n",
        "            if data[col] == \"No\":\n",
        "                exclude_sugar = False\n",
        "        else:\n",
        "            if data[col] == \"Yes\" or data[col] == \"Not provided\":\n",
        "                other_diets = True\n",
        "    \n",
        "    if specialized_diet:\n",
        "        \n",
        "        if exclude_sugar is not None and exclude_sugar and not other_diets:\n",
        "            return \"Treatment\"\n",
        "        else:\n",
        "            return \"None\"\n",
        "    \n",
        "    else:\n",
        "        if exclude_sugar is None or exclude_sugar or other_diets:\n",
        "            return \"None\"\n",
        "        \n",
        "        \n",
        "        return \"Control\"\n",
        "        "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 33,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_f['group'] = data_f.apply(get_groups, axis=1)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Only 110 subjects have the characteristics to be in the Treatment group"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 34,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "None         7171\n",
              "Control      2440\n",
              "Treatment     110\n",
              "Name: group, dtype: int64"
            ]
          },
          "execution_count": 34,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "data_f.group.value_counts()"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Excluding all subjects that are neither in the control or treatment group"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 35,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_groups = data_f.loc[((data_f.group == \"Control\") | (data_f.group == \"Treatment\"))]"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Sex distribution in the two groups"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 36,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "group      sex   \n",
            "Control    female    1202\n",
            "           male      1238\n",
            "Treatment  female      53\n",
            "           male        57\n",
            "Name: sample_name, dtype: int64\n"
          ]
        }
      ],
      "source": [
        "print(data_groups.groupby([\"group\",\"sex\"]).sample_name.nunique())"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Sex and age distribution in the two groups"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 37,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "group      sex     age_cat\n",
            "Control    female  20s        183\n",
            "                   30s        271\n",
            "                   40s        281\n",
            "                   50s        289\n",
            "                   60s        178\n",
            "           male    20s        199\n",
            "                   30s        299\n",
            "                   40s        264\n",
            "                   50s        263\n",
            "                   60s        213\n",
            "Treatment  female  20s          3\n",
            "                   30s         14\n",
            "                   40s         11\n",
            "                   50s         12\n",
            "                   60s         13\n",
            "           male    20s         13\n",
            "                   30s          9\n",
            "                   40s         11\n",
            "                   50s         12\n",
            "                   60s         12\n",
            "Name: sample_name, dtype: int64\n"
          ]
        }
      ],
      "source": [
        "print(data_groups.groupby([\"group\",\"sex\",\"age_cat\"]).sample_name.nunique())"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 38,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "group      sex   \n",
            "Control    female    44.46875\n",
            "           male      44.56250\n",
            "Treatment  female    48.62500\n",
            "           male      44.90625\n",
            "Name: age_years, dtype: float16\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.8/dist-packages/pandas/core/generic.py:5516: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  self[name] = value\n"
          ]
        }
      ],
      "source": [
        "data_groups.age_years = data_groups.age_years.astype('float16')\n",
        "print(data_groups.groupby([\"group\",\"sex\"]).age_years.mean())"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "FastQ pull:"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 43,
      "metadata": {},
      "outputs": [],
      "source": [
        "def get_fastq(sample_names, download = False):\n",
        "    \n",
        "    sample_name = list(sample_names)\n",
        "    biosample_ids = list(sample_names)\n",
        "    run_ids = list(sample_names) \n",
        "    i = 0\n",
        "    \n",
        "    for sample in sample_name:\n",
        "        i+=1\n",
        "        print(f\"({i})\\tGetting {sample_name[i-1]} Biosample ID\")\n",
        "        samplehandle = Entrez.esearch(db='biosample', term=sample)\n",
        "        sampleread = Entrez.read(samplehandle)\n",
        "                \n",
        "        if len(sampleread['IdList']) == 0:\n",
        "            print(\"No BiosampleID found\")\n",
        "            biosample_ids[i-1] = \"Not found\"\n",
        "            run_ids[i-1] = \"Not found\"\n",
        "\n",
        "        else: \n",
        "            print(f\"\\tFound {len(sampleread['IdList'])} BiosampleID: {sampleread['IdList']}\")\n",
        "\n",
        "            for bioid in sampleread['IdList']:\n",
        "                try:\n",
        "                    biohandle = Entrez.efetch(db='biosample', id=bioid, retmode='xml')\n",
        "                    bioxml = ET.fromstring(biohandle.read())\n",
        "                    sraids = bioxml.findall('.//BioSample//Ids//Id')\n",
        "                    sraids = [x.text for x in sraids if x.attrib['db']==\"SRA\"]\n",
        "                except:\n",
        "                    run_ids[i-1] = \"Not found\"\n",
        "                    continue \n",
        "                \n",
        "                for sraid in sraids:\n",
        "                    srahandle = Entrez.esearch(db='sra', term=sraid,retmode=\"xml\")\n",
        "                    srareads = Entrez.read(srahandle)[\"IdList\"]\n",
        "                    print(f\"\\tFound {len(srareads)} SRA ID: {srareads}\")\n",
        "\n",
        "                    sraidhandles =[Entrez.efetch(db='sra', id=s) for s in srareads]\n",
        "                    root = [ET.fromstring(handle.read()) for handle in sraidhandles]\n",
        "                    identifier = [r.find('.//EXPERIMENT_PACKAGE//RUN_SET//RUN') for r in root]\n",
        "                    instrument = [r.find('.//EXPERIMENT_PACKAGE//EXPERIMENT//PLATFORM//ILLUMINA//INSTRUMENT_MODEL').text for r in root]\n",
        "\n",
        "                    filt_id = [identifier[k] for k in range(len(identifier)) if instrument[k] == \"Illumina MiSeq\"]\n",
        "\n",
        "                    if len(filt_id)>0 and filt_id[0] is not None:\n",
        "                        one_id = filt_id[0].attrib[\"accession\"]\n",
        "                        print('\\tSelected 1 Run ID:', one_id)\n",
        "                        biosample_ids[i-1]=bioid\n",
        "                        run_ids[i-1] = one_id\n",
        "                    else:\n",
        "                        print('\\tNo RunID found')\n",
        "                        run_ids[i-1] = \"Not found\"\n",
        "\n",
        "                \n",
        "                    # Download FASTQ dei RUNid\n",
        "                    if download:\n",
        "                        if not os.path.exists(f\"fastq\"):\n",
        "                            os.mkdir(f\"fastq\")\n",
        "\n",
        "                        print('\\tDownloading FASTQ...')\n",
        "                        command = f'sratoolkit.3.0.5-ubuntu64/bin/fastq-dump --outdir fastq/ --readids ' + one_id\n",
        "                        os.system(command)\n",
        "\n",
        "\n",
        "    dict = {'sample_names': sample_name, \n",
        "                'biosample_ids': biosample_ids,\n",
        "                'run_ids': run_ids}  \n",
        "\n",
        "    return dict    "
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Treatment group data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 44,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_treatment = data_groups.loc[(data_groups.group == \"Treatment\")]"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Control group data (selecting only 120 subjects)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 45,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_control = data_groups.loc[(data_groups.group == \"Control\")]\n",
        "data_control = data_control.groupby([\"sex\",\"age_cat\"]).head(12)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Treatment group fastq pulling"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 46,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "(1)\tGetting 10317.000023139 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353729']\n",
            "\tFound 1 SRA ID: ['4279800']\n",
            "\tSelected 1 Run ID: ERR2032495\n",
            "\tDownloading FASTQ...\n",
            "Read 16572 spots for ERR2032495\n",
            "Written 16572 spots for ERR2032495\n",
            "(2)\tGetting 10317.000027811 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353735']\n",
            "\tFound 1 SRA ID: ['4279806']\n",
            "\tSelected 1 Run ID: ERR2032501\n",
            "\tDownloading FASTQ...\n",
            "Read 19562 spots for ERR2032501\n",
            "Written 19562 spots for ERR2032501\n",
            "(3)\tGetting 10317.000033280 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577201']\n",
            "\tFound 1 SRA ID: ['5144891']\n",
            "\tSelected 1 Run ID: ERR2313963\n",
            "\tDownloading FASTQ...\n",
            "Read 27488 spots for ERR2313963\n",
            "Written 27488 spots for ERR2313963\n",
            "(4)\tGetting 10317.000038261 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614208']\n",
            "\tFound 1 SRA ID: ['5162312']\n",
            "\tSelected 1 Run ID: ERR2318007\n",
            "\tDownloading FASTQ...\n",
            "Read 39992 spots for ERR2318007\n",
            "Written 39992 spots for ERR2318007\n",
            "(5)\tGetting 10317.000041075 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8728556']\n",
            "\tFound 5 SRA ID: ['14635658', '14635351', '14634860', '14631050', '5245039']\n",
            "\tSelected 1 Run ID: ERR2404913\n",
            "\tDownloading FASTQ...\n",
            "Read 34268 spots for ERR2404913\n",
            "Written 34268 spots for ERR2404913\n",
            "(6)\tGetting 10317.000050189 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366098']\n",
            "\tFound 1 SRA ID: ['3736480']\n",
            "\tSelected 1 Run ID: ERR1842797\n",
            "\tDownloading FASTQ...\n",
            "Read 28329 spots for ERR1842797\n",
            "Written 28329 spots for ERR1842797\n",
            "(7)\tGetting 10317.000051222 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9653436']\n",
            "\tFound 1 SRA ID: ['5960036']\n",
            "\tSelected 1 Run ID: ERR2696825\n",
            "\tDownloading FASTQ...\n",
            "Read 12170 spots for ERR2696825\n",
            "Written 12170 spots for ERR2696825\n",
            "(8)\tGetting 10317.000052017 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366109']\n",
            "\tFound 2 SRA ID: ['3736491', '5144952']\n",
            "\tSelected 1 Run ID: ERR1842808\n",
            "\tDownloading FASTQ...\n",
            "Read 16660 spots for ERR1842808\n",
            "Written 16660 spots for ERR1842808\n",
            "(9)\tGetting 10317.000052437 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8945357']\n",
            "\tFound 1 SRA ID: ['5424259']\n",
            "\tSelected 1 Run ID: ERR2523875\n",
            "\tDownloading FASTQ...\n",
            "Read 40808 spots for ERR2523875\n",
            "Written 40808 spots for ERR2523875\n",
            "(10)\tGetting 10317.000054059 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569437']\n",
            "\tFound 1 SRA ID: ['5134324']\n",
            "\tSelected 1 Run ID: ERR2306103\n",
            "\tDownloading FASTQ...\n",
            "Read 78069 spots for ERR2306103\n",
            "Written 78069 spots for ERR2306103\n",
            "(11)\tGetting 10317.000058376 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569441']\n",
            "\tFound 1 SRA ID: ['5134328']\n",
            "\tSelected 1 Run ID: ERR2306107\n",
            "\tDownloading FASTQ...\n",
            "Read 75366 spots for ERR2306107\n",
            "Written 75366 spots for ERR2306107\n",
            "(12)\tGetting 10317.000058489 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366178']\n",
            "\tFound 1 SRA ID: ['3736560']\n",
            "\tSelected 1 Run ID: ERR1842877\n",
            "\tDownloading FASTQ...\n",
            "Read 17826 spots for ERR1842877\n",
            "Written 17826 spots for ERR1842877\n",
            "(13)\tGetting 10317.000058560 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18047848']\n",
            "\tFound 1 SRA ID: ['13603924']\n",
            "\tSelected 1 Run ID: ERR5332667\n",
            "\tDownloading FASTQ...\n",
            "Read 13714 spots for ERR5332667\n",
            "Written 13714 spots for ERR5332667\n",
            "(14)\tGetting 10317.000058561 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18047689']\n",
            "\tFound 1 SRA ID: ['13603925']\n",
            "\tSelected 1 Run ID: ERR5332668\n",
            "\tDownloading FASTQ...\n",
            "Read 14608 spots for ERR5332668\n",
            "Written 14608 spots for ERR5332668\n",
            "(15)\tGetting 10317.000058846 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353978']\n",
            "\tFound 1 SRA ID: ['4280059']\n",
            "\tSelected 1 Run ID: ERR2032754\n",
            "\tDownloading FASTQ...\n",
            "Read 13714 spots for ERR2032754\n",
            "Written 13714 spots for ERR2032754\n",
            "(16)\tGetting 10317.000058966 Biosample ID\n",
            "No BiosampleID found\n",
            "(17)\tGetting 10317.000060330 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9653442']\n",
            "\tFound 1 SRA ID: ['5960043']\n",
            "\tSelected 1 Run ID: ERR2696832\n",
            "\tDownloading FASTQ...\n",
            "Read 9065 spots for ERR2696832\n",
            "Written 9065 spots for ERR2696832\n",
            "(18)\tGetting 10317.000062130 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439226']\n",
            "\tFound 4 SRA ID: ['4337331', '4939745', '4932868', '4932490']\n",
            "\tSelected 1 Run ID: ERR2056970\n",
            "\tDownloading FASTQ...\n",
            "Read 27824 spots for ERR2056970\n",
            "Written 27824 spots for ERR2056970\n",
            "(19)\tGetting 10317.000062131 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614281']\n",
            "\tFound 1 SRA ID: ['5162387']\n",
            "\tSelected 1 Run ID: ERR2318082\n",
            "\tDownloading FASTQ...\n",
            "Read 34048 spots for ERR2318082\n",
            "Written 34048 spots for ERR2318082\n",
            "(20)\tGetting 10317.000062968 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354021']\n",
            "\tFound 1 SRA ID: ['4280102']\n",
            "\tSelected 1 Run ID: ERR2032797\n",
            "\tDownloading FASTQ...\n",
            "Read 48855 spots for ERR2032797\n",
            "Written 48855 spots for ERR2032797\n",
            "(21)\tGetting 10317.000063010 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354028']\n",
            "\tFound 1 SRA ID: ['4280109']\n",
            "\tSelected 1 Run ID: ERR2032804\n",
            "\tDownloading FASTQ...\n",
            "Read 24596 spots for ERR2032804\n",
            "Written 24596 spots for ERR2032804\n",
            "(22)\tGetting 10317.000063234 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577314']\n",
            "\tFound 1 SRA ID: ['5145034']\n",
            "\tSelected 1 Run ID: ERR2314106\n",
            "\tDownloading FASTQ...\n",
            "Read 33833 spots for ERR2314106\n",
            "Written 33833 spots for ERR2314106\n",
            "(23)\tGetting 10317.000063235 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354290']\n",
            "\tFound 1 SRA ID: ['4280391']\n",
            "\tSelected 1 Run ID: ERR2033290\n",
            "\tDownloading FASTQ...\n",
            "Read 18027 spots for ERR2033290\n",
            "Written 18027 spots for ERR2033290\n",
            "(24)\tGetting 10317.000063294 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577317']\n",
            "\tFound 1 SRA ID: ['5145037']\n",
            "\tSelected 1 Run ID: ERR2314109\n",
            "\tDownloading FASTQ...\n",
            "Read 34149 spots for ERR2314109\n",
            "Written 34149 spots for ERR2314109\n",
            "(25)\tGetting 10317.000063295 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618451']\n",
            "\tFound 1 SRA ID: ['10591987']\n",
            "\tSelected 1 Run ID: ERR4018935\n",
            "\tDownloading FASTQ...\n",
            "Read 31148 spots for ERR4018935\n",
            "Written 31148 spots for ERR4018935\n",
            "(26)\tGetting 10317.000063573 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577327']\n",
            "\tFound 1 SRA ID: ['5145047']\n",
            "\tSelected 1 Run ID: ERR2314119\n",
            "\tDownloading FASTQ...\n",
            "Read 37817 spots for ERR2314119\n",
            "Written 37817 spots for ERR2314119\n",
            "(27)\tGetting 10317.000065297 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354042']\n",
            "\tFound 1 SRA ID: ['4280123']\n",
            "\tSelected 1 Run ID: ERR2032818\n",
            "\tDownloading FASTQ...\n",
            "Read 20563 spots for ERR2032818\n",
            "Written 20563 spots for ERR2032818\n",
            "(28)\tGetting 10317.000066580 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577353']\n",
            "\tFound 1 SRA ID: ['5145080']\n",
            "\tSelected 1 Run ID: ERR2314152\n",
            "\tDownloading FASTQ...\n",
            "Read 49733 spots for ERR2314152\n",
            "Written 49733 spots for ERR2314152\n",
            "(29)\tGetting 10317.000067000 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8728581']\n",
            "\tFound 1 SRA ID: ['5245068']\n",
            "\tSelected 1 Run ID: ERR2404942\n",
            "\tDownloading FASTQ...\n",
            "Read 16994 spots for ERR2404942\n",
            "Written 16994 spots for ERR2404942\n",
            "(30)\tGetting 10317.000067091 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8568972']\n",
            "\tFound 1 SRA ID: ['5125319']\n",
            "\tSelected 1 Run ID: ERR2303921\n",
            "\tDownloading FASTQ...\n",
            "Read 31496 spots for ERR2303921\n",
            "Written 31496 spots for ERR2303921\n",
            "(31)\tGetting 10317.000067525 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7497010']\n",
            "\tFound 5 SRA ID: ['14637457', '14635704', '14634906', '14631096', '4376044']\n",
            "\tSelected 1 Run ID: ERR2092192\n",
            "\tDownloading FASTQ...\n",
            "Read 26891 spots for ERR2092192\n",
            "Written 26891 spots for ERR2092192\n",
            "(32)\tGetting 10317.000067526 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7497011']\n",
            "\tFound 1 SRA ID: ['4376045']\n",
            "\tSelected 1 Run ID: ERR2092193\n",
            "\tDownloading FASTQ...\n",
            "Read 26561 spots for ERR2092193\n",
            "Written 26561 spots for ERR2092193\n",
            "(33)\tGetting 10317.000067527 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7497012']\n",
            "\tFound 1 SRA ID: ['4376046']\n",
            "\tSelected 1 Run ID: ERR2092194\n",
            "\tDownloading FASTQ...\n",
            "Read 24008 spots for ERR2092194\n",
            "Written 24008 spots for ERR2092194\n",
            "(34)\tGetting 10317.000067528 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7497013']\n",
            "\tFound 1 SRA ID: ['4376047']\n",
            "\tSelected 1 Run ID: ERR2092195\n",
            "\tDownloading FASTQ...\n",
            "Read 27911 spots for ERR2092195\n",
            "Written 27911 spots for ERR2092195\n",
            "(35)\tGetting 10317.000068165 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614770']\n",
            "\tFound 1 SRA ID: ['5162917']\n",
            "\tSelected 1 Run ID: ERR2319477\n",
            "\tDownloading FASTQ...\n",
            "Read 35379 spots for ERR2319477\n",
            "Written 35379 spots for ERR2319477\n",
            "(36)\tGetting 10317.000068166 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614771']\n",
            "\tFound 1 SRA ID: ['5162918']\n",
            "\tSelected 1 Run ID: ERR2319478\n",
            "\tDownloading FASTQ...\n",
            "Read 54801 spots for ERR2319478\n",
            "Written 54801 spots for ERR2319478\n",
            "(37)\tGetting 10317.000068167 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354467']\n",
            "\tFound 1 SRA ID: ['4280568']\n",
            "\tSelected 1 Run ID: ERR2033467\n",
            "\tDownloading FASTQ...\n",
            "Read 30274 spots for ERR2033467\n",
            "Written 30274 spots for ERR2033467\n",
            "(38)\tGetting 10317.000068168 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354468']\n",
            "\tFound 1 SRA ID: ['4280569']\n",
            "\tSelected 1 Run ID: ERR2033468\n",
            "\tDownloading FASTQ...\n",
            "Read 25167 spots for ERR2033468\n",
            "Written 25167 spots for ERR2033468\n",
            "(39)\tGetting 10317.000068405 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541236']\n",
            "\tFound 1 SRA ID: ['5840526']\n",
            "\tSelected 1 Run ID: ERR2579788\n",
            "\tDownloading FASTQ...\n",
            "Read 12129 spots for ERR2579788\n",
            "Written 12129 spots for ERR2579788\n",
            "(40)\tGetting 10317.000068406 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541237']\n",
            "\tFound 1 SRA ID: ['5840527']\n",
            "\tSelected 1 Run ID: ERR2579789\n",
            "\tDownloading FASTQ...\n",
            "Read 3583 spots for ERR2579789\n",
            "Written 3583 spots for ERR2579789\n",
            "(41)\tGetting 10317.000068407 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614348']\n",
            "\tFound 1 SRA ID: ['5162455']\n",
            "\tSelected 1 Run ID: ERR2318150\n",
            "\tDownloading FASTQ...\n",
            "Read 13 spots for ERR2318150\n",
            "Written 13 spots for ERR2318150\n",
            "(42)\tGetting 10317.000068408 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614349']\n",
            "\tFound 1 SRA ID: ['5162456']\n",
            "\tSelected 1 Run ID: ERR2318151\n",
            "\tDownloading FASTQ...\n",
            "Read 32755 spots for ERR2318151\n",
            "Written 32755 spots for ERR2318151\n",
            "(43)\tGetting 10317.000068409 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541238']\n",
            "\tFound 1 SRA ID: ['5840528']\n",
            "\tSelected 1 Run ID: ERR2579790\n",
            "\tDownloading FASTQ...\n",
            "Read 18951 spots for ERR2579790\n",
            "Written 18951 spots for ERR2579790\n",
            "(44)\tGetting 10317.000068410 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614350']\n",
            "\tFound 1 SRA ID: ['5162457']\n",
            "\tSelected 1 Run ID: ERR2318152\n",
            "\tDownloading FASTQ...\n",
            "Read 57 spots for ERR2318152\n",
            "Written 57 spots for ERR2318152\n",
            "(45)\tGetting 10317.000068411 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614351']\n",
            "\tFound 1 SRA ID: ['5162458']\n",
            "\tSelected 1 Run ID: ERR2318153\n",
            "\tDownloading FASTQ...\n",
            "Read 36283 spots for ERR2318153\n",
            "Written 36283 spots for ERR2318153\n",
            "(46)\tGetting 10317.000069874 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614381']\n",
            "\tFound 1 SRA ID: ['5162490']\n",
            "\tSelected 1 Run ID: ERR2318185\n",
            "\tDownloading FASTQ...\n",
            "Read 46640 spots for ERR2318185\n",
            "Written 46640 spots for ERR2318185\n",
            "(47)\tGetting 10317.000071460 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541302']\n",
            "\tFound 1 SRA ID: ['5840628']\n",
            "\tSelected 1 Run ID: ERR2579848\n",
            "\tDownloading FASTQ...\n",
            "Read 38207 spots for ERR2579848\n",
            "Written 38207 spots for ERR2579848\n",
            "(48)\tGetting 10317.000071514 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577053']\n",
            "\tFound 1 SRA ID: ['5144320']\n",
            "\tSelected 1 Run ID: ERR2309320\n",
            "\tDownloading FASTQ...\n",
            "Read 54341 spots for ERR2309320\n",
            "Written 54341 spots for ERR2309320\n",
            "(49)\tGetting 10317.000071552 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541315']\n",
            "\tFound 1 SRA ID: ['5840641']\n",
            "\tSelected 1 Run ID: ERR2579861\n",
            "\tDownloading FASTQ...\n",
            "Read 58736 spots for ERR2579861\n",
            "Written 58736 spots for ERR2579861\n",
            "(50)\tGetting 10317.000071913 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541322']\n",
            "\tFound 1 SRA ID: ['5840648']\n",
            "\tSelected 1 Run ID: ERR2579868\n",
            "\tDownloading FASTQ...\n",
            "Read 3162 spots for ERR2579868\n",
            "Written 3162 spots for ERR2579868\n",
            "(51)\tGetting 10317.000072012 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577525']\n",
            "\tFound 1 SRA ID: ['5145259']\n",
            "\tSelected 1 Run ID: ERR2314331\n",
            "\tDownloading FASTQ...\n",
            "Read 38741 spots for ERR2314331\n",
            "Written 38741 spots for ERR2314331\n",
            "(52)\tGetting 10317.000072056 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618518']\n",
            "\tFound 1 SRA ID: ['10592056']\n",
            "\tSelected 1 Run ID: ERR4019004\n",
            "\tDownloading FASTQ...\n",
            "Read 25164 spots for ERR4019004\n",
            "Written 25164 spots for ERR4019004\n",
            "(53)\tGetting 10317.000072635 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577546']\n",
            "\tFound 1 SRA ID: ['5145280']\n",
            "\tSelected 1 Run ID: ERR2314352\n",
            "\tDownloading FASTQ...\n",
            "Read 32221 spots for ERR2314352\n",
            "Written 32221 spots for ERR2314352\n",
            "(54)\tGetting 10317.000072639 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569068']\n",
            "\tFound 1 SRA ID: ['5125415']\n",
            "\tSelected 1 Run ID: ERR2304017\n",
            "\tDownloading FASTQ...\n",
            "Read 19620 spots for ERR2304017\n",
            "Written 19620 spots for ERR2304017\n",
            "(55)\tGetting 10317.000072641 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569069']\n",
            "\tFound 1 SRA ID: ['5125416']\n",
            "\tSelected 1 Run ID: ERR2304018\n",
            "\tDownloading FASTQ...\n",
            "Read 27533 spots for ERR2304018\n",
            "Written 27533 spots for ERR2304018\n",
            "(56)\tGetting 10317.000073896 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541509']\n",
            "\tFound 1 SRA ID: ['5840836']\n",
            "\tSelected 1 Run ID: ERR2580056\n",
            "\tDownloading FASTQ...\n",
            "Read 34401 spots for ERR2580056\n",
            "Written 34401 spots for ERR2580056\n",
            "(57)\tGetting 10317.000074288 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618569']\n",
            "\tFound 1 SRA ID: ['10592109']\n",
            "\tSelected 1 Run ID: ERR4019057\n",
            "\tDownloading FASTQ...\n",
            "Read 34078 spots for ERR4019057\n",
            "Written 34078 spots for ERR4019057\n",
            "(58)\tGetting 10317.000074478 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046105']\n",
            "\tFound 1 SRA ID: ['13600894']\n",
            "\tSelected 1 Run ID: ERR5327749\n",
            "\tDownloading FASTQ...\n",
            "Read 20192 spots for ERR5327749\n",
            "Written 20192 spots for ERR5327749\n",
            "(59)\tGetting 10317.000074600 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14621245']\n",
            "\tFound 1 SRA ID: ['10593178']\n",
            "\tSelected 1 Run ID: ERR4020390\n",
            "\tDownloading FASTQ...\n",
            "Read 18243 spots for ERR4020390\n",
            "Written 18243 spots for ERR4020390\n",
            "(60)\tGetting 10317.000074640 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577110']\n",
            "\tFound 1 SRA ID: ['5144377']\n",
            "\tSelected 1 Run ID: ERR2309377\n",
            "\tDownloading FASTQ...\n",
            "Read 75349 spots for ERR2309377\n",
            "Written 75349 spots for ERR2309377\n",
            "(61)\tGetting 10317.000075877 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8945516']\n",
            "\tFound 1 SRA ID: ['5424310']\n",
            "\tSelected 1 Run ID: ERR2523926\n",
            "\tDownloading FASTQ...\n",
            "Read 19120 spots for ERR2523926\n",
            "Written 19120 spots for ERR2523926\n",
            "(62)\tGetting 10317.000075899 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18018905']\n",
            "\tFound 1 SRA ID: ['13595239']\n",
            "\tNo RunID found\n",
            "\tDownloading FASTQ...\n",
            "Read 19120 spots for ERR2523926\n",
            "Written 19120 spots for ERR2523926\n",
            "(63)\tGetting 10317.000076338 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9653106']\n",
            "\tFound 1 SRA ID: ['5959705']\n",
            "\tSelected 1 Run ID: ERR2696489\n",
            "\tDownloading FASTQ...\n",
            "Read 39151 spots for ERR2696489\n",
            "Written 39151 spots for ERR2696489\n",
            "(64)\tGetting 10317.000079394 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569131']\n",
            "\tFound 1 SRA ID: ['5125478']\n",
            "\tSelected 1 Run ID: ERR2304080\n",
            "\tDownloading FASTQ...\n",
            "Read 29432 spots for ERR2304080\n",
            "Written 29432 spots for ERR2304080\n",
            "(65)\tGetting 10317.000084428 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569186']\n",
            "\tFound 2 SRA ID: ['5840123', '5125533']\n",
            "\tSelected 1 Run ID: ERR2579425\n",
            "\tDownloading FASTQ...\n",
            "Read 26033 spots for ERR2579425\n",
            "Written 26033 spots for ERR2579425\n",
            "(66)\tGetting 10317.000087655 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18018465']\n",
            "\tFound 1 SRA ID: ['13595338']\n",
            "\tNo RunID found\n",
            "\tDownloading FASTQ...\n",
            "Read 26033 spots for ERR2579425\n",
            "Written 26033 spots for ERR2579425\n",
            "(67)\tGetting 10317.000087669 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8729296']\n",
            "\tFound 1 SRA ID: ['5245453']\n",
            "\tSelected 1 Run ID: ERR2405327\n",
            "\tDownloading FASTQ...\n",
            "Read 17131 spots for ERR2405327\n",
            "Written 17131 spots for ERR2405327\n",
            "(68)\tGetting 10317.000092676 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657576']\n",
            "\tFound 1 SRA ID: ['5964271']\n",
            "\tSelected 1 Run ID: ERR2697711\n",
            "\tDownloading FASTQ...\n",
            "Read 29816 spots for ERR2697711\n",
            "Written 29816 spots for ERR2697711\n",
            "(69)\tGetting 10317.000092766 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9653213']\n",
            "\tFound 1 SRA ID: ['5959813']\n",
            "\tSelected 1 Run ID: ERR2696597\n",
            "\tDownloading FASTQ...\n",
            "Read 57555 spots for ERR2696597\n",
            "Written 57555 spots for ERR2696597\n",
            "(70)\tGetting 10317.000092851 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046234']\n",
            "\tFound 1 SRA ID: ['13599458']\n",
            "\tSelected 1 Run ID: ERR5326262\n",
            "\tDownloading FASTQ...\n",
            "Read 10300 spots for ERR5326262\n",
            "Written 10300 spots for ERR5326262\n",
            "(71)\tGetting 10317.000092929 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14621278']\n",
            "\tFound 1 SRA ID: ['10593894']\n",
            "\tSelected 1 Run ID: ERR4020425\n",
            "\tDownloading FASTQ...\n",
            "Read 12946 spots for ERR4020425\n",
            "Written 12946 spots for ERR4020425\n",
            "(72)\tGetting 10317.000093616 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657681']\n",
            "\tFound 1 SRA ID: ['5964376']\n",
            "\tSelected 1 Run ID: ERR2697816\n",
            "\tDownloading FASTQ...\n",
            "Read 19136 spots for ERR2697816\n",
            "Written 19136 spots for ERR2697816\n",
            "(73)\tGetting 10317.000094618 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9653287']\n",
            "\tFound 1 SRA ID: ['5959888']\n",
            "\tSelected 1 Run ID: ERR2696671\n",
            "\tDownloading FASTQ...\n",
            "Read 20838 spots for ERR2696671\n",
            "Written 20838 spots for ERR2696671\n",
            "(74)\tGetting 10317.000094643 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9653667']\n",
            "\tFound 1 SRA ID: ['5960269']\n",
            "\tSelected 1 Run ID: ERR2697056\n",
            "\tDownloading FASTQ...\n",
            "Read 38000 spots for ERR2697056\n",
            "Written 38000 spots for ERR2697056\n",
            "(75)\tGetting 10317.000097194 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14621288']\n",
            "\tFound 1 SRA ID: ['10593904']\n",
            "\tSelected 1 Run ID: ERR4020435\n",
            "\tDownloading FASTQ...\n",
            "Read 26478 spots for ERR4020435\n",
            "Written 26478 spots for ERR4020435\n",
            "(76)\tGetting 10317.000097336 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657727']\n",
            "\tFound 1 SRA ID: ['5964422']\n",
            "\tSelected 1 Run ID: ERR2697862\n",
            "\tDownloading FASTQ...\n",
            "Read 18485 spots for ERR2697862\n",
            "Written 18485 spots for ERR2697862\n",
            "(77)\tGetting 10317.000097337 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657728']\n",
            "\tFound 1 SRA ID: ['5964423']\n",
            "\tSelected 1 Run ID: ERR2697863\n",
            "\tDownloading FASTQ...\n",
            "Read 21535 spots for ERR2697863\n",
            "Written 21535 spots for ERR2697863\n",
            "(78)\tGetting 10317.000097499 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14621297']\n",
            "\tFound 1 SRA ID: ['10593913']\n",
            "\tSelected 1 Run ID: ERR4020444\n",
            "\tDownloading FASTQ...\n",
            "Read 23110 spots for ERR4020444\n",
            "Written 23110 spots for ERR4020444\n",
            "(79)\tGetting 10317.000097731 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18050158']\n",
            "\tFound 1 SRA ID: ['13600941']\n",
            "\tSelected 1 Run ID: ERR5327796\n",
            "\tDownloading FASTQ...\n",
            "Read 23637 spots for ERR5327796\n",
            "Written 23637 spots for ERR5327796\n",
            "(80)\tGetting 10317.000097781 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14621308']\n",
            "\tFound 1 SRA ID: ['10593924']\n",
            "\tSelected 1 Run ID: ERR4020455\n",
            "\tDownloading FASTQ...\n",
            "Read 17643 spots for ERR4020455\n",
            "Written 17643 spots for ERR4020455\n",
            "(81)\tGetting 10317.000097818 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048171']\n",
            "\tFound 1 SRA ID: ['13598892']\n",
            "\tSelected 1 Run ID: ERR5325637\n",
            "\tDownloading FASTQ...\n",
            "Read 23733 spots for ERR5325637\n",
            "Written 23733 spots for ERR5325637\n",
            "(82)\tGetting 10317.000097954 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046240']\n",
            "\tFound 1 SRA ID: ['13599513']\n",
            "\tSelected 1 Run ID: ERR5326317\n",
            "\tDownloading FASTQ...\n",
            "Read 21842 spots for ERR5326317\n",
            "Written 21842 spots for ERR5326317\n",
            "(83)\tGetting 10317.000098040 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046397']\n",
            "\tFound 1 SRA ID: ['13599529']\n",
            "\tSelected 1 Run ID: ERR5326333\n",
            "\tDownloading FASTQ...\n",
            "Read 20575 spots for ERR5326333\n",
            "Written 20575 spots for ERR5326333\n",
            "(84)\tGetting 10317.000098041 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046231']\n",
            "\tFound 1 SRA ID: ['13599530']\n",
            "\tSelected 1 Run ID: ERR5326334\n",
            "\tDownloading FASTQ...\n",
            "Read 18749 spots for ERR5326334\n",
            "Written 18749 spots for ERR5326334\n",
            "(85)\tGetting 10317.000100523 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14619466']\n",
            "\tFound 2 SRA ID: ['13837477', '10592773']\n",
            "\tSelected 1 Run ID: ERR4019687\n",
            "\tDownloading FASTQ...\n",
            "Read 42714 spots for ERR4019687\n",
            "Written 42714 spots for ERR4019687\n",
            "(86)\tGetting 10317.000101328 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14619488']\n",
            "\tFound 2 SRA ID: ['13837633', '10592795']\n",
            "\tSelected 1 Run ID: ERR4019709\n",
            "\tDownloading FASTQ...\n",
            "Read 31564 spots for ERR4019709\n",
            "Written 31564 spots for ERR4019709\n",
            "(87)\tGetting 10317.000102702 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046321']\n",
            "\tFound 1 SRA ID: ['13599678']\n",
            "\tSelected 1 Run ID: ERR5326482\n",
            "\tDownloading FASTQ...\n",
            "Read 20175 spots for ERR5326482\n",
            "Written 20175 spots for ERR5326482\n",
            "(88)\tGetting 10317.000102703 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046316']\n",
            "\tFound 1 SRA ID: ['13599679']\n",
            "\tSelected 1 Run ID: ERR5326483\n",
            "\tDownloading FASTQ...\n",
            "Read 22192 spots for ERR5326483\n",
            "Written 22192 spots for ERR5326483\n",
            "(89)\tGetting 10317.000102935 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18018254']\n",
            "\tFound 1 SRA ID: ['13595501']\n",
            "\tNo RunID found\n",
            "\tDownloading FASTQ...\n",
            "Read 22192 spots for ERR5326483\n",
            "Written 22192 spots for ERR5326483\n",
            "(90)\tGetting 10317.000103064 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18049272']\n",
            "\tFound 1 SRA ID: ['13603378']\n",
            "\tSelected 1 Run ID: ERR5332123\n",
            "\tDownloading FASTQ...\n",
            "Read 17584 spots for ERR5332123\n",
            "Written 17584 spots for ERR5332123\n",
            "(91)\tGetting 10317.000103689 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618736']\n",
            "\tFound 1 SRA ID: ['10592334']\n",
            "\tSelected 1 Run ID: ERR4019252\n",
            "\tDownloading FASTQ...\n",
            "Read 24589 spots for ERR4019252\n",
            "Written 24589 spots for ERR4019252\n",
            "(92)\tGetting 10317.000103790 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618746']\n",
            "\tFound 1 SRA ID: ['10592344']\n",
            "\tSelected 1 Run ID: ERR4019262\n",
            "\tDownloading FASTQ...\n",
            "Read 16699 spots for ERR4019262\n",
            "Written 16699 spots for ERR4019262\n",
            "(93)\tGetting 10317.000105499 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18050474']\n",
            "\tFound 1 SRA ID: ['13601512']\n",
            "\tSelected 1 Run ID: ERR5328798\n",
            "\tDownloading FASTQ...\n",
            "Read 14172 spots for ERR5328798\n",
            "Written 14172 spots for ERR5328798\n",
            "(94)\tGetting 10317.000105572 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18018207']\n",
            "\tFound 1 SRA ID: ['13595594']\n",
            "\tNo RunID found\n",
            "\tDownloading FASTQ...\n",
            "Read 14172 spots for ERR5328798\n",
            "Written 14172 spots for ERR5328798\n",
            "(95)\tGetting 10317.000106410 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618766']\n",
            "\tFound 5 SRA ID: ['14637527', '14635765', '14634976', '14630964', '10592364']\n",
            "\tSelected 1 Run ID: ERR4019282\n",
            "\tDownloading FASTQ...\n",
            "Read 27182 spots for ERR4019282\n",
            "Written 27182 spots for ERR4019282\n",
            "(96)\tGetting 10317.000107324 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18050266']\n",
            "\tFound 1 SRA ID: ['13601073']\n",
            "\tSelected 1 Run ID: ERR5327928\n",
            "\tDownloading FASTQ...\n",
            "Read 21681 spots for ERR5327928\n",
            "Written 21681 spots for ERR5327928\n",
            "(97)\tGetting 10317.000107446 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046708']\n",
            "\tFound 1 SRA ID: ['13603464']\n",
            "\tSelected 1 Run ID: ERR5332209\n",
            "\tDownloading FASTQ...\n",
            "Read 18654 spots for ERR5332209\n",
            "Written 18654 spots for ERR5332209\n",
            "(98)\tGetting 10317.000107636 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048901']\n",
            "\tFound 1 SRA ID: ['13601627']\n",
            "\tSelected 1 Run ID: ERR5328913\n",
            "\tDownloading FASTQ...\n",
            "Read 27063 spots for ERR5328913\n",
            "Written 27063 spots for ERR5328913\n",
            "(99)\tGetting 10317.000107686 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046885']\n",
            "\tFound 1 SRA ID: ['13603487']\n",
            "\tSelected 1 Run ID: ERR5332232\n",
            "\tDownloading FASTQ...\n",
            "Read 21965 spots for ERR5332232\n",
            "Written 21965 spots for ERR5332232\n",
            "(100)\tGetting 10317.000108028 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048830']\n",
            "\tFound 1 SRA ID: ['13601676']\n",
            "\tSelected 1 Run ID: ERR5328962\n",
            "\tDownloading FASTQ...\n",
            "Read 30935 spots for ERR5328962\n",
            "Written 30935 spots for ERR5328962\n",
            "(101)\tGetting 10317.000108434 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18050057']\n",
            "\tFound 1 SRA ID: ['13601178']\n",
            "\tSelected 1 Run ID: ERR5328033\n",
            "\tDownloading FASTQ...\n",
            "Read 44168 spots for ERR5328033\n",
            "Written 44168 spots for ERR5328033\n",
            "(102)\tGetting 10317.000108496 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18047336']\n",
            "\tFound 1 SRA ID: ['13601706']\n",
            "\tSelected 1 Run ID: ERR5328992\n",
            "\tDownloading FASTQ...\n",
            "Read 23346 spots for ERR5328992\n",
            "Written 23346 spots for ERR5328992\n",
            "(103)\tGetting 10317.000108903 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18047274']\n",
            "\tFound 1 SRA ID: ['13601741']\n",
            "\tSelected 1 Run ID: ERR5329027\n",
            "\tDownloading FASTQ...\n",
            "Read 23250 spots for ERR5329027\n",
            "Written 23250 spots for ERR5329027\n",
            "(104)\tGetting 10317.000108906 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18047334']\n",
            "\tFound 1 SRA ID: ['13601742']\n",
            "\tSelected 1 Run ID: ERR5329028\n",
            "\tDownloading FASTQ...\n",
            "Read 19996 spots for ERR5329028\n",
            "Written 19996 spots for ERR5329028\n",
            "(105)\tGetting 10317.000109791 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18047300']\n",
            "\tFound 1 SRA ID: ['13601781']\n",
            "\tSelected 1 Run ID: ERR5329067\n",
            "\tDownloading FASTQ...\n",
            "Read 21709 spots for ERR5329067\n",
            "Written 21709 spots for ERR5329067\n",
            "(106)\tGetting 10317.000109807 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046629']\n",
            "\tFound 1 SRA ID: ['13603622']\n",
            "\tSelected 1 Run ID: ERR5332367\n",
            "\tDownloading FASTQ...\n",
            "Read 25177 spots for ERR5332367\n",
            "Written 25177 spots for ERR5332367\n",
            "(107)\tGetting 10317.000110177 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046932']\n",
            "\tFound 1 SRA ID: ['13601829']\n",
            "\tSelected 1 Run ID: ERR5329115\n",
            "\tDownloading FASTQ...\n",
            "Read 31256 spots for ERR5329115\n",
            "Written 31256 spots for ERR5329115\n",
            "(108)\tGetting 10317.000112912 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046822']\n",
            "\tFound 2 SRA ID: ['13604310', '13603725']\n",
            "\tSelected 1 Run ID: ERR5333108\n",
            "\tDownloading FASTQ...\n",
            "Read 30634 spots for ERR5333108\n",
            "Written 30634 spots for ERR5333108\n",
            "(109)\tGetting 10317.000113088 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18045873']\n",
            "\tFound 1 SRA ID: ['13603982']\n",
            "\tSelected 1 Run ID: ERR5332725\n",
            "\tDownloading FASTQ...\n",
            "Read 19828 spots for ERR5332725\n",
            "Written 19828 spots for ERR5332725\n",
            "(110)\tGetting 10317.000115199 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18166725']\n",
            "\tFound 1 SRA ID: ['13609683']\n",
            "\tSelected 1 Run ID: ERR5342276\n",
            "\tDownloading FASTQ...\n",
            "Read 49848 spots for ERR5342276\n",
            "Written 49848 spots for ERR5342276\n"
          ]
        }
      ],
      "source": [
        "dict_treatment = get_fastq(data_treatment.sample_name, download = True)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Control group fastq pulling"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 47,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "(1)\tGetting 10317.000001166 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353694']\n",
            "\tFound 1 SRA ID: ['4279766']\n",
            "\tSelected 1 Run ID: ERR2032461\n",
            "\tDownloading FASTQ...\n",
            "Read 36116 spots for ERR2032461\n",
            "Written 36116 spots for ERR2032461\n",
            "(2)\tGetting 10317.000003100 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618395']\n",
            "\tFound 1 SRA ID: ['10591931']\n",
            "\tSelected 1 Run ID: ERR4018879\n",
            "\tDownloading FASTQ...\n",
            "Read 27847 spots for ERR4018879\n",
            "Written 27847 spots for ERR4018879\n",
            "(3)\tGetting 10317.000004936 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353699']\n",
            "\tFound 2 SRA ID: ['4279771', '5162666']\n",
            "\tSelected 1 Run ID: ERR2032466\n",
            "\tDownloading FASTQ...\n",
            "Read 5 spots for ERR2032466\n",
            "Written 5 spots for ERR2032466\n",
            "(4)\tGetting 10317.000010862 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353704']\n",
            "\tFound 1 SRA ID: ['4279776']\n",
            "\tSelected 1 Run ID: ERR2032471\n",
            "\tDownloading FASTQ...\n",
            "Read 37468 spots for ERR2032471\n",
            "Written 37468 spots for ERR2032471\n",
            "(5)\tGetting 10317.000012018 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048072']\n",
            "\tFound 1 SRA ID: ['13598743']\n",
            "\tSelected 1 Run ID: ERR5325492\n",
            "\tDownloading FASTQ...\n",
            "Read 19774 spots for ERR5325492\n",
            "Written 19774 spots for ERR5325492\n",
            "(6)\tGetting 10317.000013727 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618399']\n",
            "\tFound 1 SRA ID: ['10591935']\n",
            "\tSelected 1 Run ID: ERR4018883\n",
            "\tDownloading FASTQ...\n",
            "Read 39345 spots for ERR4018883\n",
            "Written 39345 spots for ERR4018883\n",
            "(7)\tGetting 10317.000014885 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8576909']\n",
            "\tFound 1 SRA ID: ['5144175']\n",
            "\tSelected 1 Run ID: ERR2309175\n",
            "\tDownloading FASTQ...\n",
            "Read 45754 spots for ERR2309175\n",
            "Written 45754 spots for ERR2309175\n",
            "(8)\tGetting 10317.000017272 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614558']\n",
            "\tFound 5 SRA ID: ['14635607', '14635300', '14635285', '14630999', '5162671']\n",
            "\tSelected 1 Run ID: ERR2319231\n",
            "\tDownloading FASTQ...\n",
            "Read 31174 spots for ERR2319231\n",
            "Written 31174 spots for ERR2319231\n",
            "(9)\tGetting 10317.000020655 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439067']\n",
            "\tFound 4 SRA ID: ['4337128', '4939685', '4932705', '4932429']\n",
            "\tSelected 1 Run ID: ERR2056767\n",
            "\tDownloading FASTQ...\n",
            "Read 19066 spots for ERR2056767\n",
            "Written 19066 spots for ERR2056767\n",
            "(10)\tGetting 10317.000020667 Biosample ID\n",
            "No BiosampleID found\n",
            "(11)\tGetting 10317.000022184 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439071']\n",
            "\tFound 4 SRA ID: ['4337132', '4943479', '4939688', '4932432']\n",
            "\tSelected 1 Run ID: ERR2056771\n",
            "\tDownloading FASTQ...\n",
            "Read 27322 spots for ERR2056771\n",
            "Written 27322 spots for ERR2056771\n",
            "(12)\tGetting 10317.000022185 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439072']\n",
            "\tFound 4 SRA ID: ['4337133', '4943480', '4939689', '4932433']\n",
            "\tSelected 1 Run ID: ERR2056772\n",
            "\tDownloading FASTQ...\n",
            "Read 20489 spots for ERR2056772\n",
            "Written 20489 spots for ERR2056772\n",
            "(13)\tGetting 10317.000022186 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439073']\n",
            "\tFound 3 SRA ID: ['4337134', '4943481', '4932434']\n",
            "\tSelected 1 Run ID: ERR2056773\n",
            "\tDownloading FASTQ...\n",
            "Read 31481 spots for ERR2056773\n",
            "Written 31481 spots for ERR2056773\n",
            "(14)\tGetting 10317.000022239 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614563']\n",
            "\tFound 1 SRA ID: ['5162676']\n",
            "\tSelected 1 Run ID: ERR2319236\n",
            "\tDownloading FASTQ...\n",
            "Read 37841 spots for ERR2319236\n",
            "Written 37841 spots for ERR2319236\n",
            "(15)\tGetting 10317.000022282 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614567']\n",
            "\tFound 1 SRA ID: ['5162680']\n",
            "\tSelected 1 Run ID: ERR2319240\n",
            "\tDownloading FASTQ...\n",
            "Read 27980 spots for ERR2319240\n",
            "Written 27980 spots for ERR2319240\n",
            "(16)\tGetting 10317.000022400 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614577']\n",
            "\tFound 1 SRA ID: ['5162690']\n",
            "\tSelected 1 Run ID: ERR2319250\n",
            "\tDownloading FASTQ...\n",
            "Read 40026 spots for ERR2319250\n",
            "Written 40026 spots for ERR2319250\n",
            "(17)\tGetting 10317.000022404 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8568860']\n",
            "(18)\tGetting 10317.000022548 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048120']\n",
            "\tFound 1 SRA ID: ['13676209']\n",
            "\tSelected 1 Run ID: ERR5325493\n",
            "\tDownloading FASTQ...\n",
            "Read 20233 spots for ERR5325493\n",
            "Written 20233 spots for ERR5325493\n",
            "(19)\tGetting 10317.000023134 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046054']\n",
            "\tFound 1 SRA ID: ['13600857']\n",
            "\tSelected 1 Run ID: ERR5327712\n",
            "\tDownloading FASTQ...\n",
            "Read 18731 spots for ERR5327712\n",
            "Written 18731 spots for ERR5327712\n",
            "(20)\tGetting 10317.000023566 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614583']\n",
            "\tFound 1 SRA ID: ['5162696']\n",
            "\tSelected 1 Run ID: ERR2319256\n",
            "\tDownloading FASTQ...\n",
            "Read 28192 spots for ERR2319256\n",
            "Written 28192 spots for ERR2319256\n",
            "(21)\tGetting 10317.000023571 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048759']\n",
            "\tFound 1 SRA ID: ['13599327']\n",
            "\tSelected 1 Run ID: ERR5326131\n",
            "\tDownloading FASTQ...\n",
            "Read 19138 spots for ERR5326131\n",
            "Written 19138 spots for ERR5326131\n",
            "(22)\tGetting 10317.000023577 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8568868']\n",
            "\tFound 1 SRA ID: ['5125214']\n",
            "\tSelected 1 Run ID: ERR2303816\n",
            "\tDownloading FASTQ...\n",
            "Read 29912 spots for ERR2303816\n",
            "Written 29912 spots for ERR2303816\n",
            "(23)\tGetting 10317.000023586 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614586']\n",
            "\tFound 1 SRA ID: ['5162699']\n",
            "\tSelected 1 Run ID: ERR2319259\n",
            "\tDownloading FASTQ...\n",
            "Read 18660 spots for ERR2319259\n",
            "Written 18660 spots for ERR2319259\n",
            "(24)\tGetting 10317.000023611 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614593']\n",
            "\tFound 1 SRA ID: ['5162706']\n",
            "\tSelected 1 Run ID: ERR2319266\n",
            "\tDownloading FASTQ...\n",
            "Read 44105 spots for ERR2319266\n",
            "Written 44105 spots for ERR2319266\n",
            "(25)\tGetting 10317.000023614 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614596']\n",
            "\tFound 1 SRA ID: ['5162709']\n",
            "\tSelected 1 Run ID: ERR2319269\n",
            "\tDownloading FASTQ...\n",
            "Read 56223 spots for ERR2319269\n",
            "Written 56223 spots for ERR2319269\n",
            "(26)\tGetting 10317.000023698 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614599']\n",
            "\tFound 1 SRA ID: ['5162712']\n",
            "\tSelected 1 Run ID: ERR2319272\n",
            "\tDownloading FASTQ...\n",
            "Read 41132 spots for ERR2319272\n",
            "Written 41132 spots for ERR2319272\n",
            "(27)\tGetting 10317.000023718 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614603']\n",
            "\tFound 1 SRA ID: ['5162716']\n",
            "\tSelected 1 Run ID: ERR2319276\n",
            "\tDownloading FASTQ...\n",
            "Read 28642 spots for ERR2319276\n",
            "Written 28642 spots for ERR2319276\n",
            "(28)\tGetting 10317.000023719 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614201']\n",
            "\tFound 2 SRA ID: ['5162717', '5162304']\n",
            "\tSelected 1 Run ID: ERR2319277\n",
            "\tDownloading FASTQ...\n",
            "Read 34 spots for ERR2319277\n",
            "Written 34 spots for ERR2319277\n",
            "(29)\tGetting 10317.000027676 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048082']\n",
            "\tFound 1 SRA ID: ['13598750']\n",
            "\tSelected 1 Run ID: ERR5325495\n",
            "\tDownloading FASTQ...\n",
            "Read 28579 spots for ERR5325495\n",
            "Written 28579 spots for ERR5325495\n",
            "(30)\tGetting 10317.000028319 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18169541']\n",
            "\tFound 1 SRA ID: ['13606066']\n",
            "\tSelected 1 Run ID: ERR5337729\n",
            "\tDownloading FASTQ...\n",
            "Read 41269 spots for ERR5337729\n",
            "Written 41269 spots for ERR5337729\n",
            "(31)\tGetting 10317.000028429 Biosample ID\n",
            "No BiosampleID found\n",
            "(32)\tGetting 10317.000028606 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18046083']\n",
            "\tFound 1 SRA ID: ['13600863']\n",
            "\tSelected 1 Run ID: ERR5327718\n",
            "\tDownloading FASTQ...\n",
            "Read 27629 spots for ERR5327718\n",
            "Written 27629 spots for ERR5327718\n",
            "(33)\tGetting 10317.000028713 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353739']\n",
            "(34)\tGetting 10317.000029491 Biosample ID\n",
            "No BiosampleID found\n",
            "(35)\tGetting 10317.000029517 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8576355']\n",
            "\tFound 1 SRA ID: ['5143630']\n",
            "\tSelected 1 Run ID: ERR2308525\n",
            "\tDownloading FASTQ...\n",
            "Read 18754 spots for ERR2308525\n",
            "Written 18754 spots for ERR2308525\n",
            "(36)\tGetting 10317.000029560 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6365986']\n",
            "\tFound 1 SRA ID: ['3736367']\n",
            "\tSelected 1 Run ID: ERR1842684\n",
            "\tDownloading FASTQ...\n",
            "Read 16396 spots for ERR1842684\n",
            "Written 16396 spots for ERR1842684\n",
            "(37)\tGetting 10317.000030451 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657422']\n",
            "\tFound 1 SRA ID: ['5964114']\n",
            "\tSelected 1 Run ID: ERR2697554\n",
            "\tDownloading FASTQ...\n",
            "Read 23316 spots for ERR2697554\n",
            "Written 23316 spots for ERR2697554\n",
            "(38)\tGetting 10317.000030452 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657423']\n",
            "\tFound 1 SRA ID: ['5964115']\n",
            "\tSelected 1 Run ID: ERR2697555\n",
            "\tDownloading FASTQ...\n",
            "Read 27024 spots for ERR2697555\n",
            "Written 27024 spots for ERR2697555\n",
            "(39)\tGetting 10317.000030453 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657424']\n",
            "\tFound 1 SRA ID: ['5964116']\n",
            "\tSelected 1 Run ID: ERR2697556\n",
            "\tDownloading FASTQ...\n",
            "Read 18286 spots for ERR2697556\n",
            "Written 18286 spots for ERR2697556\n",
            "(40)\tGetting 10317.000030729 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439086']\n",
            "\tFound 1 SRA ID: ['4337149']\n",
            "\tSelected 1 Run ID: ERR2056788\n",
            "\tDownloading FASTQ...\n",
            "Read 33053 spots for ERR2056788\n",
            "Written 33053 spots for ERR2056788\n",
            "(41)\tGetting 10317.000031291 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6365993']\n",
            "\tFound 1 SRA ID: ['3736374']\n",
            "\tSelected 1 Run ID: ERR1842691\n",
            "\tDownloading FASTQ...\n",
            "Read 29286 spots for ERR1842691\n",
            "Written 29286 spots for ERR1842691\n",
            "(42)\tGetting 10317.000033024 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353750']\n",
            "\tFound 1 SRA ID: ['4279825']\n",
            "\tSelected 1 Run ID: ERR2032520\n",
            "\tDownloading FASTQ...\n",
            "Read 29033 spots for ERR2032520\n",
            "Written 29033 spots for ERR2032520\n",
            "(43)\tGetting 10317.000033123 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18049478']\n",
            "\tFound 1 SRA ID: ['13603283']\n",
            "\tSelected 1 Run ID: ERR5332028\n",
            "\tDownloading FASTQ...\n",
            "Read 26126 spots for ERR5332028\n",
            "Written 26126 spots for ERR5332028\n",
            "(44)\tGetting 10317.000033125 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18049458']\n",
            "\tFound 1 SRA ID: ['13603285']\n",
            "\tSelected 1 Run ID: ERR5332030\n",
            "\tDownloading FASTQ...\n",
            "Read 21946 spots for ERR5332030\n",
            "Written 21946 spots for ERR5332030\n",
            "(45)\tGetting 10317.000033170 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353752']\n",
            "\tFound 1 SRA ID: ['4279827']\n",
            "\tSelected 1 Run ID: ERR2032522\n",
            "\tDownloading FASTQ...\n",
            "Read 40633 spots for ERR2032522\n",
            "Written 40633 spots for ERR2032522\n",
            "(46)\tGetting 10317.000033178 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18018884']\n",
            "\tFound 1 SRA ID: ['13595204']\n",
            "\tNo RunID found\n",
            "\tDownloading FASTQ...\n",
            "Read 40633 spots for ERR2032522\n",
            "Written 40633 spots for ERR2032522\n",
            "(47)\tGetting 10317.000033209 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6365999']\n",
            "\tFound 1 SRA ID: ['3736381']\n",
            "\tSelected 1 Run ID: ERR1842698\n",
            "\tDownloading FASTQ...\n",
            "Read 7525 spots for ERR1842698\n",
            "Written 7525 spots for ERR1842698\n",
            "(48)\tGetting 10317.000033274 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353757']\n",
            "\tFound 1 SRA ID: ['4279833']\n",
            "\tSelected 1 Run ID: ERR2032528\n",
            "\tDownloading FASTQ...\n",
            "Read 25782 spots for ERR2032528\n",
            "Written 25782 spots for ERR2032528\n",
            "(49)\tGetting 10317.000033275 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353758']\n",
            "\tFound 1 SRA ID: ['4279834']\n",
            "\tSelected 1 Run ID: ERR2032529\n",
            "\tDownloading FASTQ...\n",
            "Read 41382 spots for ERR2032529\n",
            "Written 41382 spots for ERR2032529\n",
            "(50)\tGetting 10317.000033279 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577200']\n",
            "\tFound 1 SRA ID: ['5144890']\n",
            "\tSelected 1 Run ID: ERR2313962\n",
            "\tDownloading FASTQ...\n",
            "Read 53069 spots for ERR2313962\n",
            "Written 53069 spots for ERR2313962\n",
            "(51)\tGetting 10317.000033290 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614614']\n",
            "\tFound 1 SRA ID: ['5162730']\n",
            "\tSelected 1 Run ID: ERR2319290\n",
            "\tDownloading FASTQ...\n",
            "Read 43957 spots for ERR2319290\n",
            "Written 43957 spots for ERR2319290\n",
            "(52)\tGetting 10317.000033363 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9657425']\n",
            "\tFound 1 SRA ID: ['5964117']\n",
            "\tSelected 1 Run ID: ERR2697557\n",
            "\tDownloading FASTQ...\n",
            "Read 20155 spots for ERR2697557\n",
            "Written 20155 spots for ERR2697557\n",
            "(53)\tGetting 10317.000033376 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618415']\n",
            "\tFound 1 SRA ID: ['10591951']\n",
            "\tSelected 1 Run ID: ERR4018899\n",
            "\tDownloading FASTQ...\n",
            "Read 65085 spots for ERR4018899\n",
            "Written 65085 spots for ERR4018899\n",
            "(54)\tGetting 10317.000033762 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366002']\n",
            "\tFound 1 SRA ID: ['3736384']\n",
            "\tSelected 1 Run ID: ERR1842701\n",
            "\tDownloading FASTQ...\n",
            "Read 34735 spots for ERR1842701\n",
            "Written 34735 spots for ERR1842701\n",
            "(55)\tGetting 10317.000036152 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366007']\n",
            "\tFound 1 SRA ID: ['3736389']\n",
            "\tSelected 1 Run ID: ERR1842706\n",
            "\tDownloading FASTQ...\n",
            "Read 30148 spots for ERR1842706\n",
            "Written 30148 spots for ERR1842706\n",
            "(56)\tGetting 10317.000036470 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353762']\n",
            "\tFound 1 SRA ID: ['4279838']\n",
            "\tSelected 1 Run ID: ERR2032533\n",
            "\tDownloading FASTQ...\n",
            "Read 33971 spots for ERR2032533\n",
            "Written 33971 spots for ERR2032533\n",
            "(57)\tGetting 10317.000036669 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366009']\n",
            "\tFound 1 SRA ID: ['3736391']\n",
            "\tSelected 1 Run ID: ERR1842708\n",
            "\tDownloading FASTQ...\n",
            "Read 18937 spots for ERR1842708\n",
            "Written 18937 spots for ERR1842708\n",
            "(58)\tGetting 10317.000039523 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6445651']\n",
            "\tFound 1 SRA ID: ['3747618']\n",
            "\tSelected 1 Run ID: ERR1854671\n",
            "\tDownloading FASTQ...\n",
            "Read 28326 spots for ERR1854671\n",
            "Written 28326 spots for ERR1854671\n",
            "(59)\tGetting 10317.000039524 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6445652']\n",
            "\tFound 1 SRA ID: ['3747619']\n",
            "\tSelected 1 Run ID: ERR1854672\n",
            "\tDownloading FASTQ...\n",
            "Read 21955 spots for ERR1854672\n",
            "Written 21955 spots for ERR1854672\n",
            "(60)\tGetting 10317.000039529 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353767']\n",
            "\tFound 1 SRA ID: ['4279843']\n",
            "\tSelected 1 Run ID: ERR2032538\n",
            "\tDownloading FASTQ...\n",
            "Read 36381 spots for ERR2032538\n",
            "Written 36381 spots for ERR2032538\n",
            "(61)\tGetting 10317.000039535 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353770']\n",
            "\tFound 1 SRA ID: ['4279846']\n",
            "\tSelected 1 Run ID: ERR2032541\n",
            "\tDownloading FASTQ...\n",
            "Read 25891 spots for ERR2032541\n",
            "Written 25891 spots for ERR2032541\n",
            "(62)\tGetting 10317.000039536 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353771']\n",
            "\tFound 1 SRA ID: ['4279847']\n",
            "\tSelected 1 Run ID: ERR2032542\n",
            "\tDownloading FASTQ...\n",
            "Read 13493 spots for ERR2032542\n",
            "Written 13493 spots for ERR2032542\n",
            "(63)\tGetting 10317.000039538 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353773']\n",
            "\tFound 1 SRA ID: ['4279849']\n",
            "\tSelected 1 Run ID: ERR2032544\n",
            "\tDownloading FASTQ...\n",
            "Read 28353 spots for ERR2032544\n",
            "Written 28353 spots for ERR2032544\n",
            "(64)\tGetting 10317.000039540 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353775']\n",
            "\tFound 1 SRA ID: ['4279851']\n",
            "\tSelected 1 Run ID: ERR2032546\n",
            "\tDownloading FASTQ...\n",
            "Read 16103 spots for ERR2032546\n",
            "Written 16103 spots for ERR2032546\n",
            "(65)\tGetting 10317.000039541 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353776']\n",
            "\tFound 1 SRA ID: ['4279852']\n",
            "\tSelected 1 Run ID: ERR2032547\n",
            "\tDownloading FASTQ...\n",
            "Read 10616 spots for ERR2032547\n",
            "Written 10616 spots for ERR2032547\n",
            "(66)\tGetting 10317.000039544 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6445653']\n",
            "\tFound 5 SRA ID: ['14635623', '14635316', '14634825', '14631015', '3747620']\n",
            "\tSelected 1 Run ID: ERR1854673\n",
            "\tDownloading FASTQ...\n",
            "Read 19092 spots for ERR1854673\n",
            "Written 19092 spots for ERR1854673\n",
            "(67)\tGetting 10317.000039549 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353779']\n",
            "\tFound 5 SRA ID: ['14635624', '14635317', '14634826', '14631016', '4279855']\n",
            "\tSelected 1 Run ID: ERR2032550\n",
            "\tDownloading FASTQ...\n",
            "Read 48887 spots for ERR2032550\n",
            "Written 48887 spots for ERR2032550\n",
            "(68)\tGetting 10317.000039550 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353780']\n",
            "\tFound 1 SRA ID: ['4279856']\n",
            "\tSelected 1 Run ID: ERR2032551\n",
            "\tDownloading FASTQ...\n",
            "Read 32990 spots for ERR2032551\n",
            "Written 32990 spots for ERR2032551\n",
            "(69)\tGetting 10317.000039558 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353787']\n",
            "\tFound 1 SRA ID: ['4279863']\n",
            "\tSelected 1 Run ID: ERR2032558\n",
            "\tDownloading FASTQ...\n",
            "Read 36816 spots for ERR2032558\n",
            "Written 36816 spots for ERR2032558\n",
            "(70)\tGetting 10317.000039572 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353790']\n",
            "\tFound 1 SRA ID: ['4279866']\n",
            "\tSelected 1 Run ID: ERR2032561\n",
            "\tDownloading FASTQ...\n",
            "Read 25763 spots for ERR2032561\n",
            "Written 25763 spots for ERR2032561\n",
            "(71)\tGetting 10317.000039579 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18050427']\n",
            "\tFound 1 SRA ID: ['13601330']\n",
            "\tSelected 1 Run ID: ERR5328190\n",
            "\tDownloading FASTQ...\n",
            "Read 15288 spots for ERR5328190\n",
            "Written 15288 spots for ERR5328190\n",
            "(72)\tGetting 10317.000039600 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353799']\n",
            "\tFound 1 SRA ID: ['4279875']\n",
            "\tSelected 1 Run ID: ERR2032570\n",
            "\tDownloading FASTQ...\n",
            "Read 39948 spots for ERR2032570\n",
            "Written 39948 spots for ERR2032570\n",
            "(73)\tGetting 10317.000039612 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577207']\n",
            "\tFound 1 SRA ID: ['5144900']\n",
            "\tSelected 1 Run ID: ERR2313972\n",
            "\tDownloading FASTQ...\n",
            "Read 34663 spots for ERR2313972\n",
            "Written 34663 spots for ERR2313972\n",
            "(74)\tGetting 10317.000039622 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353803']\n",
            "\tFound 1 SRA ID: ['4279879']\n",
            "\tSelected 1 Run ID: ERR2032574\n",
            "\tDownloading FASTQ...\n",
            "Read 55275 spots for ERR2032574\n",
            "Written 55275 spots for ERR2032574\n",
            "(75)\tGetting 10317.000039626 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18050444']\n",
            "\tFound 5 SRA ID: ['14635632', '14635325', '14634834', '14631024', '13601332']\n",
            "\tSelected 1 Run ID: ERR5328192\n",
            "\tDownloading FASTQ...\n",
            "Read 41110 spots for ERR5328192\n",
            "Written 41110 spots for ERR5328192\n",
            "(76)\tGetting 10317.000039629 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7496959']\n",
            "\tFound 1 SRA ID: ['4375992']\n",
            "\tSelected 1 Run ID: ERR2092140\n",
            "\tDownloading FASTQ...\n",
            "Read 16184 spots for ERR2092140\n",
            "Written 16184 spots for ERR2092140\n",
            "(77)\tGetting 10317.000039636 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353805']\n",
            "\tFound 1 SRA ID: ['4279881']\n",
            "\tSelected 1 Run ID: ERR2032576\n",
            "\tDownloading FASTQ...\n",
            "Read 49757 spots for ERR2032576\n",
            "Written 49757 spots for ERR2032576\n",
            "(78)\tGetting 10317.000039642 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353807']\n",
            "\tFound 1 SRA ID: ['4279883']\n",
            "\tSelected 1 Run ID: ERR2032578\n",
            "\tDownloading FASTQ...\n",
            "Read 25052 spots for ERR2032578\n",
            "Written 25052 spots for ERR2032578\n",
            "(79)\tGetting 10317.000039646 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353810']\n",
            "\tFound 5 SRA ID: ['14635635', '14635328', '14634837', '14631027', '4279886']\n",
            "\tSelected 1 Run ID: ERR2032581\n",
            "\tDownloading FASTQ...\n",
            "Read 33623 spots for ERR2032581\n",
            "Written 33623 spots for ERR2032581\n",
            "(80)\tGetting 10317.000039664 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614627']\n",
            "\tFound 1 SRA ID: ['5162743']\n",
            "\tSelected 1 Run ID: ERR2319303\n",
            "\tDownloading FASTQ...\n",
            "Read 48180 spots for ERR2319303\n",
            "Written 48180 spots for ERR2319303\n",
            "(81)\tGetting 10317.000039677 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354223']\n",
            "\tFound 1 SRA ID: ['4280313']\n",
            "\tSelected 1 Run ID: ERR2033212\n",
            "\tDownloading FASTQ...\n",
            "Read 25538 spots for ERR2033212\n",
            "Written 25538 spots for ERR2033212\n",
            "(82)\tGetting 10317.000039685 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6445673']\n",
            "\tFound 1 SRA ID: ['3747634']\n",
            "\tSelected 1 Run ID: ERR1854687\n",
            "\tDownloading FASTQ...\n",
            "Read 3249 spots for ERR1854687\n",
            "Written 3249 spots for ERR1854687\n",
            "(83)\tGetting 10317.000039686 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6445675']\n",
            "\tFound 1 SRA ID: ['3747635']\n",
            "\tSelected 1 Run ID: ERR1854688\n",
            "\tDownloading FASTQ...\n",
            "Read 24899 spots for ERR1854688\n",
            "Written 24899 spots for ERR1854688\n",
            "(84)\tGetting 10317.000039689 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353812']\n",
            "\tFound 1 SRA ID: ['4279888']\n",
            "\tSelected 1 Run ID: ERR2032583\n",
            "\tDownloading FASTQ...\n",
            "Read 22069 spots for ERR2032583\n",
            "Written 22069 spots for ERR2032583\n",
            "(85)\tGetting 10317.000039693 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353816']\n",
            "\tFound 1 SRA ID: ['4279892']\n",
            "\tSelected 1 Run ID: ERR2032587\n",
            "\tDownloading FASTQ...\n",
            "Read 20950 spots for ERR2032587\n",
            "Written 20950 spots for ERR2032587\n",
            "(86)\tGetting 10317.000039701 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6445679']\n",
            "\tFound 1 SRA ID: ['3747638']\n",
            "\tSelected 1 Run ID: ERR1854691\n",
            "\tDownloading FASTQ...\n",
            "Read 21775 spots for ERR1854691\n",
            "Written 21775 spots for ERR1854691\n",
            "(87)\tGetting 10317.000039719 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614630']\n",
            "\tFound 1 SRA ID: ['5162746']\n",
            "\tSelected 1 Run ID: ERR2319306\n",
            "\tDownloading FASTQ...\n",
            "Read 39414 spots for ERR2319306\n",
            "Written 39414 spots for ERR2319306\n",
            "(88)\tGetting 10317.000039733 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439100']\n",
            "\tFound 3 SRA ID: ['4337173', '4943486', '4932442']\n",
            "\tSelected 1 Run ID: ERR2056812\n",
            "\tDownloading FASTQ...\n",
            "Read 28953 spots for ERR2056812\n",
            "Written 28953 spots for ERR2056812\n",
            "(89)\tGetting 10317.000039746 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8576920']\n",
            "\tFound 1 SRA ID: ['5144187']\n",
            "\tSelected 1 Run ID: ERR2309187\n",
            "\tDownloading FASTQ...\n",
            "Read 41910 spots for ERR2309187\n",
            "Written 41910 spots for ERR2309187\n",
            "(90)\tGetting 10317.000039750 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8569428']\n",
            "\tFound 5 SRA ID: ['14635644', '14635337', '14634846', '14631036', '5134315']\n",
            "\tSelected 1 Run ID: ERR2306094\n",
            "\tDownloading FASTQ...\n",
            "Read 87796 spots for ERR2306094\n",
            "Written 87796 spots for ERR2306094\n",
            "(91)\tGetting 10317.000039754 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8614212']\n",
            "\tFound 1 SRA ID: ['5162316']\n",
            "\tSelected 1 Run ID: ERR2318011\n",
            "\tDownloading FASTQ...\n",
            "Read 48653 spots for ERR2318011\n",
            "Written 48653 spots for ERR2318011\n",
            "(92)\tGetting 10317.000040070 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577218']\n",
            "\tFound 1 SRA ID: ['5144911']\n",
            "\tSelected 1 Run ID: ERR2313983\n",
            "\tDownloading FASTQ...\n",
            "Read 52592 spots for ERR2313983\n",
            "Written 52592 spots for ERR2313983\n",
            "(93)\tGetting 10317.000040182 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353826']\n",
            "\tFound 1 SRA ID: ['4279902']\n",
            "\tSelected 1 Run ID: ERR2032597\n",
            "\tDownloading FASTQ...\n",
            "Read 24395 spots for ERR2032597\n",
            "Written 24395 spots for ERR2032597\n",
            "(94)\tGetting 10317.000040185 Biosample ID\n",
            "\tFound 1 BiosampleID: ['14618426']\n",
            "(95)\tGetting 10317.000041098 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8728557']\n",
            "\tFound 1 SRA ID: ['5245040']\n",
            "\tSelected 1 Run ID: ERR2404914\n",
            "\tDownloading FASTQ...\n",
            "Read 31973 spots for ERR2404914\n",
            "Written 31973 spots for ERR2404914\n",
            "(96)\tGetting 10317.000041290 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048085']\n",
            "\tFound 1 SRA ID: ['13598757']\n",
            "\tSelected 1 Run ID: ERR5325502\n",
            "\tDownloading FASTQ...\n",
            "Read 17121 spots for ERR5325502\n",
            "Written 17121 spots for ERR5325502\n",
            "(97)\tGetting 10317.000041545 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8568886']\n",
            "\tFound 1 SRA ID: ['5125232']\n",
            "\tSelected 1 Run ID: ERR2303834\n",
            "\tDownloading FASTQ...\n",
            "Read 24859 spots for ERR2303834\n",
            "Written 24859 spots for ERR2303834\n",
            "(98)\tGetting 10317.000041787 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366023']\n",
            "\tFound 1 SRA ID: ['3736405']\n",
            "\tSelected 1 Run ID: ERR1842722\n",
            "\tDownloading FASTQ...\n",
            "Read 40861 spots for ERR1842722\n",
            "Written 40861 spots for ERR1842722\n",
            "(99)\tGetting 10317.000041830 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366027']\n",
            "\tFound 1 SRA ID: ['3736409']\n",
            "\tSelected 1 Run ID: ERR1842726\n",
            "\tDownloading FASTQ...\n",
            "Read 12478 spots for ERR1842726\n",
            "Written 12478 spots for ERR1842726\n",
            "(100)\tGetting 10317.000041831 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366028']\n",
            "\tFound 1 SRA ID: ['3736410']\n",
            "\tSelected 1 Run ID: ERR1842727\n",
            "\tDownloading FASTQ...\n",
            "Read 37171 spots for ERR1842727\n",
            "Written 37171 spots for ERR1842727\n",
            "(101)\tGetting 10317.000041832 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366029']\n",
            "\tFound 1 SRA ID: ['3736411']\n",
            "\tSelected 1 Run ID: ERR1842728\n",
            "\tDownloading FASTQ...\n",
            "Read 31170 spots for ERR1842728\n",
            "Written 31170 spots for ERR1842728\n",
            "(102)\tGetting 10317.000041837 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366032']\n",
            "\tFound 1 SRA ID: ['3736414']\n",
            "\tSelected 1 Run ID: ERR1842731\n",
            "\tDownloading FASTQ...\n",
            "Read 18850 spots for ERR1842731\n",
            "Written 18850 spots for ERR1842731\n",
            "(103)\tGetting 10317.000042579 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366033']\n",
            "\tFound 1 SRA ID: ['3736415']\n",
            "\tSelected 1 Run ID: ERR1842732\n",
            "\tDownloading FASTQ...\n",
            "Read 22844 spots for ERR1842732\n",
            "Written 22844 spots for ERR1842732\n",
            "(104)\tGetting 10317.000042849 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18045803']\n",
            "\tFound 1 SRA ID: ['13604013']\n",
            "\tSelected 1 Run ID: ERR5332811\n",
            "\tDownloading FASTQ...\n",
            "Read 33533 spots for ERR5332811\n",
            "Written 33533 spots for ERR5332811\n",
            "(105)\tGetting 10317.000047222 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366062']\n",
            "\tFound 1 SRA ID: ['3736444']\n",
            "\tSelected 1 Run ID: ERR1842761\n",
            "\tDownloading FASTQ...\n",
            "Read 33037 spots for ERR1842761\n",
            "Written 33037 spots for ERR1842761\n",
            "(106)\tGetting 10317.000047387 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366065']\n",
            "\tFound 1 SRA ID: ['3736447']\n",
            "\tSelected 1 Run ID: ERR1842764\n",
            "\tDownloading FASTQ...\n",
            "Read 30837 spots for ERR1842764\n",
            "Written 30837 spots for ERR1842764\n",
            "(107)\tGetting 10317.000047463 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353923']\n",
            "\tFound 1 SRA ID: ['4280003']\n",
            "\tSelected 1 Run ID: ERR2032698\n",
            "\tDownloading FASTQ...\n",
            "Read 21901 spots for ERR2032698\n",
            "Written 21901 spots for ERR2032698\n",
            "(108)\tGetting 10317.000047501 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7354229']\n",
            "\tFound 1 SRA ID: ['4280322']\n",
            "\tSelected 1 Run ID: ERR2033221\n",
            "\tDownloading FASTQ...\n",
            "Read 22634 spots for ERR2033221\n",
            "Written 22634 spots for ERR2033221\n",
            "(109)\tGetting 10317.000047540 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048774']\n",
            "\tFound 1 SRA ID: ['13599342']\n",
            "\tSelected 1 Run ID: ERR5326146\n",
            "\tDownloading FASTQ...\n",
            "Read 25369 spots for ERR5326146\n",
            "Written 25369 spots for ERR5326146\n",
            "(110)\tGetting 10317.000047574 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353929']\n",
            "\tFound 1 SRA ID: ['4280009']\n",
            "\tSelected 1 Run ID: ERR2032704\n",
            "\tDownloading FASTQ...\n",
            "Read 29449 spots for ERR2032704\n",
            "Written 29449 spots for ERR2032704\n",
            "(111)\tGetting 10317.000051556 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18048055']\n",
            "\tFound 1 SRA ID: ['13598762']\n",
            "\tSelected 1 Run ID: ERR5325507\n",
            "\tDownloading FASTQ...\n",
            "Read 23235 spots for ERR5325507\n",
            "Written 23235 spots for ERR5325507\n",
            "(112)\tGetting 10317.000052116 Biosample ID\n",
            "\tFound 1 BiosampleID: ['18169542']\n",
            "\tFound 1 SRA ID: ['13606078']\n",
            "\tSelected 1 Run ID: ERR5337741\n",
            "\tDownloading FASTQ...\n",
            "Read 41157 spots for ERR5337741\n",
            "Written 41157 spots for ERR5337741\n",
            "(113)\tGetting 10317.000052323 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7439140']\n",
            "\tFound 4 SRA ID: ['4337234', '4943490', '4939705', '4932455']\n",
            "\tSelected 1 Run ID: ERR2056873\n",
            "\tDownloading FASTQ...\n",
            "Read 37164 spots for ERR2056873\n",
            "Written 37164 spots for ERR2056873\n",
            "(114)\tGetting 10317.000052472 Biosample ID\n",
            "\tFound 1 BiosampleID: ['9541145']\n",
            "\tFound 1 SRA ID: ['5840428']\n",
            "\tSelected 1 Run ID: ERR2579690\n",
            "\tDownloading FASTQ...\n",
            "Read 69941 spots for ERR2579690\n",
            "Written 69941 spots for ERR2579690\n",
            "(115)\tGetting 10317.000052511 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8577247']\n",
            "\tFound 1 SRA ID: ['5144958']\n",
            "\tSelected 1 Run ID: ERR2314030\n",
            "\tDownloading FASTQ...\n",
            "Read 42978 spots for ERR2314030\n",
            "Written 42978 spots for ERR2314030\n",
            "(116)\tGetting 10317.000053409 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8568900']\n",
            "\tFound 1 SRA ID: ['5125246']\n",
            "\tSelected 1 Run ID: ERR2303848\n",
            "\tDownloading FASTQ...\n",
            "Read 30646 spots for ERR2303848\n",
            "Written 30646 spots for ERR2303848\n",
            "(117)\tGetting 10317.000058850 Biosample ID\n",
            "\tFound 1 BiosampleID: ['7353979']\n",
            "\tFound 1 SRA ID: ['4280060']\n",
            "\tSelected 1 Run ID: ERR2032755\n",
            "\tDownloading FASTQ...\n",
            "Read 23632 spots for ERR2032755\n",
            "Written 23632 spots for ERR2032755\n",
            "(118)\tGetting 10317.000059011 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366198']\n",
            "\tFound 1 SRA ID: ['3736580']\n",
            "\tSelected 1 Run ID: ERR1842897\n",
            "\tDownloading FASTQ...\n",
            "Read 28549 spots for ERR1842897\n",
            "Written 28549 spots for ERR1842897\n",
            "(119)\tGetting 10317.000059235 Biosample ID\n",
            "\tFound 1 BiosampleID: ['8945359']\n",
            "\tFound 1 SRA ID: ['5424261']\n",
            "\tSelected 1 Run ID: ERR2523877\n",
            "\tDownloading FASTQ...\n",
            "Read 15366 spots for ERR2523877\n",
            "Written 15366 spots for ERR2523877\n",
            "(120)\tGetting 10317.000062080 Biosample ID\n",
            "\tFound 1 BiosampleID: ['6366218']\n",
            "\tFound 1 SRA ID: ['3736600']\n",
            "\tSelected 1 Run ID: ERR1842917\n",
            "\tDownloading FASTQ...\n",
            "Read 24508 spots for ERR1842917\n",
            "Written 24508 spots for ERR1842917\n"
          ]
        }
      ],
      "source": [
        "dict_control = get_fastq(data_control.sample_name, download = True)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Id dataframe for both groups"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 48,
      "metadata": {},
      "outputs": [],
      "source": [
        "treatment_fastq = pd.DataFrame.from_dict(dict_treatment)\n",
        "control_fastq = pd.DataFrame.from_dict(dict_control)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Selecting only subjects for whom fastq was found"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 49,
      "metadata": {},
      "outputs": [],
      "source": [
        "treatment_fastq = treatment_fastq.loc[treatment_fastq['run_ids'] != \"Not found\"].reset_index(drop= True)\n",
        "control_fastq = control_fastq.loc[control_fastq['run_ids'] != \"Not found\"].reset_index(drop= True)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Selecting only 100 subjects for each group"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 50,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_control = data_control.loc[data_control['sample_name'].isin(control_fastq.sample_names)]\n",
        "data_treatment = data_treatment.loc[data_treatment['sample_name'].isin(treatment_fastq.sample_names)]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 51,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_control = data_control.groupby([\"sex\",\"age_cat\"]).head(10).reset_index(drop=True)\n",
        "data_treatment = data_treatment.head(100).reset_index(drop=True)"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Concatenation of the two groups"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 52,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_final_groups = pd.concat([data_treatment,data_control]).reset_index(drop=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 53,
      "metadata": {},
      "outputs": [],
      "source": [
        "control_fastq = control_fastq.loc[control_fastq.sample_names.isin(data_control['sample_name'])]\n",
        "treatment_fastq = treatment_fastq.loc[treatment_fastq.sample_names.isin(data_treatment['sample_name'])]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 54,
      "metadata": {},
      "outputs": [],
      "source": [
        "all_fastq = pd.concat([control_fastq,treatment_fastq]).reset_index(drop=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 55,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>sample_names</th>\n",
              "      <th>biosample_ids</th>\n",
              "      <th>run_ids</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>10317.000001166</td>\n",
              "      <td>7353694</td>\n",
              "      <td>ERR2032461</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>10317.000003100</td>\n",
              "      <td>14618395</td>\n",
              "      <td>ERR4018879</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>10317.000004936</td>\n",
              "      <td>7353699</td>\n",
              "      <td>ERR2032466</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>10317.000010862</td>\n",
              "      <td>7353704</td>\n",
              "      <td>ERR2032471</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>10317.000012018</td>\n",
              "      <td>18048072</td>\n",
              "      <td>ERR5325492</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>195</th>\n",
              "      <td>10317.000108434</td>\n",
              "      <td>18050057</td>\n",
              "      <td>ERR5328033</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>196</th>\n",
              "      <td>10317.000108496</td>\n",
              "      <td>18047336</td>\n",
              "      <td>ERR5328992</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>197</th>\n",
              "      <td>10317.000108903</td>\n",
              "      <td>18047274</td>\n",
              "      <td>ERR5329027</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>198</th>\n",
              "      <td>10317.000108906</td>\n",
              "      <td>18047334</td>\n",
              "      <td>ERR5329028</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>199</th>\n",
              "      <td>10317.000109791</td>\n",
              "      <td>18047300</td>\n",
              "      <td>ERR5329067</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>200 rows × 3 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "        sample_names biosample_ids     run_ids\n",
              "0    10317.000001166       7353694  ERR2032461\n",
              "1    10317.000003100      14618395  ERR4018879\n",
              "2    10317.000004936       7353699  ERR2032466\n",
              "3    10317.000010862       7353704  ERR2032471\n",
              "4    10317.000012018      18048072  ERR5325492\n",
              "..               ...           ...         ...\n",
              "195  10317.000108434      18050057  ERR5328033\n",
              "196  10317.000108496      18047336  ERR5328992\n",
              "197  10317.000108903      18047274  ERR5329027\n",
              "198  10317.000108906      18047334  ERR5329028\n",
              "199  10317.000109791      18047300  ERR5329067\n",
              "\n",
              "[200 rows x 3 columns]"
            ]
          },
          "execution_count": 55,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "all_fastq"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Saving the dataframe with all needed info"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 57,
      "metadata": {},
      "outputs": [],
      "source": [
        "data_final_groups = pd.merge(data_final_groups,\n",
        "                             all_fastq,\n",
        "                             how = \"inner\",\n",
        "                             left_on = \"sample_name\",\n",
        "                             right_on=\"sample_names\")\n",
        "data_final_groups.to_csv(\"data_final_groups.csv\")"
      ]
    },
    {
      "attachments": {},
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# FASTQ - Quality check"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 58,
      "metadata": {},
      "outputs": [],
      "source": [
        "from Bio import SeqIO\n",
        "from os import listdir\n",
        "from os.path import isfile, join\n",
        "import os\n",
        "import pandas as pd"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 59,
      "metadata": {},
      "outputs": [],
      "source": [
        "def filter_quality(fastq_path,min_quality = 20,verbose = True):\n",
        "    \n",
        "    tot_reads = []\n",
        "    saved_reads = []\n",
        "\n",
        "    if not os.path.exists('good_quality_' + str(min_quality)):\n",
        "        os.mkdir(f'good_quality_{str(min_quality)}')\n",
        "    \n",
        "    for f in os.listdir(fastq_path):\n",
        "        print(f\"Reading {f}\")\n",
        "        num_reads = 0 \n",
        "        for rec in SeqIO.parse(os.path.join(fastq_path,f), \"fastq\"):\n",
        "            num_reads +=1\n",
        "            good_reads = (rec for rec in \\\n",
        "                            SeqIO.parse(os.path.join(fastq_path,f), \"fastq\") \\\n",
        "                                if min(rec.letter_annotations[\"phred_quality\"]) >= min_quality)\n",
        "\n",
        "        count = SeqIO.write(good_reads, f\"good_quality_{str(min_quality)}/{f}\", \"fastq\")\n",
        "        print(f\"Saved {count} reads\")\n",
        "        print(f\"{num_reads} reads\\n\\n\")\n",
        "        tot_reads.append(num_reads)\n",
        "        saved_reads.append(count)\n",
        "    \n",
        "    return os.listdir(fastq_path), tot_reads, saved_reads\n",
        "    \n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 60,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading ERR2579789.fastq\n",
            "Saved 822 reads\n",
            "3583 reads\n",
            "\n",
            "\n",
            "Reading ERR2032558.fastq\n",
            "Saved 11121 reads\n",
            "36816 reads\n",
            "\n",
            "\n",
            "Reading ERR2318153.fastq\n",
            "Saved 6909 reads\n",
            "36283 reads\n",
            "\n",
            "\n",
            "Reading ERR5328190.fastq\n",
            "Saved 4367 reads\n",
            "15288 reads\n",
            "\n",
            "\n",
            "Reading ERR2032538.fastq\n",
            "Saved 9420 reads\n",
            "36381 reads\n",
            "\n",
            "\n",
            "Reading ERR5326482.fastq\n",
            "Saved 9406 reads\n",
            "20175 reads\n",
            "\n",
            "\n",
            "Reading ERR2404913.fastq\n",
            "Saved 16881 reads\n",
            "34268 reads\n",
            "\n",
            "\n",
            "Reading ERR2314152.fastq\n",
            "Saved 30183 reads\n",
            "49733 reads\n",
            "\n",
            "\n",
            "Reading ERR5332232.fastq\n",
            "Saved 8168 reads\n",
            "21965 reads\n",
            "\n",
            "\n",
            "Reading ERR1842727.fastq\n",
            "Saved 162 reads\n",
            "37171 reads\n",
            "\n",
            "\n",
            "Reading ERR1854687.fastq\n",
            "Saved 212 reads\n",
            "3249 reads\n",
            "\n",
            "\n",
            "Reading ERR5328798.fastq\n",
            "Saved 6696 reads\n",
            "14172 reads\n",
            "\n",
            "\n",
            "Reading ERR2579848.fastq\n",
            "Saved 8218 reads\n",
            "38207 reads\n",
            "\n",
            "\n",
            "Reading ERR1842731.fastq\n",
            "Saved 22 reads\n",
            "18850 reads\n",
            "\n",
            "\n",
            "Reading ERR2032818.fastq\n",
            "Saved 7711 reads\n",
            "20563 reads\n",
            "\n",
            "\n",
            "Reading ERR2405327.fastq\n",
            "Saved 8663 reads\n",
            "17131 reads\n",
            "\n",
            "\n",
            "Reading ERR2523877.fastq\n",
            "Saved 7210 reads\n",
            "15366 reads\n",
            "\n",
            "\n",
            "Reading ERR2304018.fastq\n",
            "Saved 13434 reads\n",
            "27533 reads\n",
            "\n",
            "\n",
            "Reading ERR5327749.fastq\n",
            "Saved 1528 reads\n",
            "20192 reads\n",
            "\n",
            "\n",
            "Reading ERR2314030.fastq\n",
            "Saved 26204 reads\n",
            "42978 reads\n",
            "\n",
            "\n",
            "Reading ERR2319240.fastq\n",
            "Saved 9432 reads\n",
            "27980 reads\n",
            "\n",
            "\n",
            "Reading ERR2056788.fastq\n",
            "Saved 9117 reads\n",
            "33053 reads\n",
            "\n",
            "\n",
            "Reading ERR2032576.fastq\n",
            "Saved 15205 reads\n",
            "49757 reads\n",
            "\n",
            "\n",
            "Reading ERR2308525.fastq\n",
            "Saved 8409 reads\n",
            "18754 reads\n",
            "\n",
            "\n",
            "Reading ERR2303921.fastq\n",
            "Saved 15418 reads\n",
            "31496 reads\n",
            "\n",
            "\n",
            "Reading ERR2056873.fastq\n",
            "Saved 11979 reads\n",
            "37164 reads\n",
            "\n",
            "\n",
            "Reading ERR5327796.fastq\n",
            "Saved 2379 reads\n",
            "23637 reads\n",
            "\n",
            "\n",
            "Reading ERR2319250.fastq\n",
            "Saved 11788 reads\n",
            "40026 reads\n",
            "\n",
            "\n",
            "Reading ERR2313983.fastq\n",
            "Saved 33418 reads\n",
            "52592 reads\n",
            "\n",
            "\n",
            "Reading ERR2319266.fastq\n",
            "Saved 12775 reads\n",
            "44105 reads\n",
            "\n",
            "\n",
            "Reading ERR2404914.fastq\n",
            "Saved 17216 reads\n",
            "31973 reads\n",
            "\n",
            "\n",
            "Reading ERR1842917.fastq\n",
            "Saved 55 reads\n",
            "24508 reads\n",
            "\n",
            "\n",
            "Reading ERR4019252.fastq\n",
            "Saved 9883 reads\n",
            "24589 reads\n",
            "\n",
            "\n",
            "Reading ERR2697862.fastq\n",
            "Saved 7246 reads\n",
            "18485 reads\n",
            "\n",
            "\n",
            "Reading ERR2318152.fastq\n",
            "Saved 20 reads\n",
            "57 reads\n",
            "\n",
            "\n",
            "Reading ERR2032561.fastq\n",
            "Saved 7496 reads\n",
            "25763 reads\n",
            "\n",
            "\n",
            "Reading ERR2303834.fastq\n",
            "Saved 12614 reads\n",
            "24859 reads\n",
            "\n",
            "\n",
            "Reading ERR1842877.fastq\n",
            "Saved 24 reads\n",
            "17826 reads\n",
            "\n",
            "\n",
            "Reading ERR2314109.fastq\n",
            "Saved 21704 reads\n",
            "34149 reads\n",
            "\n",
            "\n",
            "Reading ERR2580056.fastq\n",
            "Saved 8270 reads\n",
            "34401 reads\n",
            "\n",
            "\n",
            "Reading ERR2314331.fastq\n",
            "Saved 23334 reads\n",
            "38741 reads\n",
            "\n",
            "\n",
            "Reading ERR2579425.fastq\n",
            "Saved 13131 reads\n",
            "26033 reads\n",
            "\n",
            "\n",
            "Reading ERR2318151.fastq\n",
            "Saved 7735 reads\n",
            "32755 reads\n",
            "\n",
            "\n",
            "Reading ERR2697555.fastq\n",
            "Saved 10750 reads\n",
            "27024 reads\n",
            "\n",
            "\n",
            "Reading ERR2318007.fastq\n",
            "Saved 6413 reads\n",
            "39992 reads\n",
            "\n",
            "\n",
            "Reading ERR2306107.fastq\n",
            "Saved 5314 reads\n",
            "75366 reads\n",
            "\n",
            "\n",
            "Reading ERR2032541.fastq\n",
            "Saved 8103 reads\n",
            "25891 reads\n",
            "\n",
            "\n",
            "Reading ERR5329028.fastq\n",
            "Saved 7304 reads\n",
            "19996 reads\n",
            "\n",
            "\n",
            "Reading ERR2032544.fastq\n",
            "Saved 8264 reads\n",
            "28353 reads\n",
            "\n",
            "\n",
            "Reading ERR4018879.fastq\n",
            "Saved 2056 reads\n",
            "27847 reads\n",
            "\n",
            "\n",
            "Reading ERR2032522.fastq\n",
            "Saved 14146 reads\n",
            "40633 reads\n",
            "\n",
            "\n",
            "Reading ERR5325502.fastq\n",
            "Saved 9656 reads\n",
            "17121 reads\n",
            "\n",
            "\n",
            "Reading ERR4018935.fastq\n",
            "Saved 3832 reads\n",
            "31148 reads\n",
            "\n",
            "\n",
            "Reading ERR2579790.fastq\n",
            "Saved 4421 reads\n",
            "18951 reads\n",
            "\n",
            "\n",
            "Reading ERR2033467.fastq\n",
            "Saved 2269 reads\n",
            "30274 reads\n",
            "\n",
            "\n",
            "Reading ERR2056812.fastq\n",
            "Saved 9934 reads\n",
            "28953 reads\n",
            "\n",
            "\n",
            "Reading ERR2697557.fastq\n",
            "Saved 8197 reads\n",
            "20155 reads\n",
            "\n",
            "\n",
            "Reading ERR5328913.fastq\n",
            "Saved 11027 reads\n",
            "27063 reads\n",
            "\n",
            "\n",
            "Reading ERR4019282.fastq\n",
            "Saved 9600 reads\n",
            "27182 reads\n",
            "\n",
            "\n",
            "Reading ERR2032583.fastq\n",
            "Saved 6980 reads\n",
            "22069 reads\n",
            "\n",
            "\n",
            "Reading ERR5325507.fastq\n",
            "Saved 11187 reads\n",
            "23235 reads\n",
            "\n",
            "\n",
            "Reading ERR1842808.fastq\n",
            "Saved 98 reads\n",
            "16660 reads\n",
            "\n",
            "\n",
            "Reading ERR2314352.fastq\n",
            "Saved 21139 reads\n",
            "32221 reads\n",
            "\n",
            "\n",
            "Reading ERR2696825.fastq\n",
            "Saved 6182 reads\n",
            "12170 reads\n",
            "\n",
            "\n",
            "Reading ERR2033212.fastq\n",
            "Saved 1961 reads\n",
            "25538 reads\n",
            "\n",
            "\n",
            "Reading ERR5329115.fastq\n",
            "Saved 12172 reads\n",
            "31256 reads\n",
            "\n",
            "\n",
            "Reading ERR2032698.fastq\n",
            "Saved 6434 reads\n",
            "21901 reads\n",
            "\n",
            "\n",
            "Reading ERR2319259.fastq\n",
            "Saved 3933 reads\n",
            "18660 reads\n",
            "\n",
            "\n",
            "Reading ERR2032546.fastq\n",
            "Saved 5156 reads\n",
            "16103 reads\n",
            "\n",
            "\n",
            "Reading ERR2303848.fastq\n",
            "Saved 14288 reads\n",
            "30646 reads\n",
            "\n",
            "\n",
            "Reading ERR2319477.fastq\n",
            "Saved 10262 reads\n",
            "35379 reads\n",
            "\n",
            "\n",
            "Reading ERR5332209.fastq\n",
            "Saved 8236 reads\n",
            "18654 reads\n",
            "\n",
            "\n",
            "Reading ERR2032547.fastq\n",
            "Saved 4172 reads\n",
            "10616 reads\n",
            "\n",
            "\n",
            "Reading ERR5326334.fastq\n",
            "Saved 8273 reads\n",
            "18749 reads\n",
            "\n",
            "\n",
            "Reading ERR2314119.fastq\n",
            "Saved 22644 reads\n",
            "37817 reads\n",
            "\n",
            "\n",
            "Reading ERR2306103.fastq\n",
            "Saved 3102 reads\n",
            "78069 reads\n",
            "\n",
            "\n",
            "Reading ERR1842761.fastq\n",
            "Saved 46 reads\n",
            "33037 reads\n",
            "\n",
            "\n",
            "Reading ERR4020425.fastq\n",
            "Saved 7242 reads\n",
            "12946 reads\n",
            "\n",
            "\n",
            "Reading ERR4020435.fastq\n",
            "Saved 12935 reads\n",
            "26478 reads\n",
            "\n",
            "\n",
            "Reading ERR2318185.fastq\n",
            "Saved 10528 reads\n",
            "46640 reads\n",
            "\n",
            "\n",
            "Reading ERR4019687.fastq\n",
            "Saved 18676 reads\n",
            "42714 reads\n",
            "\n",
            "\n",
            "Reading ERR2314106.fastq\n",
            "Saved 20267 reads\n",
            "33833 reads\n",
            "\n",
            "\n",
            "Reading ERR2319256.fastq\n",
            "Saved 9993 reads\n",
            "28192 reads\n",
            "\n",
            "\n",
            "Reading ERR2032754.fastq\n",
            "Saved 4182 reads\n",
            "13714 reads\n",
            "\n",
            "\n",
            "Reading ERR2056970.fastq\n",
            "Saved 10271 reads\n",
            "27824 reads\n",
            "\n",
            "\n",
            "Reading ERR5325637.fastq\n",
            "Saved 13609 reads\n",
            "23733 reads\n",
            "\n",
            "\n",
            "Reading ERR5337729.fastq\n",
            "Saved 19598 reads\n",
            "41269 reads\n",
            "\n",
            "\n",
            "Reading ERR1854691.fastq\n",
            "Saved 1187 reads\n",
            "21775 reads\n",
            "\n",
            "\n",
            "Reading ERR2092195.fastq\n",
            "Saved 5801 reads\n",
            "27911 reads\n",
            "\n",
            "\n",
            "Reading ERR2092140.fastq\n",
            "Saved 3107 reads\n",
            "16184 reads\n",
            "\n",
            "\n",
            "Reading ERR2319272.fastq\n",
            "Saved 11923 reads\n",
            "41132 reads\n",
            "\n",
            "\n",
            "Reading ERR5326317.fastq\n",
            "Saved 10264 reads\n",
            "21842 reads\n",
            "\n",
            "\n",
            "Reading ERR2404942.fastq\n",
            "Saved 8166 reads\n",
            "16994 reads\n",
            "\n",
            "\n",
            "Reading ERR2579690.fastq\n",
            "Saved 17024 reads\n",
            "69941 reads\n",
            "\n",
            "\n",
            "Reading ERR2319478.fastq\n",
            "Saved 13215 reads\n",
            "54801 reads\n",
            "\n",
            "\n",
            "Reading ERR5325493.fastq\n",
            "Saved 11516 reads\n",
            "20233 reads\n",
            "\n",
            "\n",
            "Reading ERR5332367.fastq\n",
            "Saved 10232 reads\n",
            "25177 reads\n",
            "\n",
            "\n",
            "Reading ERR2319306.fastq\n",
            "Saved 6201 reads\n",
            "39414 reads\n",
            "\n",
            "\n",
            "Reading ERR5325495.fastq\n",
            "Saved 13364 reads\n",
            "28579 reads\n",
            "\n",
            "\n",
            "Reading ERR2697554.fastq\n",
            "Saved 9339 reads\n",
            "23316 reads\n",
            "\n",
            "\n",
            "Reading ERR2032551.fastq\n",
            "Saved 9459 reads\n",
            "32990 reads\n",
            "\n",
            "\n",
            "Reading ERR2033221.fastq\n",
            "Saved 2109 reads\n",
            "22634 reads\n",
            "\n",
            "\n",
            "Reading ERR5326483.fastq\n",
            "Saved 10173 reads\n",
            "22192 reads\n",
            "\n",
            "\n",
            "Reading ERR5333108.fastq\n",
            "Saved 10056 reads\n",
            "30634 reads\n",
            "\n",
            "\n",
            "Reading ERR2032804.fastq\n",
            "Saved 6763 reads\n",
            "24596 reads\n",
            "\n",
            "\n",
            "Reading ERR2309175.fastq\n",
            "Saved 12799 reads\n",
            "45754 reads\n",
            "\n",
            "\n",
            "Reading ERR5328192.fastq\n",
            "Saved 12443 reads\n",
            "41110 reads\n",
            "\n",
            "\n",
            "Reading ERR2319269.fastq\n",
            "Saved 17666 reads\n",
            "56223 reads\n",
            "\n",
            "\n",
            "Reading ERR2092194.fastq\n",
            "Saved 5215 reads\n",
            "24008 reads\n",
            "\n",
            "\n",
            "Reading ERR1854673.fastq\n",
            "Saved 1094 reads\n",
            "19092 reads\n",
            "\n",
            "\n",
            "Reading ERR2313962.fastq\n",
            "Saved 33274 reads\n",
            "53069 reads\n",
            "\n",
            "\n",
            "Reading ERR2032501.fastq\n",
            "Saved 5371 reads\n",
            "19562 reads\n",
            "\n",
            "\n",
            "Reading ERR5326333.fastq\n",
            "Saved 8366 reads\n",
            "20575 reads\n",
            "\n",
            "\n",
            "Reading ERR1842764.fastq\n",
            "Saved 127 reads\n",
            "30837 reads\n",
            "\n",
            "\n",
            "Reading ERR1842698.fastq\n",
            "Saved 7 reads\n",
            "7525 reads\n",
            "\n",
            "\n",
            "Reading ERR1842797.fastq\n",
            "Saved 23 reads\n",
            "28329 reads\n",
            "\n",
            "\n",
            "Reading ERR2032597.fastq\n",
            "Saved 6737 reads\n",
            "24395 reads\n",
            "\n",
            "\n",
            "Reading ERR2032587.fastq\n",
            "Saved 6868 reads\n",
            "20950 reads\n",
            "\n",
            "\n",
            "Reading ERR5332725.fastq\n",
            "Saved 8818 reads\n",
            "19828 reads\n",
            "\n",
            "\n",
            "Reading ERR2306094.fastq\n",
            "Saved 7145 reads\n",
            "87796 reads\n",
            "\n",
            "\n",
            "Reading ERR1854671.fastq\n",
            "Saved 1460 reads\n",
            "28326 reads\n",
            "\n",
            "\n",
            "Reading ERR5337741.fastq\n",
            "Saved 17214 reads\n",
            "41157 reads\n",
            "\n",
            "\n",
            "Reading ERR2696671.fastq\n",
            "Saved 3381 reads\n",
            "20838 reads\n",
            "\n",
            "\n",
            "Reading ERR5326146.fastq\n",
            "Saved 12233 reads\n",
            "25369 reads\n",
            "\n",
            "\n",
            "Reading ERR5327712.fastq\n",
            "Saved 1356 reads\n",
            "18731 reads\n",
            "\n",
            "\n",
            "Reading ERR2032755.fastq\n",
            "Saved 7651 reads\n",
            "23632 reads\n",
            "\n",
            "\n",
            "Reading ERR5327718.fastq\n",
            "Saved 2908 reads\n",
            "27629 reads\n",
            "\n",
            "\n",
            "Reading ERR2032581.fastq\n",
            "Saved 10276 reads\n",
            "33623 reads\n",
            "\n",
            "\n",
            "Reading ERR5326131.fastq\n",
            "Saved 8256 reads\n",
            "19138 reads\n",
            "\n",
            "\n",
            "Reading ERR2696489.fastq\n",
            "Saved 4427 reads\n",
            "39151 reads\n",
            "\n",
            "\n",
            "Reading ERR2032471.fastq\n",
            "Saved 13707 reads\n",
            "37468 reads\n",
            "\n",
            "\n",
            "Reading ERR2579868.fastq\n",
            "Saved 332 reads\n",
            "3162 reads\n",
            "\n",
            "\n",
            "Reading ERR5328962.fastq\n",
            "Saved 12007 reads\n",
            "30935 reads\n",
            "\n",
            "\n",
            "Reading ERR2032528.fastq\n",
            "Saved 7614 reads\n",
            "25782 reads\n",
            "\n",
            "\n",
            "Reading ERR2319276.fastq\n",
            "Saved 9441 reads\n",
            "28642 reads\n",
            "\n",
            "\n",
            "Reading ERR1842728.fastq\n",
            "Saved 54 reads\n",
            "31170 reads\n",
            "\n",
            "\n",
            "Reading ERR2697863.fastq\n",
            "Saved 9053 reads\n",
            "21535 reads\n",
            "\n",
            "\n",
            "Reading ERR1842691.fastq\n",
            "Saved 46 reads\n",
            "29286 reads\n",
            "\n",
            "\n",
            "Reading ERR1842708.fastq\n",
            "Saved 52 reads\n",
            "18937 reads\n",
            "\n",
            "\n",
            "Reading ERR1842722.fastq\n",
            "Saved 29 reads\n",
            "40861 reads\n",
            "\n",
            "\n",
            "Reading ERR2032797.fastq\n",
            "Saved 16262 reads\n",
            "48855 reads\n",
            "\n",
            "\n",
            "Reading ERR2056772.fastq\n",
            "Saved 5675 reads\n",
            "20489 reads\n",
            "\n",
            "\n",
            "Reading ERR1842726.fastq\n",
            "Saved 8 reads\n",
            "12478 reads\n",
            "\n",
            "\n",
            "Reading ERR5328033.fastq\n",
            "Saved 19831 reads\n",
            "44168 reads\n",
            "\n",
            "\n",
            "Reading ERR2304080.fastq\n",
            "Saved 15004 reads\n",
            "29432 reads\n",
            "\n",
            "\n",
            "Reading ERR2309377.fastq\n",
            "Saved 21387 reads\n",
            "75349 reads\n",
            "\n",
            "\n",
            "Reading ERR5332667.fastq\n",
            "Saved 4439 reads\n",
            "13714 reads\n",
            "\n",
            "\n",
            "Reading ERR2319303.fastq\n",
            "Saved 15799 reads\n",
            "48180 reads\n",
            "\n",
            "\n",
            "Reading ERR5327928.fastq\n",
            "Saved 1972 reads\n",
            "21681 reads\n",
            "\n",
            "\n",
            "Reading ERR2579788.fastq\n",
            "Saved 3676 reads\n",
            "12129 reads\n",
            "\n",
            "\n",
            "Reading ERR5329067.fastq\n",
            "Saved 8142 reads\n",
            "21709 reads\n",
            "\n",
            "\n",
            "Reading ERR2318082.fastq\n",
            "Saved 5452 reads\n",
            "34048 reads\n",
            "\n",
            "\n",
            "Reading ERR5332668.fastq\n",
            "Saved 4962 reads\n",
            "14608 reads\n",
            "\n",
            "\n",
            "Reading ERR5342276.fastq\n",
            "Saved 26886 reads\n",
            "49848 reads\n",
            "\n",
            "\n",
            "Reading ERR2033290.fastq\n",
            "Saved 1134 reads\n",
            "18027 reads\n",
            "\n",
            "\n",
            "Reading ERR2309320.fastq\n",
            "Saved 15555 reads\n",
            "54341 reads\n",
            "\n",
            "\n",
            "Reading ERR4019004.fastq\n",
            "Saved 2186 reads\n",
            "25164 reads\n",
            "\n",
            "\n",
            "Reading ERR4020455.fastq\n",
            "Saved 9340 reads\n",
            "17643 reads\n",
            "\n",
            "\n",
            "Reading ERR1842701.fastq\n",
            "Saved 53 reads\n",
            "34735 reads\n",
            "\n",
            "\n",
            "Reading ERR2032461.fastq\n",
            "Saved 11144 reads\n",
            "36116 reads\n",
            "\n",
            "\n",
            "Reading ERR2092193.fastq\n",
            "Saved 5911 reads\n",
            "26561 reads\n",
            "\n",
            "\n",
            "Reading ERR2032520.fastq\n",
            "Saved 8200 reads\n",
            "29033 reads\n",
            "\n",
            "\n",
            "Reading ERR4020444.fastq\n",
            "Saved 10663 reads\n",
            "23110 reads\n",
            "\n",
            "\n",
            "Reading ERR2319231.fastq\n",
            "Saved 8945 reads\n",
            "31174 reads\n",
            "\n",
            "\n",
            "Reading ERR2318011.fastq\n",
            "Saved 12320 reads\n",
            "48653 reads\n",
            "\n",
            "\n",
            "Reading ERR2032542.fastq\n",
            "Saved 3861 reads\n",
            "13493 reads\n",
            "\n",
            "\n",
            "Reading ERR2032529.fastq\n",
            "Saved 12795 reads\n",
            "41382 reads\n",
            "\n",
            "\n",
            "Reading ERR2319290.fastq\n",
            "Saved 12335 reads\n",
            "43957 reads\n",
            "\n",
            "\n",
            "Reading ERR4019709.fastq\n",
            "Saved 16494 reads\n",
            "31564 reads\n",
            "\n",
            "\n",
            "Reading ERR2697556.fastq\n",
            "Saved 8119 reads\n",
            "18286 reads\n",
            "\n",
            "\n",
            "Reading ERR5332811.fastq\n",
            "Saved 10917 reads\n",
            "33533 reads\n",
            "\n",
            "\n",
            "Reading ERR2092192.fastq\n",
            "Saved 5983 reads\n",
            "26891 reads\n",
            "\n",
            "\n",
            "Reading ERR5325492.fastq\n",
            "Saved 10729 reads\n",
            "19774 reads\n",
            "\n",
            "\n",
            "Reading ERR5332030.fastq\n",
            "Saved 8303 reads\n",
            "21946 reads\n",
            "\n",
            "\n",
            "Reading ERR1842684.fastq\n",
            "Saved 42 reads\n",
            "16396 reads\n",
            "\n",
            "\n",
            "Reading ERR4018899.fastq\n",
            "Saved 7382 reads\n",
            "65085 reads\n",
            "\n",
            "\n",
            "Reading ERR2523875.fastq\n",
            "Saved 21223 reads\n",
            "40808 reads\n",
            "\n",
            "\n",
            "Reading ERR2313972.fastq\n",
            "Saved 21836 reads\n",
            "34663 reads\n",
            "\n",
            "\n",
            "Reading ERR2056771.fastq\n",
            "Saved 7864 reads\n",
            "27322 reads\n",
            "\n",
            "\n",
            "Reading ERR2309187.fastq\n",
            "Saved 11914 reads\n",
            "41910 reads\n",
            "\n",
            "\n",
            "Reading ERR1854688.fastq\n",
            "Saved 1882 reads\n",
            "24899 reads\n",
            "\n",
            "\n",
            "Reading ERR5332028.fastq\n",
            "Saved 11333 reads\n",
            "26126 reads\n",
            "\n",
            "\n",
            "Reading ERR4019057.fastq\n",
            "Saved 3193 reads\n",
            "34078 reads\n",
            "\n",
            "\n",
            "Reading ERR2319236.fastq\n",
            "Saved 10745 reads\n",
            "37841 reads\n",
            "\n",
            "\n",
            "Reading ERR1842897.fastq\n",
            "Saved 85 reads\n",
            "28549 reads\n",
            "\n",
            "\n",
            "Reading ERR2032533.fastq\n",
            "Saved 10344 reads\n",
            "33971 reads\n",
            "\n",
            "\n",
            "Reading ERR2579861.fastq\n",
            "Saved 13276 reads\n",
            "58736 reads\n",
            "\n",
            "\n",
            "Reading ERR2303816.fastq\n",
            "Saved 15618 reads\n",
            "29912 reads\n",
            "\n",
            "\n",
            "Reading ERR4020390.fastq\n",
            "Saved 9086 reads\n",
            "18243 reads\n",
            "\n",
            "\n",
            "Reading ERR2032466.fastq\n",
            "Saved 2 reads\n",
            "5 reads\n",
            "\n",
            "\n",
            "Reading ERR2304017.fastq\n",
            "Saved 9419 reads\n",
            "19620 reads\n",
            "\n",
            "\n",
            "Reading ERR2523926.fastq\n",
            "Saved 9136 reads\n",
            "19120 reads\n",
            "\n",
            "\n",
            "Reading ERR5326262.fastq\n",
            "Saved 4224 reads\n",
            "10300 reads\n",
            "\n",
            "\n",
            "Reading ERR4019262.fastq\n",
            "Saved 7061 reads\n",
            "16699 reads\n",
            "\n",
            "\n",
            "Reading ERR2313963.fastq\n",
            "Saved 16933 reads\n",
            "27488 reads\n",
            "\n",
            "\n",
            "Reading ERR2032578.fastq\n",
            "Saved 8537 reads\n",
            "25052 reads\n",
            "\n",
            "\n",
            "Reading ERR2033468.fastq\n",
            "Saved 1912 reads\n",
            "25167 reads\n",
            "\n",
            "\n",
            "Reading ERR2032704.fastq\n",
            "Saved 9184 reads\n",
            "29449 reads\n",
            "\n",
            "\n",
            "Reading ERR5329027.fastq\n",
            "Saved 9597 reads\n",
            "23250 reads\n",
            "\n",
            "\n",
            "Reading ERR1842706.fastq\n",
            "Saved 68 reads\n",
            "30148 reads\n",
            "\n",
            "\n",
            "Reading ERR2056767.fastq\n",
            "Saved 7888 reads\n",
            "19066 reads\n",
            "\n",
            "\n",
            "Reading ERR2319277.fastq\n",
            "Saved 4 reads\n",
            "34 reads\n",
            "\n",
            "\n",
            "Reading ERR4018883.fastq\n",
            "Saved 5774 reads\n",
            "39345 reads\n",
            "\n",
            "\n",
            "Reading ERR1854672.fastq\n",
            "Saved 1859 reads\n",
            "21955 reads\n",
            "\n",
            "\n",
            "Reading ERR2318150.fastq\n",
            "Saved 1 reads\n",
            "13 reads\n",
            "\n",
            "\n",
            "Reading ERR5332123.fastq\n",
            "Saved 7932 reads\n",
            "17584 reads\n",
            "\n",
            "\n",
            "Reading ERR2696597.fastq\n",
            "Saved 7157 reads\n",
            "57555 reads\n",
            "\n",
            "\n",
            "Reading ERR2032550.fastq\n",
            "Saved 14169 reads\n",
            "48887 reads\n",
            "\n",
            "\n",
            "Reading ERR1842732.fastq\n",
            "Saved 53 reads\n",
            "22844 reads\n",
            "\n",
            "\n",
            "Reading ERR2697816.fastq\n",
            "Saved 9325 reads\n",
            "19136 reads\n",
            "\n",
            "\n",
            "Reading ERR2032570.fastq\n",
            "Saved 10993 reads\n",
            "39948 reads\n",
            "\n",
            "\n",
            "Reading ERR2696832.fastq\n",
            "Saved 4316 reads\n",
            "9065 reads\n",
            "\n",
            "\n",
            "Reading ERR2697056.fastq\n",
            "Saved 18978 reads\n",
            "38000 reads\n",
            "\n",
            "\n",
            "Reading ERR2032495.fastq\n",
            "Saved 4813 reads\n",
            "16572 reads\n",
            "\n",
            "\n",
            "Reading ERR2056773.fastq\n",
            "Saved 9018 reads\n",
            "31481 reads\n",
            "\n",
            "\n",
            "Reading ERR2697711.fastq\n",
            "Saved 12265 reads\n",
            "29816 reads\n",
            "\n",
            "\n",
            "Reading ERR5328992.fastq\n",
            "Saved 9449 reads\n",
            "23346 reads\n",
            "\n",
            "\n",
            "Reading ERR2032574.fastq\n",
            "Saved 17118 reads\n",
            "55275 reads\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "names, tot_reads, good_reads = filter_quality(\"fastq\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 61,
      "metadata": {},
      "outputs": [],
      "source": [
        "good_reads_df = pd.DataFrame({\"id\":names,\"tot_reads\":tot_reads,\"good_reads\":good_reads})"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 62,
      "metadata": {},
      "outputs": [],
      "source": [
        "good_reads_df.to_csv('good_reads.csv')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [],
      "source": [
        "good_reads_df = pd.read_csv(\"good_reads.csv\",index_col=0)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>id</th>\n",
              "      <th>tot_reads</th>\n",
              "      <th>good_reads</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>204</th>\n",
              "      <td>ERR2318150.fastq</td>\n",
              "      <td>13</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>189</th>\n",
              "      <td>ERR2032466.fastq</td>\n",
              "      <td>5</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>201</th>\n",
              "      <td>ERR2319277.fastq</td>\n",
              "      <td>34</td>\n",
              "      <td>4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>114</th>\n",
              "      <td>ERR1842698.fastq</td>\n",
              "      <td>7525</td>\n",
              "      <td>7</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>142</th>\n",
              "      <td>ERR1842726.fastq</td>\n",
              "      <td>12478</td>\n",
              "      <td>8</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>ERR2318152.fastq</td>\n",
              "      <td>57</td>\n",
              "      <td>20</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>ERR1842731.fastq</td>\n",
              "      <td>18850</td>\n",
              "      <td>22</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>115</th>\n",
              "      <td>ERR1842797.fastq</td>\n",
              "      <td>28329</td>\n",
              "      <td>23</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>ERR1842877.fastq</td>\n",
              "      <td>17826</td>\n",
              "      <td>24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>139</th>\n",
              "      <td>ERR1842722.fastq</td>\n",
              "      <td>40861</td>\n",
              "      <td>29</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>174</th>\n",
              "      <td>ERR1842684.fastq</td>\n",
              "      <td>16396</td>\n",
              "      <td>42</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>137</th>\n",
              "      <td>ERR1842691.fastq</td>\n",
              "      <td>29286</td>\n",
              "      <td>46</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>76</th>\n",
              "      <td>ERR1842761.fastq</td>\n",
              "      <td>33037</td>\n",
              "      <td>46</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>138</th>\n",
              "      <td>ERR1842708.fastq</td>\n",
              "      <td>18937</td>\n",
              "      <td>52</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>208</th>\n",
              "      <td>ERR1842732.fastq</td>\n",
              "      <td>22844</td>\n",
              "      <td>53</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                   id  tot_reads  good_reads\n",
              "204  ERR2318150.fastq         13           1\n",
              "189  ERR2032466.fastq          5           2\n",
              "201  ERR2319277.fastq         34           4\n",
              "114  ERR1842698.fastq       7525           7\n",
              "142  ERR1842726.fastq      12478           8\n",
              "34   ERR2318152.fastq         57          20\n",
              "13   ERR1842731.fastq      18850          22\n",
              "115  ERR1842797.fastq      28329          23\n",
              "37   ERR1842877.fastq      17826          24\n",
              "139  ERR1842722.fastq      40861          29\n",
              "174  ERR1842684.fastq      16396          42\n",
              "137  ERR1842691.fastq      29286          46\n",
              "76   ERR1842761.fastq      33037          46\n",
              "138  ERR1842708.fastq      18937          52\n",
              "208  ERR1842732.fastq      22844          53"
            ]
          },
          "execution_count": 12,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "good_reads_df.sort_values(by=\"good_reads\").head(15)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.8.15"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
