{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "from qiime2 import Artifact\n",
    "import os"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import of the final sample dataset obtained in the EDA notebook: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>acid_reflux</th>\n",
       "      <th>acne_medication</th>\n",
       "      <th>acne_medication_otc</th>\n",
       "      <th>add_adhd</th>\n",
       "      <th>age_cat</th>\n",
       "      <th>age_corrected</th>\n",
       "      <th>age_years</th>\n",
       "      <th>alcohol_consumption</th>\n",
       "      <th>alcohol_frequency</th>\n",
       "      <th>...</th>\n",
       "      <th>vitamin_b_supplement_frequency</th>\n",
       "      <th>vitamin_d_supplement_frequency</th>\n",
       "      <th>vivid_dreams</th>\n",
       "      <th>weight_cat</th>\n",
       "      <th>weight_change</th>\n",
       "      <th>weight_kg</th>\n",
       "      <th>weight_units</th>\n",
       "      <th>whole_eggs</th>\n",
       "      <th>whole_grain_frequency</th>\n",
       "      <th>group</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10317.000023139</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>60s</td>\n",
       "      <td>69.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>...</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Daily</td>\n",
       "      <td>Never</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>63</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10317.000027811</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>40s</td>\n",
       "      <td>46.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>...</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>81</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10317.000031513</td>\n",
       "      <td>Self-diagnosed</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>30s</td>\n",
       "      <td>37.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>49</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10317.000033280</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>60s</td>\n",
       "      <td>60.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>68</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10317.000038261</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>40s</td>\n",
       "      <td>48.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>70</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>10317.000065667</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>20s</td>\n",
       "      <td>22.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>61</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Never</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>10317.000065673</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>20s</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>No</td>\n",
       "      <td>Never</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>79</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Never</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>10317.000065675</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>20s</td>\n",
       "      <td>21.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>No</td>\n",
       "      <td>Never</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Daily</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>51</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Never</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>10317.000065681</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>20s</td>\n",
       "      <td>21.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>No</td>\n",
       "      <td>Never</td>\n",
       "      <td>...</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>137</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>10317.000065684</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>20s</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>No</td>\n",
       "      <td>Never</td>\n",
       "      <td>...</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>Occasionally (1-2 times/week)</td>\n",
       "      <td>Rarely (a few times/month)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Remained stable</td>\n",
       "      <td>58</td>\n",
       "      <td>kilograms</td>\n",
       "      <td>Rarely (less than once/week)</td>\n",
       "      <td>Regularly (3-5 times/week)</td>\n",
       "      <td>Control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>380 rows × 661 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name     acid_reflux acne_medication acne_medication_otc  \\\n",
       "0    10317.000023139              No              No                  No   \n",
       "1    10317.000027811              No              No                  No   \n",
       "2    10317.000031513  Self-diagnosed              No                  No   \n",
       "3    10317.000033280              No              No                  No   \n",
       "4    10317.000038261              No              No                  No   \n",
       "..               ...             ...             ...                 ...   \n",
       "375  10317.000065667              No              No                  No   \n",
       "376  10317.000065673              No              No                 Yes   \n",
       "377  10317.000065675              No              No                  No   \n",
       "378  10317.000065681              No              No                 Yes   \n",
       "379  10317.000065684              No              No                  No   \n",
       "\n",
       "         add_adhd age_cat  age_corrected  age_years alcohol_consumption  \\\n",
       "0              No     60s           69.0       69.0                 Yes   \n",
       "1              No     40s           46.0       46.0                 Yes   \n",
       "2    Not provided     30s           37.0       37.0                 Yes   \n",
       "3              No     60s           60.0       60.0                 Yes   \n",
       "4              No     40s           48.0       48.0                 Yes   \n",
       "..            ...     ...            ...        ...                 ...   \n",
       "375            No     20s           22.0       22.0                 Yes   \n",
       "376            No     20s           20.0       20.0                  No   \n",
       "377            No     20s           21.0       21.0                  No   \n",
       "378            No     20s           21.0       21.0                  No   \n",
       "379            No     20s           20.0       20.0                  No   \n",
       "\n",
       "                 alcohol_frequency  ... vitamin_b_supplement_frequency  \\\n",
       "0       Rarely (a few times/month)  ...     Regularly (3-5 times/week)   \n",
       "1       Rarely (a few times/month)  ...     Rarely (a few times/month)   \n",
       "2       Regularly (3-5 times/week)  ...                          Never   \n",
       "3       Regularly (3-5 times/week)  ...                          Never   \n",
       "4       Regularly (3-5 times/week)  ...                          Never   \n",
       "..                             ...  ...                            ...   \n",
       "375  Occasionally (1-2 times/week)  ...                          Never   \n",
       "376                          Never  ...                          Never   \n",
       "377                          Never  ...                          Never   \n",
       "378                          Never  ...                          Never   \n",
       "379                          Never  ...     Rarely (a few times/month)   \n",
       "\n",
       "    vitamin_d_supplement_frequency                   vivid_dreams weight_cat  \\\n",
       "0                            Daily                          Never        NaN   \n",
       "1       Regularly (3-5 times/week)     Rarely (a few times/month)        NaN   \n",
       "2       Rarely (a few times/month)     Rarely (a few times/month)        NaN   \n",
       "3                            Never                          Never        NaN   \n",
       "4                            Never  Occasionally (1-2 times/week)        NaN   \n",
       "..                             ...                            ...        ...   \n",
       "375                          Never     Rarely (a few times/month)        NaN   \n",
       "376                          Never  Occasionally (1-2 times/week)        NaN   \n",
       "377                          Never                          Daily        NaN   \n",
       "378                          Never     Rarely (a few times/month)        NaN   \n",
       "379  Occasionally (1-2 times/week)     Rarely (a few times/month)        NaN   \n",
       "\n",
       "       weight_change weight_kg weight_units                     whole_eggs  \\\n",
       "0    Remained stable        63    kilograms     Regularly (3-5 times/week)   \n",
       "1    Remained stable        81    kilograms  Occasionally (1-2 times/week)   \n",
       "2    Remained stable        49    kilograms  Occasionally (1-2 times/week)   \n",
       "3    Remained stable        68    kilograms  Occasionally (1-2 times/week)   \n",
       "4    Remained stable        70    kilograms   Rarely (less than once/week)   \n",
       "..               ...       ...          ...                            ...   \n",
       "375  Remained stable        61    kilograms     Regularly (3-5 times/week)   \n",
       "376  Remained stable        79    kilograms                          Never   \n",
       "377  Remained stable        51    kilograms                          Never   \n",
       "378  Remained stable       137    kilograms   Rarely (less than once/week)   \n",
       "379  Remained stable        58    kilograms   Rarely (less than once/week)   \n",
       "\n",
       "             whole_grain_frequency      group  \n",
       "0       Regularly (3-5 times/week)  Treatment  \n",
       "1     Rarely (less than once/week)  Treatment  \n",
       "2     Rarely (less than once/week)  Treatment  \n",
       "3       Regularly (3-5 times/week)  Treatment  \n",
       "4    Occasionally (1-2 times/week)  Treatment  \n",
       "..                             ...        ...  \n",
       "375                          Never    Control  \n",
       "376   Rarely (less than once/week)    Control  \n",
       "377     Regularly (3-5 times/week)    Control  \n",
       "378   Rarely (less than once/week)    Control  \n",
       "379     Regularly (3-5 times/week)    Control  \n",
       "\n",
       "[380 rows x 661 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_csv('data_final_groups.csv',dtype = {\"sample_name\": \"str\"},index_col=0)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Yes    190\n",
       "No     190\n",
       "Name: specialized_diet_exclude_refined_sugars, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.specialized_diet_exclude_refined_sugars.value_counts()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Metadata file creation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = data.sample_name.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = metadata.rename(columns={'sample_name': 'sample-id'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_cat</th>\n",
       "      <th>refined_sugars</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10317.000023139</td>\n",
       "      <td>female</td>\n",
       "      <td>60s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10317.000027811</td>\n",
       "      <td>male</td>\n",
       "      <td>40s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10317.000031513</td>\n",
       "      <td>female</td>\n",
       "      <td>30s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10317.000033280</td>\n",
       "      <td>female</td>\n",
       "      <td>60s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10317.000038261</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>10317.000065667</td>\n",
       "      <td>female</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>10317.000065673</td>\n",
       "      <td>female</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>10317.000065675</td>\n",
       "      <td>female</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>10317.000065681</td>\n",
       "      <td>female</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>10317.000065684</td>\n",
       "      <td>female</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>380 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           sample-id     sex age_cat refined_sugars\n",
       "0    10317.000023139  female     60s            Yes\n",
       "1    10317.000027811    male     40s            Yes\n",
       "2    10317.000031513  female     30s            Yes\n",
       "3    10317.000033280  female     60s            Yes\n",
       "4    10317.000038261  female     40s            Yes\n",
       "..               ...     ...     ...            ...\n",
       "375  10317.000065667  female     20s             No\n",
       "376  10317.000065673  female     20s             No\n",
       "377  10317.000065675  female     20s             No\n",
       "378  10317.000065681  female     20s             No\n",
       "379  10317.000065684  female     20s             No\n",
       "\n",
       "[380 rows x 4 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Aggiungiamo al manifest le informazioni riguardo l'età, il sesso e l'uso degli antibiotici\n",
    "metadata['sex'] = data['sex']\n",
    "metadata['age_cat'] = data['age_cat']\n",
    "metadata['refined_sugars'] = data['specialized_diet_exclude_refined_sugars']\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata['sample-id'].to_csv(\"ids.txt\",header=None,index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Yes    190\n",
       "No     190\n",
       "Name: refined_sugars, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.refined_sugars.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata.to_csv('sample_metadata.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from qiime2 import Metadata\n",
    "sample_metadata = Metadata.load(os.getcwd()+'/sample_metadata.tsv')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import Fasta 99% otus containing Taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported gg_13_8_otus/rep_set/99_otus.fasta as DNASequencesDirectoryFormat to rep_set.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime tools import --type 'FeatureData[Sequence]' --input-path gg_13_8_otus/rep_set/99_otus.fasta \\\n",
    "    --output-path rep_set.qza"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Getting otus from sample ids with biom:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "!export ctx=Pick_closed-reference_OTUs-Greengenes-Illumina-16S-V4-5c6506"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!redbiom fetch samples --from ids.txt --context $ctx --output otu_table.biom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "!biom convert -i otu_table.biom -o otu_table.tsv --to-tsv "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_table = pd.read_csv(\"sample_metadata.tsv\",sep=\"\\t\",dtype = {\"sample-id\": \"str\"})\n",
    "meta_table = meta_table.sort_values(by=\"sample-id\").reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported otu_table.biom as BIOMV210Format to feature_table.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path otu_table.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --input-format BIOMV210Format \\\n",
    "  --output-path feature_table.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "from qiime2 import Artifact\n",
    "otu_table = Artifact.load(\"feature_table.qza\")\n",
    "rep_set = Artifact.load(\"rep_set.qza\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "otu_index = otu_table.view(pd.DataFrame).index.sort_values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "ot_df = otu_table.view(pd.DataFrame) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "non_match = []\n",
    "for i in range(meta_table.shape[0]):\n",
    "    result = [k for k in otu_index if k.startswith(meta_table[\"sample-id\"][i])]\n",
    "    if len(result) == 0:\n",
    "        non_match.append(meta_table[\"sample-id\"][i] )\n",
    "    else: \n",
    "        meta_table[\"sample-id\"][i] = meta_table[\"sample-id\"][i] + result[0][-6:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_table_filt = meta_table.loc[~meta_table[\"sample-id\"].isin(non_match)] \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_cat</th>\n",
       "      <th>refined_sugars</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10317.000001166.25853</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10317.000003100.69016</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10317.000004936.25853</td>\n",
       "      <td>male</td>\n",
       "      <td>50s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10317.000010862.25853</td>\n",
       "      <td>male</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10317.000012018.56079</td>\n",
       "      <td>female</td>\n",
       "      <td>50s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>10317.000109791.74729</td>\n",
       "      <td>male</td>\n",
       "      <td>20s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>10317.000109807.76952</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>10317.000110177.74729</td>\n",
       "      <td>male</td>\n",
       "      <td>30s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>376</th>\n",
       "      <td>10317.000112912.76952</td>\n",
       "      <td>female</td>\n",
       "      <td>50s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>10317.000113026.81168</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>359 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample-id     sex age_cat refined_sugars\n",
       "0    10317.000001166.25853  female     40s             No\n",
       "1    10317.000003100.69016  female     40s             No\n",
       "2    10317.000004936.25853    male     50s             No\n",
       "3    10317.000010862.25853    male     20s             No\n",
       "4    10317.000012018.56079  female     50s             No\n",
       "..                     ...     ...     ...            ...\n",
       "373  10317.000109791.74729    male     20s            Yes\n",
       "374  10317.000109807.76952  female     40s            Yes\n",
       "375  10317.000110177.74729    male     30s            Yes\n",
       "376  10317.000112912.76952  female     50s            Yes\n",
       "377  10317.000113026.81168  female     40s            Yes\n",
       "\n",
       "[359 rows x 4 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_table_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_table_filt.to_csv(\"sample_metadata_filt.tsv\",sep=\"\\t\",index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "ot_df_filt = ot_df.loc[ot_df.index.isin(meta_table_filt[\"sample-id\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "ot_df_filt.to_csv(\"otu_table_filt.tsv\",sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "!biom convert -i otu_table_filt.tsv -o otu_table_filt.biom --table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Sequence] to: rep_set_filt.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-seqs --i-data rep_set.qza --m-metadata-file otu_table.tsv --o-filtered-data rep_set_filt.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "!biom convert -i otu_table_reorder.tsv -o otu_table_reorder.biom --table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "otu_df = otu_table.view(pd.DataFrame)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OTU filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>383659</th>\n",
       "      <th>227697</th>\n",
       "      <th>318777</th>\n",
       "      <th>193551</th>\n",
       "      <th>297629</th>\n",
       "      <th>4359590</th>\n",
       "      <th>356052</th>\n",
       "      <th>3392847</th>\n",
       "      <th>263834</th>\n",
       "      <th>294909</th>\n",
       "      <th>...</th>\n",
       "      <th>4344207</th>\n",
       "      <th>3251419</th>\n",
       "      <th>4414478</th>\n",
       "      <th>350503</th>\n",
       "      <th>165257</th>\n",
       "      <th>512791</th>\n",
       "      <th>1718658</th>\n",
       "      <th>577679</th>\n",
       "      <th>3568710</th>\n",
       "      <th>178957</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10317.000023139.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000027811.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000031513.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000033280.31503</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000038261.33800</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000065667.26478</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000065673.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000065675.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000065681.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000065684.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1191.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>371 rows × 6904 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       383659  227697  318777  193551  297629  4359590  \\\n",
       "10317.000023139.25853     0.0     0.0     0.0     1.0     0.0      4.0   \n",
       "10317.000027811.25853     0.0     0.0     0.0     3.0     0.0      2.0   \n",
       "10317.000031513.25853     0.0     0.0     0.0     0.0     0.0      0.0   \n",
       "10317.000033280.31503     0.0     0.0     0.0     1.0     0.0      2.0   \n",
       "10317.000038261.33800     0.0     0.0     0.0     0.0     0.0      1.0   \n",
       "...                       ...     ...     ...     ...     ...      ...   \n",
       "10317.000065667.26478     0.0     0.0     0.0     0.0     0.0      1.0   \n",
       "10317.000065673.25853     0.0     0.0     0.0     1.0     0.0      0.0   \n",
       "10317.000065675.25853     0.0     0.0     0.0     2.0     0.0      0.0   \n",
       "10317.000065681.25853     0.0     0.0     0.0     8.0     0.0      0.0   \n",
       "10317.000065684.25853     0.0     0.0     0.0     1.0     0.0   1191.0   \n",
       "\n",
       "                       356052  3392847  263834  294909  ...  4344207  3251419  \\\n",
       "10317.000023139.25853     0.0      0.0     0.0     0.0  ...      0.0      0.0   \n",
       "10317.000027811.25853     0.0      0.0     0.0     0.0  ...      4.0      1.0   \n",
       "10317.000031513.25853     0.0      0.0     0.0     0.0  ...      0.0      0.0   \n",
       "10317.000033280.31503     0.0      0.0     0.0     0.0  ...      0.0      0.0   \n",
       "10317.000038261.33800     0.0      0.0     0.0     0.0  ...      0.0      0.0   \n",
       "...                       ...      ...     ...     ...  ...      ...      ...   \n",
       "10317.000065667.26478     0.0      0.0     0.0     0.0  ...      0.0     67.0   \n",
       "10317.000065673.25853     0.0      0.0     0.0     0.0  ...      0.0      0.0   \n",
       "10317.000065675.25853     0.0      0.0     0.0     0.0  ...      0.0      0.0   \n",
       "10317.000065681.25853     0.0      0.0     0.0     0.0  ...      0.0      4.0   \n",
       "10317.000065684.25853     0.0      0.0     0.0     0.0  ...      6.0      0.0   \n",
       "\n",
       "                       4414478  350503  165257  512791  1718658  577679  \\\n",
       "10317.000023139.25853      0.0     5.0     0.0     0.0      0.0     0.0   \n",
       "10317.000027811.25853      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000031513.25853      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000033280.31503      0.0     4.0     0.0     0.0      0.0     0.0   \n",
       "10317.000038261.33800      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "...                        ...     ...     ...     ...      ...     ...   \n",
       "10317.000065667.26478      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000065673.25853      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000065675.25853      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000065681.25853      1.0     0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000065684.25853      0.0     0.0     0.0     0.0      0.0     0.0   \n",
       "\n",
       "                       3568710  178957  \n",
       "10317.000023139.25853      0.0     0.0  \n",
       "10317.000027811.25853      0.0     0.0  \n",
       "10317.000031513.25853      0.0     0.0  \n",
       "10317.000033280.31503      0.0     0.0  \n",
       "10317.000038261.33800      0.0     0.0  \n",
       "...                        ...     ...  \n",
       "10317.000065667.26478      0.0     0.0  \n",
       "10317.000065673.25853      0.0     0.0  \n",
       "10317.000065675.25853      0.0     0.0  \n",
       "10317.000065681.25853      0.0     0.0  \n",
       "10317.000065684.25853      0.0     0.0  \n",
       "\n",
       "[371 rows x 6904 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "otu_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "otu_df = otu_df.loc[otu_df.index.isin(meta_table_filt[\"sample-id\"])].sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "otu_df.to_csv(\"otu_match.tsv\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "!biom convert -i otu_match.tsv -o otu_match.biom --table-type=\"OTU table\" --to-hdf5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported otu_match.biom as BIOMV210Format to feature_table_match.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime tools import \\\n",
    "  --input-path otu_match.biom\\\n",
    "  --type 'FeatureTable[Frequency]' \\\n",
    "  --input-format BIOMV210Format \\\n",
    "  --output-path feature_table_match.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: feature_table_match_trans.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime feature-table transpose --i-table feature_table_match.qza --o-transposed-feature-table feature_table_match_trans.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'rep_set' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mqiime2\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mplugins\u001b[39;00m \u001b[39mimport\u001b[39;00m alignment, phylogeny\n\u001b[0;32m----> 2\u001b[0m mafft_alignment \u001b[39m=\u001b[39m alignment\u001b[39m.\u001b[39mmethods\u001b[39m.\u001b[39mmafft(rep_set)\n\u001b[1;32m      3\u001b[0m masked_mafft_alignment \u001b[39m=\u001b[39m alignment\u001b[39m.\u001b[39mmethods\u001b[39m.\u001b[39mmask(mafft_alignment\u001b[39m.\u001b[39malignment)\n\u001b[1;32m      4\u001b[0m unrooted_tree \u001b[39m=\u001b[39m phylogeny\u001b[39m.\u001b[39mmethods\u001b[39m.\u001b[39mfasttree(masked_mafft_alignment\u001b[39m.\u001b[39mmasked_alignment)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'rep_set' is not defined"
     ]
    }
   ],
   "source": [
    "from qiime2.plugins import alignment, phylogeny\n",
    "mafft_alignment = alignment.methods.mafft(rep_set)\n",
    "masked_mafft_alignment = alignment.methods.mask(mafft_alignment.alignment)\n",
    "unrooted_tree = phylogeny.methods.fasttree(masked_mafft_alignment.masked_alignment)\n",
    "rooted_tree = phylogeny.methods.midpoint_root(unrooted_tree.tree)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'rooted_tree.qza'"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rooted_tree.rooted_tree.save(\"rooted_tree.qza\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "from qiime2 import Artifact\n",
    "rooted_tree = Artifact.load(\"rooted_tree.qza\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "otu_table = Artifact.load(\"feature_table_match.qza\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'otu_first_analysis.qzv'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from qiime2.plugins import feature_table\n",
    "output_viz = feature_table.visualizers.summarize(otu_table)\n",
    "output_viz.visualization.save(\"otu_first_analysis.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "from qiime2 import Metadata\n",
    "metadata = Metadata.load(\"sample_metadata_filt.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-id</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_cat</th>\n",
       "      <th>refined_sugars</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10317.000001166.25853</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10317.000003100.69016</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10317.000004936.25853</td>\n",
       "      <td>male</td>\n",
       "      <td>50s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10317.000010862.25853</td>\n",
       "      <td>male</td>\n",
       "      <td>20s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10317.000012018.56079</td>\n",
       "      <td>female</td>\n",
       "      <td>50s</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>10317.000109791.74729</td>\n",
       "      <td>male</td>\n",
       "      <td>20s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>10317.000109807.76952</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>10317.000110177.74729</td>\n",
       "      <td>male</td>\n",
       "      <td>30s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>10317.000112912.76952</td>\n",
       "      <td>female</td>\n",
       "      <td>50s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>10317.000113026.81168</td>\n",
       "      <td>female</td>\n",
       "      <td>40s</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>359 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample-id     sex age_cat refined_sugars\n",
       "0    10317.000001166.25853  female     40s             No\n",
       "1    10317.000003100.69016  female     40s             No\n",
       "2    10317.000004936.25853    male     50s             No\n",
       "3    10317.000010862.25853    male     20s             No\n",
       "4    10317.000012018.56079  female     50s             No\n",
       "..                     ...     ...     ...            ...\n",
       "354  10317.000109791.74729    male     20s            Yes\n",
       "355  10317.000109807.76952  female     40s            Yes\n",
       "356  10317.000110177.74729    male     30s            Yes\n",
       "357  10317.000112912.76952  female     50s            Yes\n",
       "358  10317.000113026.81168  female     40s            Yes\n",
       "\n",
       "[359 rows x 4 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata = pd.read_csv('sample_metadata_filt.tsv', sep='\\t')\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      10317.000001166.25853\n",
       "1      10317.000003100.69016\n",
       "2      10317.000004936.25853\n",
       "3      10317.000010862.25853\n",
       "4      10317.000012018.56079\n",
       "               ...          \n",
       "354    10317.000109791.74729\n",
       "355    10317.000109807.76952\n",
       "356    10317.000110177.74729\n",
       "357    10317.000112912.76952\n",
       "358    10317.000113026.81168\n",
       "Name: sample-id, Length: 359, dtype: object"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata[\"sample-id\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "359 x 6904 <class 'biom.table.Table'> with 179471 nonzero entries (7% dense)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import biom\n",
    "\n",
    "b = biom.load_table(\"otu_match.biom\")\n",
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "otu_table_t = Artifact.load(\"feature_table_match_trans.qza\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "359"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.sum(otu_table_t.view(pd.DataFrame).columns == metadata[\"sample-id\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/sklearn/metrics/pairwise.py:1776: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:143: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.056619661459254594 and the largest is 9.316356307490718.\n",
      "  warn(\n",
      "/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:143: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.11963259840261971 and the largest is 8.192855317913514.\n",
      "  warn(\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "None of the sample identifiers match between the metadata and the coordinates. Verify that you are using metadata and coordinates corresponding to the same dataset.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[71], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mqiime2\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mplugins\u001b[39;00m \u001b[39mimport\u001b[39;00m diversity\n\u001b[0;32m----> 2\u001b[0m core_metrics \u001b[39m=\u001b[39m diversity\u001b[39m.\u001b[39;49mpipelines\u001b[39m.\u001b[39;49mcore_metrics_phylogenetic(table \u001b[39m=\u001b[39;49m otu_table_t,\n\u001b[1;32m      3\u001b[0m                                                              phylogeny \u001b[39m=\u001b[39;49m rooted_tree,\n\u001b[1;32m      4\u001b[0m                                                              sampling_depth \u001b[39m=\u001b[39;49m \u001b[39m8200\u001b[39;49m,\n\u001b[1;32m      5\u001b[0m                                                              metadata \u001b[39m=\u001b[39;49m metadata)\n",
      "File \u001b[0;32m<decorator-gen-133>:2\u001b[0m, in \u001b[0;36mcore_metrics_phylogenetic\u001b[0;34m(table, phylogeny, sampling_depth, metadata, with_replacement, n_jobs_or_threads)\u001b[0m\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/qiime2/sdk/action.py:234\u001b[0m, in \u001b[0;36mAction._bind.<locals>.bound_callable\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    230\u001b[0m         warn(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_build_deprecation_message(),\n\u001b[1;32m    231\u001b[0m              \u001b[39mFutureWarning\u001b[39;00m)\n\u001b[1;32m    233\u001b[0m \u001b[39m# Execute\u001b[39;00m\n\u001b[0;32m--> 234\u001b[0m outputs \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_callable_executor_(scope, callable_args,\n\u001b[1;32m    235\u001b[0m                                    output_types, provenance)\n\u001b[1;32m    237\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(outputs) \u001b[39m!=\u001b[39m \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msignature\u001b[39m.\u001b[39moutputs):\n\u001b[1;32m    238\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m    239\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mNumber of callable outputs must match number of \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    240\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39moutputs defined in signature: \u001b[39m\u001b[39m%d\u001b[39;00m\u001b[39m != \u001b[39m\u001b[39m%d\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m\n\u001b[1;32m    241\u001b[0m         (\u001b[39mlen\u001b[39m(outputs), \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msignature\u001b[39m.\u001b[39moutputs)))\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/qiime2/sdk/action.py:475\u001b[0m, in \u001b[0;36mPipeline._callable_executor_\u001b[0;34m(self, scope, view_args, output_types, provenance)\u001b[0m\n\u001b[1;32m    474\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_callable_executor_\u001b[39m(\u001b[39mself\u001b[39m, scope, view_args, output_types, provenance):\n\u001b[0;32m--> 475\u001b[0m     outputs \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_callable(scope\u001b[39m.\u001b[39;49mctx, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mview_args)\n\u001b[1;32m    476\u001b[0m     outputs \u001b[39m=\u001b[39m tuplize(outputs)\n\u001b[1;32m    478\u001b[0m     \u001b[39mfor\u001b[39;00m output \u001b[39min\u001b[39;00m outputs:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/q2_diversity/_core_metrics.py:58\u001b[0m, in \u001b[0;36mcore_metrics_phylogenetic\u001b[0;34m(ctx, table, phylogeny, sampling_depth, metadata, with_replacement, n_jobs_or_threads)\u001b[0m\n\u001b[1;32m     55\u001b[0m emperor_plot \u001b[39m=\u001b[39m ctx\u001b[39m.\u001b[39mget_action(\u001b[39m'\u001b[39m\u001b[39memperor\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mplot\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m     56\u001b[0m core_metrics \u001b[39m=\u001b[39m ctx\u001b[39m.\u001b[39mget_action(\u001b[39m'\u001b[39m\u001b[39mdiversity\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mcore_metrics\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m---> 58\u001b[0m cr \u001b[39m=\u001b[39m core_metrics(table\u001b[39m=\u001b[39;49mtable, sampling_depth\u001b[39m=\u001b[39;49msampling_depth,\n\u001b[1;32m     59\u001b[0m                   metadata\u001b[39m=\u001b[39;49mmetadata, with_replacement\u001b[39m=\u001b[39;49mwith_replacement,\n\u001b[1;32m     60\u001b[0m                   n_jobs\u001b[39m=\u001b[39;49mn_jobs_or_threads)\n\u001b[1;32m     62\u001b[0m faith_pd_vector, \u001b[39m=\u001b[39m faith_pd(table\u001b[39m=\u001b[39mcr\u001b[39m.\u001b[39mrarefied_table,\n\u001b[1;32m     63\u001b[0m                             phylogeny\u001b[39m=\u001b[39mphylogeny)\n\u001b[1;32m     65\u001b[0m dms \u001b[39m=\u001b[39m []\n",
      "File \u001b[0;32m<decorator-gen-487>:2\u001b[0m, in \u001b[0;36mcore_metrics\u001b[0;34m(table, sampling_depth, metadata, with_replacement, n_jobs)\u001b[0m\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/qiime2/sdk/action.py:234\u001b[0m, in \u001b[0;36mAction._bind.<locals>.bound_callable\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    230\u001b[0m         warn(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_build_deprecation_message(),\n\u001b[1;32m    231\u001b[0m              \u001b[39mFutureWarning\u001b[39;00m)\n\u001b[1;32m    233\u001b[0m \u001b[39m# Execute\u001b[39;00m\n\u001b[0;32m--> 234\u001b[0m outputs \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_callable_executor_(scope, callable_args,\n\u001b[1;32m    235\u001b[0m                                    output_types, provenance)\n\u001b[1;32m    237\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(outputs) \u001b[39m!=\u001b[39m \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msignature\u001b[39m.\u001b[39moutputs):\n\u001b[1;32m    238\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m    239\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mNumber of callable outputs must match number of \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    240\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39moutputs defined in signature: \u001b[39m\u001b[39m%d\u001b[39;00m\u001b[39m != \u001b[39m\u001b[39m%d\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m\n\u001b[1;32m    241\u001b[0m         (\u001b[39mlen\u001b[39m(outputs), \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msignature\u001b[39m.\u001b[39moutputs)))\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/qiime2/sdk/action.py:475\u001b[0m, in \u001b[0;36mPipeline._callable_executor_\u001b[0;34m(self, scope, view_args, output_types, provenance)\u001b[0m\n\u001b[1;32m    474\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_callable_executor_\u001b[39m(\u001b[39mself\u001b[39m, scope, view_args, output_types, provenance):\n\u001b[0;32m--> 475\u001b[0m     outputs \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_callable(scope\u001b[39m.\u001b[39;49mctx, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mview_args)\n\u001b[1;32m    476\u001b[0m     outputs \u001b[39m=\u001b[39m tuplize(outputs)\n\u001b[1;32m    478\u001b[0m     \u001b[39mfor\u001b[39;00m output \u001b[39min\u001b[39;00m outputs:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/q2_diversity/_core_metrics.py:42\u001b[0m, in \u001b[0;36mcore_metrics\u001b[0;34m(ctx, table, sampling_depth, metadata, with_replacement, n_jobs)\u001b[0m\n\u001b[1;32m     39\u001b[0m     pcoas \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m pcoa_results\n\u001b[1;32m     41\u001b[0m \u001b[39mfor\u001b[39;00m pcoa \u001b[39min\u001b[39;00m pcoas:\n\u001b[0;32m---> 42\u001b[0m     results \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m emperor_plot(pcoa\u001b[39m=\u001b[39;49mpcoa, metadata\u001b[39m=\u001b[39;49mmetadata)\n\u001b[1;32m     44\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mtuple\u001b[39m(results)\n",
      "File \u001b[0;32m<decorator-gen-496>:2\u001b[0m, in \u001b[0;36mplot\u001b[0;34m(pcoa, metadata, custom_axes, ignore_missing_samples, ignore_pcoa_features)\u001b[0m\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/qiime2/sdk/action.py:234\u001b[0m, in \u001b[0;36mAction._bind.<locals>.bound_callable\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    230\u001b[0m         warn(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_build_deprecation_message(),\n\u001b[1;32m    231\u001b[0m              \u001b[39mFutureWarning\u001b[39;00m)\n\u001b[1;32m    233\u001b[0m \u001b[39m# Execute\u001b[39;00m\n\u001b[0;32m--> 234\u001b[0m outputs \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_callable_executor_(scope, callable_args,\n\u001b[1;32m    235\u001b[0m                                    output_types, provenance)\n\u001b[1;32m    237\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(outputs) \u001b[39m!=\u001b[39m \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msignature\u001b[39m.\u001b[39moutputs):\n\u001b[1;32m    238\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\n\u001b[1;32m    239\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mNumber of callable outputs must match number of \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    240\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39moutputs defined in signature: \u001b[39m\u001b[39m%d\u001b[39;00m\u001b[39m != \u001b[39m\u001b[39m%d\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m\n\u001b[1;32m    241\u001b[0m         (\u001b[39mlen\u001b[39m(outputs), \u001b[39mlen\u001b[39m(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msignature\u001b[39m.\u001b[39moutputs)))\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/qiime2/sdk/action.py:443\u001b[0m, in \u001b[0;36mVisualizer._callable_executor_\u001b[0;34m(self, scope, view_args, output_types, provenance)\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_callable_executor_\u001b[39m(\u001b[39mself\u001b[39m, scope, view_args, output_types, provenance):\n\u001b[1;32m    439\u001b[0m     \u001b[39m# TODO use qiime2.plugin.OutPath when it exists, and update visualizers\u001b[39;00m\n\u001b[1;32m    440\u001b[0m     \u001b[39m# to work with OutPath instead of str. Visualization._from_data_dir\u001b[39;00m\n\u001b[1;32m    441\u001b[0m     \u001b[39m# will also need to be updated to support OutPath instead of str.\u001b[39;00m\n\u001b[1;32m    442\u001b[0m     \u001b[39mwith\u001b[39;00m tempfile\u001b[39m.\u001b[39mTemporaryDirectory(prefix\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mqiime2-temp-\u001b[39m\u001b[39m'\u001b[39m) \u001b[39mas\u001b[39;00m temp_dir:\n\u001b[0;32m--> 443\u001b[0m         ret_val \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_callable(output_dir\u001b[39m=\u001b[39;49mtemp_dir, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mview_args)\n\u001b[1;32m    444\u001b[0m         \u001b[39mif\u001b[39;00m ret_val \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m    445\u001b[0m             \u001b[39mraise\u001b[39;00m \u001b[39mTypeError\u001b[39;00m(\n\u001b[1;32m    446\u001b[0m                 \u001b[39m\"\u001b[39m\u001b[39mVisualizer \u001b[39m\u001b[39m%r\u001b[39;00m\u001b[39m should not return anything. \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    447\u001b[0m                 \u001b[39m\"\u001b[39m\u001b[39mReceived \u001b[39m\u001b[39m%r\u001b[39;00m\u001b[39m as a return value.\u001b[39m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m (\u001b[39mself\u001b[39m, ret_val))\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/q2_emperor/_plot.py:75\u001b[0m, in \u001b[0;36mplot\u001b[0;34m(output_dir, pcoa, metadata, custom_axes, ignore_missing_samples, ignore_pcoa_features)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[39mif\u001b[39;00m pcoa\u001b[39m.\u001b[39mfeatures \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m     71\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mArrows cannot be visualized with the \u001b[39m\u001b[39m'\u001b[39m\u001b[39mplot\u001b[39m\u001b[39m'\u001b[39m\u001b[39m method, \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m     72\u001b[0m                      \u001b[39m\"\u001b[39m\u001b[39muse \u001b[39m\u001b[39m'\u001b[39m\u001b[39mbiplot\u001b[39m\u001b[39m'\u001b[39m\u001b[39m instead, or enable \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m     73\u001b[0m                      \u001b[39m\"\u001b[39m\u001b[39m`ignore_pcoa_features`.\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m---> 75\u001b[0m generic_plot(output_dir, master\u001b[39m=\u001b[39;49mpcoa, metadata\u001b[39m=\u001b[39;49mmetadata, other_pcoa\u001b[39m=\u001b[39;49m\u001b[39mNone\u001b[39;49;00m,\n\u001b[1;32m     76\u001b[0m              ignore_missing_samples\u001b[39m=\u001b[39;49mignore_missing_samples,\n\u001b[1;32m     77\u001b[0m              custom_axes\u001b[39m=\u001b[39;49mcustom_axes, plot_name\u001b[39m=\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39mplot\u001b[39;49m\u001b[39m'\u001b[39;49m)\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/q2_emperor/_plot.py:41\u001b[0m, in \u001b[0;36mgeneric_plot\u001b[0;34m(output_dir, master, metadata, other_pcoa, plot_name, info, custom_axes, settings, ignore_missing_samples, feature_metadata)\u001b[0m\n\u001b[1;32m     38\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m     39\u001b[0m     procrustes \u001b[39m=\u001b[39m [other_pcoa]\n\u001b[0;32m---> 41\u001b[0m viz \u001b[39m=\u001b[39m Emperor(master, mf, feature_mapping_file\u001b[39m=\u001b[39;49mfeature_metadata,\n\u001b[1;32m     42\u001b[0m               ignore_missing_samples\u001b[39m=\u001b[39;49mignore_missing_samples,\n\u001b[1;32m     43\u001b[0m               procrustes\u001b[39m=\u001b[39;49mprocrustes, remote\u001b[39m=\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39m.\u001b[39;49m\u001b[39m'\u001b[39;49m)\n\u001b[1;32m     45\u001b[0m \u001b[39mif\u001b[39;00m custom_axes \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m     46\u001b[0m     viz\u001b[39m.\u001b[39mcustom_axes \u001b[39m=\u001b[39m custom_axes\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/emperor/core.py:220\u001b[0m, in \u001b[0;36mEmperor.__init__\u001b[0;34m(self, ordination, mapping_file, feature_mapping_file, dimensions, remote, jackknifed, procrustes, ignore_missing_samples)\u001b[0m\n\u001b[1;32m    216\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mprocrustes \u001b[39m=\u001b[39m procrustes \u001b[39mif\u001b[39;00m procrustes \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m \u001b[39melse\u001b[39;00m []\n\u001b[1;32m    218\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mmf \u001b[39m=\u001b[39m mapping_file\u001b[39m.\u001b[39mcopy()\n\u001b[0;32m--> 220\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mmf \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_validate_metadata(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mmf, \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mordination\u001b[39m.\u001b[39;49msamples,\n\u001b[1;32m    221\u001b[0m                                   ignore_missing_samples)\n\u001b[1;32m    223\u001b[0m \u001b[39m# if biplots are to be visualized\u001b[39;00m\n\u001b[1;32m    224\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mordination\u001b[39m.\u001b[39mfeatures \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/anaconda3/envs/qiime2-2023.2/lib/python3.8/site-packages/emperor/core.py:304\u001b[0m, in \u001b[0;36mEmperor._validate_metadata\u001b[0;34m(self, metadata, matrix, ignore_missing_samples, kind)\u001b[0m\n\u001b[1;32m    301\u001b[0m difference \u001b[39m=\u001b[39m ordination_elements \u001b[39m-\u001b[39m \u001b[39mset\u001b[39m(metadata\u001b[39m.\u001b[39mindex)\n\u001b[1;32m    303\u001b[0m \u001b[39mif\u001b[39;00m difference \u001b[39m==\u001b[39m ordination_elements:\n\u001b[0;32m--> 304\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\u001b[39m'\u001b[39m\u001b[39mNone of the \u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m identifiers match between the\u001b[39m\u001b[39m'\u001b[39m\n\u001b[1;32m    305\u001b[0m                      \u001b[39m'\u001b[39m\u001b[39m metadata and the coordinates. Verify that you \u001b[39m\u001b[39m'\u001b[39m\n\u001b[1;32m    306\u001b[0m                      \u001b[39m'\u001b[39m\u001b[39mare using metadata and coordinates corresponding\u001b[39m\u001b[39m'\u001b[39m\n\u001b[1;32m    307\u001b[0m                      \u001b[39m'\u001b[39m\u001b[39m to the same dataset.\u001b[39m\u001b[39m'\u001b[39m \u001b[39m%\u001b[39m kind)\n\u001b[1;32m    309\u001b[0m \u001b[39mif\u001b[39;00m difference \u001b[39mand\u001b[39;00m \u001b[39mnot\u001b[39;00m ignore_missing_samples:\n\u001b[1;32m    310\u001b[0m     \u001b[39m# sort the elements so we have a deterministic output\u001b[39;00m\n\u001b[1;32m    311\u001b[0m     difference \u001b[39m=\u001b[39m \u001b[39msorted\u001b[39m([\u001b[39mstr\u001b[39m(i) \u001b[39mfor\u001b[39;00m i \u001b[39min\u001b[39;00m difference])\n",
      "\u001b[0;31mValueError\u001b[0m: None of the sample identifiers match between the metadata and the coordinates. Verify that you are using metadata and coordinates corresponding to the same dataset."
     ]
    }
   ],
   "source": [
    "from qiime2.plugins import diversity\n",
    "core_metrics = diversity.pipelines.core_metrics_phylogenetic(table = otu_table_t,\n",
    "                                                             phylogeny = rooted_tree,\n",
    "                                                             sampling_depth = 8200,\n",
    "                                                             metadata = metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2941399</th>\n",
       "      <th>4425663</th>\n",
       "      <th>353737</th>\n",
       "      <th>363646</th>\n",
       "      <th>1090458</th>\n",
       "      <th>922239</th>\n",
       "      <th>845273</th>\n",
       "      <th>185021</th>\n",
       "      <th>293665</th>\n",
       "      <th>4453060</th>\n",
       "      <th>...</th>\n",
       "      <th>4356697</th>\n",
       "      <th>213762</th>\n",
       "      <th>179905</th>\n",
       "      <th>4324040</th>\n",
       "      <th>333726</th>\n",
       "      <th>199147</th>\n",
       "      <th>1808779</th>\n",
       "      <th>173916</th>\n",
       "      <th>435264</th>\n",
       "      <th>325586</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10317.000001166.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000003100.69016</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000010862.25853</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000012018.56079</th>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000013727.69016</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000109095.74729</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000109791.74729</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000109807.76952</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000110177.74729</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10317.000113026.81168</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>338 rows × 5284 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       2941399  4425663  353737  363646  1090458  922239  \\\n",
       "10317.000001166.25853      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000003100.69016      0.0      1.0     0.0     0.0      0.0     0.0   \n",
       "10317.000010862.25853      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000012018.56079      0.0     28.0     0.0     0.0      0.0     0.0   \n",
       "10317.000013727.69016      0.0      0.0     1.0     0.0      0.0     0.0   \n",
       "...                        ...      ...     ...     ...      ...     ...   \n",
       "10317.000109095.74729      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000109791.74729      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000109807.76952      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000110177.74729      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000113026.81168      0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "\n",
       "                       845273  185021  293665  4453060  ...  4356697  213762  \\\n",
       "10317.000001166.25853     0.0     1.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000003100.69016     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000010862.25853     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000012018.56079     1.0     0.0     0.0      1.0  ...      0.0     0.0   \n",
       "10317.000013727.69016     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "...                       ...     ...     ...      ...  ...      ...     ...   \n",
       "10317.000109095.74729     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000109791.74729     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000109807.76952     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000110177.74729     0.0     1.0     0.0      0.0  ...      0.0     0.0   \n",
       "10317.000113026.81168     0.0     0.0     0.0      0.0  ...      0.0     0.0   \n",
       "\n",
       "                       179905  4324040  333726  199147  1808779  173916  \\\n",
       "10317.000001166.25853     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000003100.69016     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000010862.25853     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000012018.56079     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000013727.69016     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "...                       ...      ...     ...     ...      ...     ...   \n",
       "10317.000109095.74729     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000109791.74729     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000109807.76952     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000110177.74729     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "10317.000113026.81168     0.0      0.0     0.0     0.0      0.0     0.0   \n",
       "\n",
       "                       435264  325586  \n",
       "10317.000001166.25853     0.0     0.0  \n",
       "10317.000003100.69016     0.0     0.0  \n",
       "10317.000010862.25853     0.0     0.0  \n",
       "10317.000012018.56079     0.0     0.0  \n",
       "10317.000013727.69016     0.0     0.0  \n",
       "...                       ...     ...  \n",
       "10317.000109095.74729     0.0     0.0  \n",
       "10317.000109791.74729     0.0     0.0  \n",
       "10317.000109807.76952     0.0     0.0  \n",
       "10317.000110177.74729     0.0     0.0  \n",
       "10317.000113026.81168     0.0     0.0  \n",
       "\n",
       "[338 rows x 5284 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "core_metrics.rarefied_table.view(pd.DataFrame)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>feature</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10317.000001166.25853</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10317.000003100.69016</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10317.000010862.25853</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10317.000012018.56079</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10317.000013727.69016</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785987</th>\n",
       "      <td>10317.000109095.74729</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785988</th>\n",
       "      <td>10317.000109791.74729</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785989</th>\n",
       "      <td>10317.000109807.76952</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785990</th>\n",
       "      <td>10317.000110177.74729</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785991</th>\n",
       "      <td>10317.000113026.81168</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1785992 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index  feature  Value\n",
       "0        10317.000001166.25853  2941399    0.0\n",
       "1        10317.000003100.69016  2941399    0.0\n",
       "2        10317.000010862.25853  2941399    0.0\n",
       "3        10317.000012018.56079  2941399    0.0\n",
       "4        10317.000013727.69016  2941399    0.0\n",
       "...                        ...      ...    ...\n",
       "1785987  10317.000109095.74729   325586    0.0\n",
       "1785988  10317.000109791.74729   325586    0.0\n",
       "1785989  10317.000109807.76952   325586    0.0\n",
       "1785990  10317.000110177.74729   325586    0.0\n",
       "1785991  10317.000113026.81168   325586    0.0\n",
       "\n",
       "[1785992 rows x 3 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "core_metrics_rarefied_table_df = core_metrics.rarefied_table.view(pd.DataFrame).reset_index().melt(id_vars=['index'],\n",
    "                                                          var_name=\"feature\", \n",
    "                                                          value_name=\"Value\")\n",
    "core_metrics_rarefied_table_df.to_csv('core_metrics_rarefied_table_df.csv')\n",
    "core_metrics_rarefied_table_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>feature</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10317.000001166.25853</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10317.000003100.69016</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10317.000010862.25853</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10317.000012018.56079</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10317.000013727.69016</td>\n",
       "      <td>2941399</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785987</th>\n",
       "      <td>10317.000109095.74729</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785988</th>\n",
       "      <td>10317.000109791.74729</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785989</th>\n",
       "      <td>10317.000109807.76952</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785990</th>\n",
       "      <td>10317.000110177.74729</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1785991</th>\n",
       "      <td>10317.000113026.81168</td>\n",
       "      <td>325586</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1785992 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         index  feature  Value\n",
       "0        10317.000001166.25853  2941399    0.0\n",
       "1        10317.000003100.69016  2941399    0.0\n",
       "2        10317.000010862.25853  2941399    0.0\n",
       "3        10317.000012018.56079  2941399    0.0\n",
       "4        10317.000013727.69016  2941399    0.0\n",
       "...                        ...      ...    ...\n",
       "1785987  10317.000109095.74729   325586    0.0\n",
       "1785988  10317.000109791.74729   325586    0.0\n",
       "1785989  10317.000109807.76952   325586    0.0\n",
       "1785990  10317.000110177.74729   325586    0.0\n",
       "1785991  10317.000113026.81168   325586    0.0\n",
       "\n",
       "[1785992 rows x 3 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "core_metrics_rarefied_table_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'Results' object has no attribute 'observed_otus_vector'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[121], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m observed_otus_vector_group_significance \u001b[39m=\u001b[39m diversity\u001b[39m.\u001b[39mactions\u001b[39m.\u001b[39malpha_group_significance(core_metrics\u001b[39m.\u001b[39;49mobserved_otus_vector,\n\u001b[1;32m      2\u001b[0m                                                                                      metadata)\n\u001b[1;32m      3\u001b[0m observed_otus_vector_group_significance\u001b[39m.\u001b[39mvisualization\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'Results' object has no attribute 'observed_otus_vector'"
     ]
    }
   ],
   "source": [
    "observed_otus_vector_group_significance = diversity.actions.alpha_group_significance(core_metrics.observed_otus_vector,\n",
    "                                                                                     metadata)\n",
    "observed_otus_vector_group_significance.visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'shannon.qzv'"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shannon_vector_group_significance = diversity.actions.alpha_group_significance(core_metrics.shannon_vector,\n",
    "                                                                         metadata)\n",
    "shannon_vector_group_significance.visualization.save(\"shannon.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'faith_pd.qzv'"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "faith_pd_group_significance = diversity.actions.alpha_group_significance(core_metrics.faith_pd_vector,\n",
    "                                                                         metadata)\n",
    "faith_pd_group_significance.visualization.save(\"faith_pd.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'evenness_group.qzv'"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "evenness_group_significance = diversity.actions.alpha_group_significance(core_metrics.evenness_vector,\n",
    "                                                                         metadata)\n",
    "evenness_group_significance.visualization.save(\"evenness_group.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'bray_curtis.qzv'"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bray_curtis_distance_significance = diversity.actions.beta_group_significance(core_metrics.bray_curtis_distance_matrix,\n",
    "                                                                           metadata.get_column('refined_sugars'))\n",
    "bray_curtis_distance_significance.visualization.save(\"bray_curtis.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'jaccard_beta.qzv'"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "jaccard_distance_significance = diversity.actions.beta_group_significance(core_metrics.jaccard_distance_matrix,\n",
    "                                                                           metadata.get_column('refined_sugars'))\n",
    "jaccard_distance_significance.visualization.save(\"jaccard_beta.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'unweighted_unifrac.qzv'"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unweighted_unifrac_significance = diversity.actions.beta_group_significance(core_metrics.unweighted_unifrac_distance_matrix,\n",
    "                                                                           metadata.get_column('refined_sugars'))\n",
    "unweighted_unifrac_significance.visualization.save(\"unweighted_unifrac.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'weighted_unifrac.qzv'"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weighted_unifrac_significance = diversity.actions.beta_group_significance(core_metrics.weighted_unifrac_distance_matrix,\n",
    "                                                                           metadata.get_column('refined_sugars'))\n",
    "weighted_unifrac_significance.visualization.save(\"weighted_unifrac.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'emperor.qzv'"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from qiime2.plugins import emperor\n",
    "emperor_plot = emperor.visualizers.plot(core_metrics.unweighted_unifrac_pcoa_results,\n",
    "                                        metadata)\n",
    "                                        #custom_axes=['DaysSinceExperimentStart'])\n",
    "emperor_plot.visualization.save('emperor.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.15"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
